// @argent/x-ui/dist/src/components/transactions/NftDetails.js
import { jsxs as s, jsx as o } from "react/jsx-runtime";
import { getNftPicture as x } from "../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/nfts/utils.js";
import { prettifyCurrencyValue as T } from "../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/prettifyCurrencyValue.js";
import { prettifyTokenAmount as w } from "../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/prettifyTokenAmount.js";
import { generateAvatarImage as F, getColor as W } from "../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/utils/avatarImage.js";
import v from "../../features/icons/WarningCircleSecondaryIcon.js";
import { useColorMode as I, AccordionButton as P, Flex as b, Image as S } from "@chakra-ui/react";
import { useMemo as h, useCallback as U } from "react";
import { useCurrency as j } from "../../features/providers/CurrencyProvider.js";
import { TextWithAmount as A } from "../TextWithAmount.js";
import { P3 as _, P2 as B, Label2Bold as M } from "../typography.generated.js";
import { parseCurrencyAbs as R } from "../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/bigdecimal/utils.js";
const p = "Unknown NFT", V = ({ contractAddress: e, safe: C, dataIndex: c, totalData: u, isDisabled: d, isDefaultNetwork: N, amount: a, usdValue: m, nft: r, collection: i }) => {
  const { colorMode: g } = I(), f = h(() => g === "dark", [g]), l = j(), n = U((t, k) => F(k ?? p, {
    background: W(t)
  }), []), y = h(() => {
    if (r && r.name)
      return {
        name: r.name,
        description: r.description,
        image_url_copy: r.image_url_copy ?? r.image_uri ?? n(e, r.name)
      };
    if (i) {
      const t = i.name ?? "Unknown Collection";
      return {
        name: `${t} NFT`,
        description: `${t} NFT`,
        image_url_copy: i.imageUri ?? n(e)
      };
    }
    return {
      name: p,
      description: p,
      image_url_copy: n(e)
    };
  }, [e, i, r, n]);
  return /* @__PURE__ */ s(P, { display: "flex", width: "100%", justifyContent: "space-between", outline: "none", px: "3", pb: c !== u - 1 ? "3" : "3.5", _expanded: {
    backgroundColor: f ? "neutrals.700" : "gray.50",
    pb: "3.5"
  }, disabled: d, _disabled: {
    cursor: "auto",
    opacity: 1
  }, _hover: {
    backgroundColor: d ? "" : f ? "neutrals.700" : "gray.50",
    borderBottomRadius: c === u - 1 ? "xl" : "0"
  }, children: [
    /* @__PURE__ */ s(b, { alignItems: "center", gap: "2", children: [
      /* @__PURE__ */ o(S, { src: x({ image_url_copy: y.image_url_copy }) || "", w: "5", h: "5", borderRadius: "base" }),
      /* @__PURE__ */ o(_, { fontWeight: "bold", color: "text-primary", children: y.name }),
      !C && /* @__PURE__ */ o(B, { color: "error.500", fontWeight: "bold", mt: "0.25", children: /* @__PURE__ */ o(v, {}) })
    ] }),
    /* @__PURE__ */ s(b, { direction: "column", gap: "0.5", alignItems: "flex-end", children: [
      /* @__PURE__ */ o(A, { amount: a, children: /* @__PURE__ */ o(_, { color: a < 0n ? "error.500" : "secondary.500", fontWeight: "bold", children: w({
        amount: a,
        decimals: 0,
        symbol: "NFT",
        showPlusSign: !0
      }) }) }),
      N && !!m && m !== "0" && /* @__PURE__ */ o(M, { color: "neutrals.300", children: T(R(m).toString(), l == null ? void 0 : l.symbol) })
    ] })
  ] });
};
export {
  V as NftDetails
};
