// @tronweb3/walletconnect-tron/lib/esm/utils.js
import { defineChain } from '@reown/appkit/networks';
export var WalletConnectChainID;
(function (WalletConnectChainID) {
    WalletConnectChainID["Mainnet"] = "tron:0x2b6653dc";
    WalletConnectChainID["Shasta"] = "tron:0x94a9059e";
    WalletConnectChainID["Nile"] = "tron:0xcd8690dc";
})(WalletConnectChainID || (WalletConnectChainID = {}));
export const mainnet = defineChain({
    id: '0x2b6653dc',
    caipNetworkId: 'tron:0x2b6653dc',
    chainNamespace: 'tron',
    name: 'Tron Mainnet',
    nativeCurrency: {
        decimals: 6,
        name: 'TRX',
        symbol: 'TRX'
    },
    rpcUrls: {
        default: {
            http: ['https://api.trongrid.io/jsonrpc'],
            webSocket: ['']
        }
    },
    blockExplorers: {
        default: { name: 'Tron BlockChain Explorer', url: 'https://tronscan.org/' }
    },
    contracts: {}
});
export const nileTestnet = defineChain({
    id: '0xcd8690dc',
    caipNetworkId: 'tron:0xcd8690dc',
    chainNamespace: 'tron',
    name: 'Tron Nile Testnet',
    nativeCurrency: {
        decimals: 6,
        name: 'TRX',
        symbol: 'TRX'
    },
    rpcUrls: {
        default: {
            http: [''],
            webSocket: ['']
        }
    },
    blockExplorers: {
        default: { name: 'Tron BlockChain Explorer', url: 'https://nile.tronscan.org/' }
    },
    contracts: {}
});
export const shastaTestnet = defineChain({
    id: '0x94a9059e',
    caipNetworkId: 'tron:0x94a9059e',
    chainNamespace: 'tron',
    name: 'Tron Shasta Testnet',
    nativeCurrency: {
        decimals: 6,
        name: 'TRX',
        symbol: 'TRX'
    },
    rpcUrls: {
        default: {
            http: [''],
            webSocket: ['']
        }
    },
    blockExplorers: {
        default: { name: 'Tron BlockChain Explorer', url: 'https://shasta.tronscan.org/' }
    },
    contracts: {}
});
export const NETWORK_MAP = new Map([
    [WalletConnectChainID.Mainnet, mainnet],
    [WalletConnectChainID.Nile, nileTestnet],
    [WalletConnectChainID.Shasta, shastaTestnet]
]);
