// @argent/x-ui/dist/src/components/Banner.js
import { jsxs as s, jsx as o } from "react/jsx-runtime";
import h from "../features/icons/CrossSecondaryIcon.js";
import { Box as c } from "@chakra-ui/react";
import { ButtonCell as u } from "./CellStack.js";
import "./Typography.js";
import { P2 as b, P3 as g } from "./typography.generated.js";
const m = ({ dark: e, ...t }) => /* @__PURE__ */ o(c, { position: "absolute", top: "0", left: "0", right: "0", bottom: "0", ...t }), k = ({ dark: e, ...t }) => /* @__PURE__ */ o(c, { position: "absolute", top: "0", right: "0", width: "24px", height: "24px", zIndex: "1", background: "black", color: "white", borderRadius: "50%", m: 1, display: "flex", justifyContent: "center", alignItems: "center", ...t, children: /* @__PURE__ */ o(h, { fontSize: "xs", "data-testid": "close-banner" }) }), F = ({ href: e, backgroundImageUrl: t, title: n = "Discover", subTitle: d = "Starknet dapps", onClose: i, dark: r, onClick: l, ...p }) => /* @__PURE__ */ s(
  u,
  {
    width: "100%",
    overflow: "hidden",
    position: "relative",
    rightIcon: null,
    bgColor: "#FFFFFF",
    background: `url("${t}")`,
    backgroundSize: "cover",
    backgroundPosition: "right top",
    backgroundRepeat: "no-repeat",
    as: "a",
    title: n,
    target: "_blank",
    _hover: {
      background: `url("${t}")`,
      backgroundSize: "cover",
      backgroundPosition: "right top",
      backgroundRepeat: "no-repeat",
      backgroundColor: "white"
    },
    href: e,
    onClick: l,
    whiteSpace: "initial",
    ...p,
    children: [
      /* @__PURE__ */ o(m, { dark: r }),
      /* @__PURE__ */ o(b, { zIndex: "1", color: r ? "white" : "black", fontWeight: "extrabold", children: n }),
      /* @__PURE__ */ o(g, { zIndex: "1", color: r ? "white" : "black", fontWeight: "medium", children: d }),
      /* @__PURE__ */ o(k, { onClick: (a) => {
        a.preventDefault(), a.stopPropagation(), i == null || i();
      }, dark: r })
    ]
  }
);
export {
  F as Banner
};
