// @argent/x-ui/dist/src/components/DetailAccordion.js
import { jsx as o, jsxs as l, Fragment as m } from "react/jsx-runtime";
import { buttonBaseStyle as s, buttonHoverStyle as f } from "../features/theme/buttonTheme.js";
import { typographyStyles as i } from "../features/theme/typographyStyles.generated.js";
import { Flex as e, Accordion as x, AccordionItem as b, useAccordionItemState as g, AccordionButton as _, AccordionPanel as w, Divider as y, Box as d } from "@chakra-ui/react";
import { CopyTooltip as p } from "./CopyTooltip.js";
import "./Typography.js";
import { P3 as k } from "./typography.generated.js";
const I = (r) => /* @__PURE__ */ o(e, { backgroundColor: "surface-elevated-hover", px: "3", py: "2.5", borderTopRadius: "xl", mb: "1px", className: "detail-accordion__header", children: /* @__PURE__ */ o(e, { ...i.P3, alignItems: "center", gap: "1", fontWeight: "medium", color: "text-secondary", ...r }) }), j = (r) => /* @__PURE__ */ o(x, { allowToggle: !0, backgroundColor: "surface-elevated", borderRadius: "xl", fontWeight: "medium", sx: {
  ".detail-accordion__header + &": {
    borderTopRadius: "none"
  }
}, overflow: "hidden", ...r }), B = (r) => /* @__PURE__ */ o(
  b,
  {
    border: "none",
    color: "white",
    sx: {
      ".chakra-accordion__button": {
        py: 1.5,
        '&[aria-expanded="true"]': {
          pt: 3
        }
      },
      _notFirst: {
        '.chakra-accordion__button[aria-expanded="true"]': {
          mt: 1.5
        }
      },
      _notLast: {
        '.chakra-accordion__button[aria-expanded="true"] + .chakra-collapse > .chakra-accordion__panel': {
          mb: 1.5
        }
      },
      _first: {
        ".chakra-accordion__button": {
          pt: 3,
          '&[aria-expanded="true"]': {
            mt: 0
          }
        }
      },
      _last: {
        ".chakra-accordion__button": {
          pb: 3,
          '&[aria-expanded="true"]': {
            pb: 1.5
          }
        }
      }
    },
    ...r
  }
), O = ({ label: r, value: t, children: a, ...n }) => {
  const { isDisabled: c } = g();
  return /* @__PURE__ */ l(_, { transitionProperty: "box-shadow, margin, padding, background", transitionDuration: "fast", display: "flex", width: "full", justifyContent: "space-between", outline: "none", px: "3", _expanded: {
    backgroundColor: "surface-elevated-hover"
  }, boxShadow: s.boxShadow, _disabled: {
    cursor: "auto",
    opacity: 1
  }, _hover: {
    boxShadow: c ? s.boxShadow : f.boxShadow
  }, ...n, children: [
    r && /* @__PURE__ */ o(k, { fontWeight: "medium", children: r }),
    t && /* @__PURE__ */ o(e, { ...i.P3, color: "neutrals.400", fontWeight: "medium", noOfLines: 1, children: t }),
    a
  ] });
}, R = (r) => /* @__PURE__ */ l(w, { backgroundColor: "surface-elevated", px: "3", pb: "0", children: [
  /* @__PURE__ */ o(y, { color: "black", opacity: "1" }),
  /* @__PURE__ */ o(e, { flexDirection: "column", gap: "3", py: "3", ...r })
] }), T = ({ header: r, label: t, copyLabel: a, value: n, copyValue: c, children: u, ...h }) => /* @__PURE__ */ l(e, { justifyContent: "space-between", gap: "2", ...h, children: [
  r && /* @__PURE__ */ o(e, { ...i.P3, alignItems: "center", gap: "1", color: "neutrals.300", fontWeight: "medium", children: r }),
  t && /* @__PURE__ */ o(e, { ...i.P3, alignItems: "center", gap: "1", color: "neutrals.400", fontWeight: "medium", children: a ? /* @__PURE__ */ o(p, { copyValue: a, children: /* @__PURE__ */ o(d, { _hover: {
    color: "text-primary"
  }, cursor: "pointer", transitionProperty: "color", transitionDuration: "fast", children: t }) }) : /* @__PURE__ */ o(m, { children: t }) }),
  n && /* @__PURE__ */ o(e, { ...i.P3, color: "neutrals.500", fontWeight: "medium", maxWidth: "70%", noOfLines: 1, children: c ? /* @__PURE__ */ o(p, { copyValue: c, children: /* @__PURE__ */ o(d, { _hover: {
    color: "text-primary"
  }, cursor: "pointer", transitionProperty: "color", transitionDuration: "fast", whiteSpace: "nowrap", textOverflow: "ellipsis", overflow: "hidden", minWidth: "0", children: n }) }) : /* @__PURE__ */ o(d, { whiteSpace: "nowrap", textOverflow: "ellipsis", overflow: "hidden", minWidth: "0", children: n }) }),
  u
] });
export {
  j as DetailAccordion,
  O as DetailAccordionButton,
  I as DetailAccordionHeader,
  B as DetailAccordionItem,
  R as DetailAccordionPanel,
  T as DetailAccordionRow
};
