// @argent/x-ui/dist/src/components/FlowHeader.js
import { jsxs as s, jsx as r } from "react/jsx-runtime";
import g from "../features/icons/CheckmarkSecondaryIcon.js";
import { Center as m, Circle as f, Spinner as p } from "@chakra-ui/react";
import "./Typography.js";
import { H3 as d, P2 as u } from "./typography.generated.js";
const b = {
  default: {
    bg: "neutrals.700",
    fg: "white"
  },
  primary: {
    bg: "primary.500",
    fg: "neutrals.900"
  },
  removed: {
    bg: "error.900",
    fg: "error.400"
  },
  success: {
    bg: "success.900",
    fg: "success.500"
  },
  warning: {
    bg: "warning.500",
    fg: "neutrals.900"
  },
  danger: {
    bg: "error.900",
    fg: "error.400"
  }
}, y = ({ title: c, subtitle: e, variant: i = "default", size: n = "md", icon: a = g, isLoading: o, ...l }) => {
  const t = b[i];
  return /* @__PURE__ */ s(m, { flexDirection: "column", textAlign: "center", pt: 4, pb: 8, px: 4, ...l, children: [
    /* @__PURE__ */ r(f, { size: n === "md" ? 18 : 24, bg: o ? "black" : t.bg, children: o ? /* @__PURE__ */ r(p, { size: "xl" }) : /* @__PURE__ */ r(a, { fontSize: n === "md" ? "4xl" : "5xl", color: t.fg }) }),
    /* @__PURE__ */ r(d, { pt: 4, children: c }),
    e && /* @__PURE__ */ r(u, { pt: 2, color: "neutrals.100", children: e })
  ] });
};
export {
  y as FlowHeader
};
