// @argent/x-ui/dist/src/components/LabelValue.js
import { jsx as r, jsxs as d } from "react/jsx-runtime";
import { isAddress as u, formatFullAddress as h } from "../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/chains/starknet/address.js";
import { formatAddress as w } from "../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/utils/address.js";
import { typographyStyles as x } from "../features/theme/typographyStyles.generated.js";
import { Flex as n, Tooltip as s, Text as l } from "@chakra-ui/react";
import { isString as g } from "lodash-es";
import { useMemo as b, isValidElement as y, useState as A, useRef as S, useCallback as V } from "react";
function W(e) {
  return /* @__PURE__ */ r(n, { direction: "column", w: "full", ...e });
}
function C({ label: e, value: t }) {
  const i = b(() => t === void 0 ? null : y(t) ? t : g(t) && u(t) ? /* @__PURE__ */ r(s, { label: h(t), placement: "top", children: /* @__PURE__ */ r(l, { ml: "auto", textAlign: "right", children: w(t) }) }) : /* @__PURE__ */ r(F, { ml: "auto", textAlign: "right", value: t }), [t]);
  return /* @__PURE__ */ d(n, { alignItems: "center", flex: 1, gap: 1, _notFirst: { mt: 2 }, ...x.P3, children: [
    /* @__PURE__ */ r(l, { color: "text-secondary", whiteSpace: "nowrap", children: e }),
    i
  ] });
}
function F({ value: e, ...t }) {
  const [i, c] = A(!1), o = S(null), a = V((f) => {
    if (o.current = f, o != null && o.current) {
      const { scrollWidth: m, offsetWidth: p } = o.current;
      c(m > p);
    }
  }, []);
  return /* @__PURE__ */ r(s, { label: e, placement: "top", isDisabled: !i, children: /* @__PURE__ */ r(l, { ref: a, overflow: "hidden", textOverflow: "ellipsis", whiteSpace: "nowrap", ...t, children: e }) });
}
export {
  F as LabelValueAutoTooltip,
  C as LabelValueRow,
  W as LabelValueStack
};
