// @argent/x-ui/dist/src/components/NavigationBar.js
import { jsx as t, jsxs as p, Fragment as g } from "react/jsx-runtime";
import w from "../features/icons/ArrowLeftPrimaryIcon.js";
import B from "../features/icons/CrossSecondaryIcon.js";
import x from "../features/icons/PlusSecondaryIcon.js";
import { Button as k, chakra as v, Center as e, Flex as u, SkeletonCircle as d, SkeletonText as S } from "@chakra-ui/react";
import { useTranslation as h } from "react-i18next";
import "./Typography.js";
import { useNavigateBack as m } from "../hooks/useNavigateBack.js";
import { H5 as y } from "./typography.generated.js";
const i = 14, C = v(u, {
  baseStyle: {
    alignItems: "center",
    bottom: "initial",
    h: i,
    px: 3,
    py: 2,
    transitionProperty: "background",
    transitionDuration: "fast",
    flexShrink: 0,
    zIndex: 1
  }
}), c = ({ children: o, ...n }) => /* @__PURE__ */ t(k, { color: "icon-secondary", colorScheme: "transparent", padding: "1.5", fontSize: "xl", size: "auto", rounded: "full", ...n, children: o }), O = (o) => {
  const { t: n } = h(), r = m();
  return /* @__PURE__ */ t(c, { "aria-label": n("navigationBar.back"), onClick: r, ...o, "data-testid": "navigation-bar-back-button", children: /* @__PURE__ */ t(w, {}) });
}, D = (o) => {
  const { t: n } = h(), r = m();
  return /* @__PURE__ */ t(c, { "aria-label": n("navigationBar.close"), onClick: r, ...o, "data-testid": "navigation-bar-close-button", children: /* @__PURE__ */ t(B, {}) });
}, q = (o) => /* @__PURE__ */ t(c, { "aria-label": "add", ...o, "data-testid": "navigation-bar-add-button", children: /* @__PURE__ */ t(x, {}) }), E = ({ isAbsolute: o, leftButton: n, rightButton: r, title: a, scroll: s, children: b, ...f }) => {
  const l = ((s == null ? void 0 : s.scrollTop) ?? 0) <= 16;
  return /* @__PURE__ */ p(C, { bg: l ? "transparent" : "surface-default", boxShadow: l ? "none" : "menu", position: o ? "absolute" : "relative", w: "full", ...f, children: [
    a && /* @__PURE__ */ t(e, { position: "absolute", left: i, right: i, top: 0, bottom: 0, children: typeof a == "string" ? /* @__PURE__ */ t(y, { w: "full", overflow: "hidden", textOverflow: "ellipsis", whiteSpace: "nowrap", textAlign: "center", children: a }) : /* @__PURE__ */ t(g, { children: a }) }),
    n && /* @__PURE__ */ t(e, { position: "absolute", left: 3, top: 2, bottom: 2, children: n }),
    r && /* @__PURE__ */ t(e, { position: "absolute", right: 3, top: 2, bottom: 2, children: r }),
    b
  ] });
}, G = () => /* @__PURE__ */ p(u, { alignItems: "center", justifyContent: "space-between", px: 3, py: 2, height: i, children: [
  /* @__PURE__ */ t(e, { w: 8, h: 8, children: /* @__PURE__ */ t(d, { w: 5, h: 5 }) }),
  /* @__PURE__ */ t(S, { h: 2, w: "33.3%", noOfLines: 1 }),
  /* @__PURE__ */ t(e, { w: 8, h: 8, children: /* @__PURE__ */ t(d, { w: 5, h: 5 }) })
] });
export {
  q as BarAddButton,
  O as BarBackButton,
  D as BarCloseButton,
  c as BarIconButton,
  E as NavigationBar,
  i as NavigationBarHeight,
  G as NavigationBarSkeleton
};
