// @argent/x-ui/dist/src/components/Typography.js
import { chakra as a } from "@chakra-ui/react";
import { L1Bold as e } from "./typography.generated.js";
import { B1 as i, B2 as g, B3 as P, Button1 as H, Button2 as n, Button3 as p, FeatureTitle as L, H1 as m, H2 as h, H3 as u, H3Medium as b, H4 as s, H4Medium as M, H5 as c, Heading1 as f, Heading2 as x, Heading3 as F, Heading3Medium as T, Heading4 as k, Heading4Medium as v, Heading5 as y, L1 as E, L2 as S, L2Bold as j, L3 as q, L3Bold as w, Label1 as z, Label1Bold as A, Label2 as C, Label2Bold as D, Label3 as G, Label3Bold as I, MassiveTitle as J, P1 as K, P2 as N, P2Bold as O, P3 as Q, P3Bold as R, P4 as U, P4Bold as V, Paragraph1 as W, Paragraph2 as X, Paragraph2Bold as Y, Paragraph3 as Z, Paragraph3Bold as _, Paragraph4 as $, Paragraph4Bold as aa } from "./typography.generated.js";
const d = a("pre"), l = a(e, {
  baseStyle: {
    color: "text-danger"
  }
});
export {
  i as B1,
  g as B2,
  P as B3,
  H as Button1,
  n as Button2,
  p as Button3,
  L as FeatureTitle,
  l as FieldError,
  m as H1,
  h as H2,
  u as H3,
  b as H3Medium,
  s as H4,
  M as H4Medium,
  c as H5,
  f as Heading1,
  x as Heading2,
  F as Heading3,
  T as Heading3Medium,
  k as Heading4,
  v as Heading4Medium,
  y as Heading5,
  E as L1,
  e as L1Bold,
  S as L2,
  j as L2Bold,
  q as L3,
  w as L3Bold,
  z as Label1,
  A as Label1Bold,
  C as Label2,
  D as Label2Bold,
  G as Label3,
  I as Label3Bold,
  J as MassiveTitle,
  K as P1,
  N as P2,
  O as P2Bold,
  Q as P3,
  R as P3Bold,
  U as P4,
  V as P4Bold,
  W as Paragraph1,
  X as Paragraph2,
  Y as Paragraph2Bold,
  Z as Paragraph3,
  _ as Paragraph3Bold,
  $ as Paragraph4,
  aa as Paragraph4Bold,
  d as Pre
};
