// @argent/x-ui/dist/src/features/theme/badgeTheme.js
import { defineStyleConfig as t } from "@chakra-ui/react";
import { typographyStyles as o } from "./typographyStyles.generated.js";
const a = t({
  baseStyle: () => ({
    textTransform: "initial"
  }),
  sizes: {
    /** span with font-size so need to specify px */
    auto: {},
    md: {
      ...o.L1Bold,
      px: "4px",
      py: "2px",
      rounded: "4px",
      letterSpacing: "normal",
      fontWeight: "bold"
    }
  },
  variants: {
    outline: {},
    solid: (e) => {
      const { colorScheme: r } = e;
      switch (r) {
        case "primary":
          return {
            bg: "button-primary",
            color: "white.white"
          };
        case "secondary":
          return {
            bg: "button-secondary",
            color: "text-primary"
          };
        case "info":
          return {
            bg: "button-info",
            color: "white.white"
          };
        case "warning":
          return {
            bg: "button-warning",
            color: "black.black"
          };
        case "danger":
          return {
            bg: "surface-danger-vibrant",
            color: "white.white"
          };
      }
      return {
        bg: `${r}.500`,
        color: "white"
      };
    }
  },
  defaultProps: {
    size: "md",
    variant: "solid",
    colorScheme: "primary"
  }
});
export {
  a as badgeTheme
};
