// @argent/x-ui/dist/src/features/theme/colors.js
import { merge as e } from "lodash-es";
import { colors as a } from "./colors.generated.js";
import { makeColorVariants as r } from "./utilities/makeColorVariants.js";
const t = e({}, a, {
  transparent: "transparent",
  current: "currentColor",
  black: "#000000",
  "black.10": "rgba(0,0,0,0.1)",
  white: "#ffffff",
  gray: r(a["neutrals.600"]),
  primary: r(a["primary.orange.600"]),
  primaryExtraDark: r(a["primary.orange.1000"]),
  secondary: r("#08a681"),
  secondaryDark: "#068063",
  tertiary: r(a["neutrals.700"]),
  accent: r("#197aa6"),
  warn: r("#ffbf3d"),
  //primary.yellow.400
  "warn-high": r("#f36a3d"),
  // primary.orange.600
  info: r("#0078a4"),
  //primary.blue.800
  danger: r(a["primary.red.dark"]),
  warning: r("#f4bc54"),
  warningDark: r(a["primary.yellow.800"]),
  success: r("#3ed373"),
  error: r("#cc3247"),
  skyBlue: r(a["secondary.sky.blue"]),
  /** used by Chakra Alert and Toast */
  blue: r("#0078a4"),
  // primary.blue.800
  orange: r("#f4bc54"),
  red: r("#cc3247"),
  green: r("#08a681")
});
export {
  t as colors
};
