// @argent/x-ui/dist/src/features/theme/listTheme.js
import { listAnatomy as e } from "@chakra-ui/anatomy";
import { createMultiStyleConfigHelpers as t } from "@chakra-ui/styled-system";
import { typographyStyles as r } from "./typographyStyles.generated.js";
const { definePartsStyle: o, defineMultiStyleConfig: n } = t(e.keys), i = {
  bordered: o(() => ({
    container: {
      border: "1px solid",
      borderColor: "neutrals.600",
      rounded: "lg",
      counterReset: "borderedList",
      marginInlineStart: 0,
      margin: 0,
      ...r.P3
    },
    item: {
      display: "flex",
      alignItems: "center",
      py: 2,
      px: 3,
      gap: 3,
      _notLast: {
        borderBottom: "1px solid",
        borderColor: "neutrals.700"
      },
      "&::before": {
        display: "flex",
        content: "counter(borderedList)",
        backgroundColor: "neutrals.700",
        counterIncrement: "borderedList",
        rounded: "full",
        width: 6,
        height: 6,
        flexShrink: 0,
        textAlign: "center",
        alignItems: "center",
        justifyContent: "center"
      }
    }
  }))
}, a = n({ variants: i });
export {
  a as listTheme
};
