// @argent/x-ui/dist/src/features/theme/theme.js
import { extendTheme as m } from "@chakra-ui/react";
import { merge as r } from "lodash-es";
import { accordionTheme as t } from "./accordionTheme.js";
import { alertTheme as i } from "./alertTheme.js";
import { badgeTheme as p } from "./badgeTheme.js";
import { breakpoints as n } from "./breakpoints.js";
import { buttonTheme as s } from "./buttonTheme.js";
import { colors as o } from "./colors.js";
import { dividerTheme as f } from "./dividerTheme.js";
import { iconButtonTheme as h } from "./iconButtonTheme.js";
import { inputTheme as a } from "./inputTheme.js";
import { listTheme as T } from "./listTheme.js";
import { menuTheme as c } from "./menuTheme.js";
import { pinInputTheme as l } from "./pinInputTheme.js";
import { progressTheme as d } from "./progressTheme.js";
import { semanticTokens as e } from "./semanticTokens.js";
import { shadows as g } from "./shadows.js";
import { sizes as u } from "./sizes.js";
import { skeletonTheme as y } from "./skeleton.js";
import { spacing as b } from "./spacing.js";
import { switchTheme as x } from "./switchTheme.js";
import { tabsTheme as k } from "./tabsTheme.js";
import { textareaTheme as B } from "./textareaTheme.js";
import { tooltipTheme as I } from "./tooltipTheme.js";
import { typography as w } from "./typography.js";
import { typographyTokens as M } from "./typographyTokens.plugin.generated.js";
const S = {
  initialColorMode: "light",
  useSystemColorMode: !1
}, v = r({}, w, M), A = m({
  config: S,
  styles: {
    global: {
      "html, body": {
        color: "text-primary",
        bg: "surface-default"
      }
    }
  },
  breakpoints: n,
  colors: o,
  semanticTokens: e,
  ...v,
  sizes: u,
  shadows: g,
  space: b,
  components: {
    Accordion: t,
    Alert: i,
    Badge: p,
    Button: s,
    Divider: f,
    IconButton: h,
    Input: a,
    List: T,
    Menu: c,
    PinInput: l,
    Progress: d,
    Skeleton: y,
    Switch: x,
    Tabs: k,
    Textarea: B,
    Tooltip: I
  }
}), eo = {
  ...A,
  colors: o,
  semanticTokens: e
};
export {
  eo as theme
};
