// @argent/x-ui/dist/src/features/theme/tooltipTheme.js
import { defineStyleConfig as r } from "@chakra-ui/react";
import { mode as o } from "@chakra-ui/theme-tools";
import { typographyStyles as t } from "./typographyStyles.generated.js";
const a = r({
  baseStyle: (e) => ({
    borderRadius: "base",
    border: "1px solid",
    borderColor: "border",
    py: "1",
    px: "2",
    ...t.L1Bold,
    color: o("black", "text")(e),
    background: o("white", "black")(e)
  }),
  variants: {
    tableInfo: () => ({
      borderRadius: "4px",
      border: "none",
      color: "white",
      background: "black",
      lineHeight: "16px",
      fontSize: "14px",
      fontWeight: "400",
      p: "3"
    }),
    tertiary: () => ({
      borderRadius: "lg",
      borderColor: "neutrals.700",
      color: "white",
      background: "neutrals.700",
      lineHeight: "21px",
      fontSize: "base",
      fontWeight: "400",
      px: "4",
      py: "4"
    }),
    web: () => ({
      p: "3",
      fontWeight: "400",
      fontSize: "sm",
      lineHeight: "4",
      color: "text-primary"
    })
  }
});
export {
  a as tooltipTheme
};
