// @argent/x-ui/dist/src/features/theme/typographyStyles.generated.js
const e = {
  B1: {
    fontSize: "b1",
    fontWeight: "bold",
    lineHeight: "b1"
  },
  B2: {
    fontSize: "b2",
    fontWeight: "bold",
    lineHeight: "b2"
  },
  B3: {
    fontSize: "b3",
    fontWeight: "bold",
    lineHeight: "b3"
  },
  Button1: {
    fontSize: "b1",
    fontWeight: "bold",
    lineHeight: "b1"
  },
  Button2: {
    fontSize: "b2",
    fontWeight: "bold",
    lineHeight: "b2"
  },
  Button3: {
    fontSize: "b3",
    fontWeight: "bold",
    lineHeight: "b3"
  },
  FeatureTitle: {
    fontSize: "feature-title",
    fontWeight: "extrabold",
    letterSpacing: "feature-title",
    lineHeight: "feature-title"
  },
  H1: {
    fontSize: "h1",
    fontWeight: "extrabold",
    lineHeight: "h1"
  },
  H2: {
    fontSize: "h2",
    fontWeight: "extrabold",
    lineHeight: "h2"
  },
  H3: {
    fontSize: "h3",
    fontWeight: "extrabold",
    lineHeight: "h3"
  },
  H4: {
    fontSize: "h4",
    fontWeight: "extrabold",
    lineHeight: "h4"
  },
  H5: {
    fontSize: "h5",
    fontWeight: "bold",
    lineHeight: "h5"
  },
  H3Medium: {
    fontSize: "h3",
    fontWeight: "semibold",
    lineHeight: "h3"
  },
  H4Medium: {
    fontSize: "h4",
    fontWeight: "semibold",
    lineHeight: "h4"
  },
  Heading1: {
    fontSize: "h1",
    fontWeight: "extrabold",
    lineHeight: "h1"
  },
  Heading2: {
    fontSize: "h2",
    fontWeight: "extrabold",
    lineHeight: "h2"
  },
  Heading3: {
    fontSize: "h3",
    fontWeight: "extrabold",
    lineHeight: "h3"
  },
  Heading4: {
    fontSize: "h4",
    fontWeight: "extrabold",
    lineHeight: "h4"
  },
  Heading5: {
    fontSize: "h5",
    fontWeight: "bold",
    lineHeight: "h5"
  },
  Heading3Medium: {
    fontSize: "h3",
    fontWeight: "semibold",
    lineHeight: "h3"
  },
  Heading4Medium: {
    fontSize: "h4",
    fontWeight: "semibold",
    lineHeight: "h4"
  },
  L1: {
    fontSize: "l1",
    fontWeight: "normal",
    lineHeight: "l1"
  },
  L2: {
    fontSize: "l2",
    fontWeight: "normal",
    lineHeight: "l2"
  },
  L3: {
    fontSize: "l3",
    fontWeight: "normal",
    lineHeight: "l3"
  },
  L1Bold: {
    fontSize: "l1",
    fontWeight: "bold",
    lineHeight: "l1"
  },
  L2Bold: {
    fontSize: "l2",
    fontWeight: "bold",
    lineHeight: "l2"
  },
  L3Bold: {
    fontSize: "l3",
    fontWeight: "bold",
    lineHeight: "l3"
  },
  Label1: {
    fontSize: "l1",
    fontWeight: "normal",
    lineHeight: "l1"
  },
  Label2: {
    fontSize: "l2",
    fontWeight: "normal",
    lineHeight: "l2"
  },
  Label3: {
    fontSize: "l3",
    fontWeight: "normal",
    lineHeight: "l3"
  },
  Label1Bold: {
    fontSize: "l1",
    fontWeight: "bold",
    lineHeight: "l1"
  },
  Label2Bold: {
    fontSize: "l2",
    fontWeight: "bold",
    lineHeight: "l2"
  },
  Label3Bold: {
    fontSize: "l3",
    fontWeight: "bold",
    lineHeight: "l3"
  },
  MassiveTitle: {
    fontSize: "massive-title",
    fontWeight: "extrabold",
    lineHeight: "massive-title"
  },
  P1: {
    fontSize: "p1",
    fontWeight: "normal",
    lineHeight: "p1"
  },
  P2: {
    fontSize: "p2",
    fontWeight: "normal",
    lineHeight: "p2"
  },
  P3: {
    fontSize: "p3",
    fontWeight: "normal",
    lineHeight: "p3"
  },
  P4: {
    fontSize: "p4",
    fontWeight: "normal",
    lineHeight: "p4"
  },
  P2Bold: {
    fontSize: "p2",
    fontWeight: "bold",
    lineHeight: "p2"
  },
  P3Bold: {
    fontSize: "p3",
    fontWeight: "bold",
    lineHeight: "p3"
  },
  P4Bold: {
    fontSize: "p4",
    fontWeight: "bold",
    lineHeight: "p4"
  },
  Paragraph1: {
    fontSize: "p1",
    fontWeight: "normal",
    lineHeight: "p1"
  },
  Paragraph2: {
    fontSize: "p2",
    fontWeight: "normal",
    lineHeight: "p2"
  },
  Paragraph3: {
    fontSize: "p3",
    fontWeight: "normal",
    lineHeight: "p3"
  },
  Paragraph4: {
    fontSize: "p4",
    fontWeight: "normal",
    lineHeight: "p4"
  },
  Paragraph2Bold: {
    fontSize: "p2",
    fontWeight: "bold",
    lineHeight: "p2"
  },
  Paragraph3Bold: {
    fontSize: "p3",
    fontWeight: "bold",
    lineHeight: "p3"
  },
  Paragraph4Bold: {
    fontSize: "p4",
    fontWeight: "bold",
    lineHeight: "p4"
  }
};
export {
  e as typographyStyles
};
