// @argent/x-ui/dist/src/components/session-keys/BackendSessionTokenInfo.js
import { jsx as d } from "react/jsx-runtime";
import { ensureDecimals as e } from "../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/ensureDecimals.js";
import { prettifyTokenAmount as n } from "../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/prettifyTokenAmount.js";
import { useToken as f } from "../../features/providers/TokenProvider.js";
import { Text as p } from "@chakra-ui/react";
import { useTranslation as k } from "react-i18next";
const b = ({ txFee: t, networkId: s, ...m }) => {
  const { t: r } = k(), { tokenAddress: i, currentAmount: a, maximumAmount: u } = t, o = f({ address: i, networkId: s });
  if (!o)
    return null;
  const l = n({
    amount: a,
    decimals: e(o.decimals)
  }), c = n({
    amount: u,
    decimals: e(o.decimals),
    symbol: o.symbol
  });
  return /* @__PURE__ */ d(p, { textOverflow: "ellipsis", overflow: "hidden", whiteSpace: "nowrap", ...m, children: r("backendSessionTokenInfo.outOf", {
    displayCurrentAmount: l,
    displayMaximumAmount: c
  }) });
};
export {
  b as BackendSessionTokenInfo
};
