// @argent/x-ui/dist/src/features/swap/SwapDetails.js
import { jsx as r, jsxs as o, Fragment as p } from "react/jsx-runtime";
import s from "../icons/AlertSecondaryIcon.js";
import y from "../icons/InfoCircleSecondaryIcon.js";
import { Box as a, Flex as c, Tooltip as g, IconButton as I } from "@chakra-ui/react";
import "../../components/Typography.js";
import { P2 as x, P3 as l } from "../../components/typography.generated.js";
function b({ children: t, tooltipLabel: e, ...i }) {
  return /* @__PURE__ */ o(c, { gap: "1", alignItems: "center", ...i, children: [
    /* @__PURE__ */ r(l, { fontSize: "sm", color: "text-primary", fontWeight: "600", children: t }),
    e && /* @__PURE__ */ r(g, { variant: "web", placement: "top", shouldWrapChildren: !0, label: e, children: /* @__PURE__ */ r(y, { color: "icon-secondary", cursor: "help" }) })
  ] });
}
function B({ ...t }) {
  return /* @__PURE__ */ r(I, { width: "6", minWidth: "6", height: "6", minHeight: "6", colorScheme: "secondary", ...t });
}
function W({ children: t, buttonProps: e, ...i }) {
  return /* @__PURE__ */ o(c, { gap: "2", alignItems: "center", ...i, children: [
    /* @__PURE__ */ r(l, { fontSize: "sm", fontWeight: "600", color: "text-primary", children: t }),
    e && /* @__PURE__ */ r(B, { ...e })
  ] });
}
function S({ children: t, alert: e, ...i }) {
  return /* @__PURE__ */ o(p, { children: [
    /* @__PURE__ */ r(c, { width: "100%", justifyContent: "space-between", py: "3", ...i, children: t }),
    e && /* @__PURE__ */ o(c, { alignItems: "center", gap: "2", color: "text-danger", children: [
      /* @__PURE__ */ r(s, { color: "inherit", minWidth: "fit-content" }),
      /* @__PURE__ */ r(l, { color: "inherit", children: e })
    ] })
  ] });
}
function w({ children: t, ...e }) {
  return /* @__PURE__ */ r(a, { bg: "surface-elevated", borderBottomRadius: "xl", px: "4", pb: "4", ...e, children: t });
}
function C({ children: t, ...e }) {
  return /* @__PURE__ */ r(a, { bg: "surface-elevated", borderTopRadius: "xl", p: "4", borderBottomWidth: "1px", borderColor: "stroke-default-web", ...e, children: t });
}
function n({ children: t, ...e }) {
  return /* @__PURE__ */ r(a, { ...e, children: t });
}
n.Header = C;
n.Body = w;
n.Item = S;
n.ItemProperty = b;
n.ItemValue = W;
n.Default = function({ header: e, items: i }) {
  return /* @__PURE__ */ o(p, { children: [
    /* @__PURE__ */ o(n.Header, { children: [
      /* @__PURE__ */ o(x, { color: "text-primary", fontWeight: "500", children: [
        e.primary.property,
        " ",
        e.primary.value
      ] }),
      /* @__PURE__ */ o(c, { color: "text-primary", alignItems: "center", gap: "2", children: [
        /* @__PURE__ */ r(l, { as: "span", fontSize: "sm", color: "inherit", fontWeight: "500", children: e.secondary.property }),
        /* @__PURE__ */ r(c, { justifyContent: "center", alignItems: "center", children: e.secondary.icon })
      ] })
    ] }),
    /* @__PURE__ */ r(n.Body, { children: i.map(({ property: m, value: h, tooltipLabel: u, alert: f, buttonProps: d }) => /* @__PURE__ */ o(n.Item, { alert: f, py: d ? "2" : "3", children: [
      /* @__PURE__ */ r(n.ItemProperty, { tooltipLabel: u, children: m }),
      /* @__PURE__ */ r(n.ItemValue, { buttonProps: d, children: h })
    ] }, m)) })
  ] });
};
export {
  n as SwapDetails
};
