// @argent/x-ui/dist/logos-deprecated.js
import { default as r } from "./src/features/logos-deprecated/ArgentXLogo.js";
import { default as a } from "./src/features/logos-deprecated/ArgentXLogoFull.js";
import { default as s } from "./src/features/logos-deprecated/AspectLogo.js";
import { default as l } from "./src/features/logos-deprecated/AvnuLogo.js";
import { default as m } from "./src/features/logos-deprecated/BanxaLogo.js";
import { default as u } from "./src/features/logos-deprecated/BlastLogo.js";
import { default as L } from "./src/features/logos-deprecated/BriqLogo.js";
import { default as i } from "./src/features/logos-deprecated/CoinbaseLogo.js";
import { default as y } from "./src/features/logos-deprecated/DiscordLogo.js";
import { default as F } from "./src/features/logos-deprecated/ElementLogo.js";
import { default as b } from "./src/features/logos-deprecated/EmptyNftLogo.js";
import { default as h } from "./src/features/logos-deprecated/EthereumLogo.js";
import { default as B } from "./src/features/logos-deprecated/FlexLogo.js";
import { default as I } from "./src/features/logos-deprecated/FrenslandsLogo.js";
import { default as q } from "./src/features/logos-deprecated/GithubLogo.js";
import { default as T } from "./src/features/logos-deprecated/ImageLogo.js";
import { default as C } from "./src/features/logos-deprecated/InfluenceLogo.js";
import { default as G } from "./src/features/logos-deprecated/JediswapLogo.js";
import { default as M } from "./src/features/logos-deprecated/LayerswapLogo.js";
import { default as O } from "./src/features/logos-deprecated/LedgerLogo.js";
import { default as U } from "./src/features/logos-deprecated/MintsquareLogo.js";
import { default as Z } from "./src/features/logos-deprecated/OrbiterLogo.js";
import { default as z } from "./src/features/logos-deprecated/PyramidLogo.js";
import { default as K } from "./src/features/logos-deprecated/RampLogo.js";
import { default as W } from "./src/features/logos-deprecated/RhinoFiLogo.js";
import { default as _ } from "./src/features/logos-deprecated/SpokLogo.js";
import { default as oo } from "./src/features/logos-deprecated/StarkgateLogo.js";
import { default as ro } from "./src/features/logos-deprecated/StarknetIdLogo.js";
import { default as ao } from "./src/features/logos-deprecated/StarknetLogo.js";
import { default as so } from "./src/features/logos-deprecated/TopperLogo.js";
import { default as po } from "./src/features/logos-deprecated/TwitterLogo.js";
import { default as go } from "./src/features/logos-deprecated/UnframedLogo.js";
import { default as xo } from "./src/features/logos-deprecated/VoyagerLogo.js";
import { default as no } from "./src/features/logos-deprecated/XLogo.js";
import { default as ko } from "./src/features/logos-deprecated/ZklendLogo.js";
export {
  r as ArgentXLogo,
  a as ArgentXLogoFull,
  s as AspectLogo,
  l as AvnuLogo,
  m as BanxaLogo,
  u as BlastLogo,
  L as BriqLogo,
  i as CoinbaseLogo,
  y as DiscordLogo,
  F as ElementLogo,
  b as EmptyNftLogo,
  h as EthereumLogo,
  B as FlexLogo,
  I as FrenslandsLogo,
  q as GithubLogo,
  T as ImageLogo,
  C as InfluenceLogo,
  G as JediswapLogo,
  M as LayerswapLogo,
  O as LedgerLogo,
  U as MintsquareLogo,
  Z as OrbiterLogo,
  z as PyramidLogo,
  K as RampLogo,
  W as RhinoFiLogo,
  _ as SpokLogo,
  oo as StarkgateLogo,
  ro as StarknetIdLogo,
  ao as StarknetLogo,
  so as TopperLogo,
  po as TwitterLogo,
  go as UnframedLogo,
  xo as VoyagerLogo,
  no as XLogo,
  ko as ZklendLogo
};
