// @argent/x-ui/dist/providers.js
import { AddressNameProvider as o, useAddressName as p } from "./src/features/providers/AddressNameProvider.js";
import { ArgentUIProvider as i } from "./src/features/providers/ArgentUIProvider.js";
import { BlockExplorerProvider as d, useBlockExplorer as l, useBlockExplorerOnOpenAddress as m, useBlockExplorerOnOpenTransaction as n, useBlockExplorerTitle as x } from "./src/features/providers/BlockExplorerProvider.js";
import { CurrencyProvider as u, useCurrency as f } from "./src/features/providers/CurrencyProvider.js";
import { DappProvider as v, useDappId as P } from "./src/features/providers/DappProvider.js";
import { ImageOptimizationProvider as O, useGetOptimizedImageUrl as y } from "./src/features/providers/ImageOptimizationProvider.js";
import { LabelsProvider as b, prettifyLabelOrKey as B, useLabel as E } from "./src/features/providers/LabelsProvider.js";
import { TokenProvider as T, useToken as g, useTokenInfo as L } from "./src/features/providers/TokenProvider.js";
import { WebAnalyticsServiceFactory as C } from "./src/features/providers/WebAmplitude/service.js";
import { WebAmplitudeProvider as N } from "./src/features/providers/WebAmplitude/WebAmplitudeProvider.js";
export {
  o as AddressNameProvider,
  i as ArgentUIProvider,
  d as BlockExplorerProvider,
  u as CurrencyProvider,
  v as DappProvider,
  O as ImageOptimizationProvider,
  b as LabelsProvider,
  T as TokenProvider,
  N as WebAmplitudeProvider,
  C as WebAnalyticsServiceFactory,
  B as prettifyLabelOrKey,
  p as useAddressName,
  l as useBlockExplorer,
  m as useBlockExplorerOnOpenAddress,
  n as useBlockExplorerOnOpenTransaction,
  x as useBlockExplorerTitle,
  f as useCurrency,
  P as useDappId,
  y as useGetOptimizedImageUrl,
  E as useLabel,
  g as useToken,
  L as useTokenInfo
};
