// @cartridge/controller/dist/session.js
import { WalletAccount as y, stark as p, ec as m, encode as u } from "starknet";
import { signerToGuid as h, subscribeCreateSession as g } from "@cartridge/controller-wasm";
export * from "@cartridge/controller-wasm";
import { n as f, B as _, K as w, A as v, c as I } from "./provider-CN6AecRF.js";
import { a as C, E as G, F as J, I as E, N as T, R as H, S as L } from "./provider-CN6AecRF.js";
import { CartridgeSessionAccount as K } from "@cartridge/controller-wasm/session";
class O extends y {
  controller;
  constructor(t, {
    rpcUrl: r,
    privateKey: e,
    address: s,
    ownerGuid: i,
    chainId: a,
    expiresAt: n,
    policies: l,
    guardianKeyGuid: o,
    metadataHash: c,
    sessionKeyGuid: d
  }) {
    super({
      provider: { nodeUrl: r },
      walletProvider: t,
      address: s
    }), this.controller = K.newAsRegistered(
      r,
      e,
      s,
      i,
      a,
      {
        expiresAt: n,
        policies: l,
        guardianKeyGuid: o,
        metadataHash: c,
        sessionKeyGuid: d
      }
    );
  }
  /**
   * Invoke execute function in account contract
   *
   * @param calls the invocation object or an array of them, containing:
   * - contractAddress - the address of the contract
   * - entrypoint - the entrypoint of the contract
   * - calldata - (defaults to []) the calldata
   * - signature - (defaults to []) the signature
   * @param abis (optional) the abi of the contract for better displaying
   *
   * @returns response from addTransaction
   */
  async execute(t) {
    return this.controller.execute(f(t));
  }
}
class x extends _ {
  id = "controller_session";
  name = "Controller Session";
  _chainId;
  _rpcUrl;
  _username;
  _redirectUrl;
  _disconnectRedirectUrl;
  _policies;
  _keychainUrl;
  _apiUrl;
  _publicKey;
  _sessionKeyGuid;
  reopenBrowser = !0;
  constructor({
    rpc: t,
    chainId: r,
    policies: e,
    redirectUrl: s,
    disconnectRedirectUrl: i,
    keychainUrl: a,
    apiUrl: n
  }) {
    if (super(), this._policies = {
      verified: !1,
      contracts: e.contracts ? Object.fromEntries(
        Object.entries(e.contracts).map(([o, c]) => [
          o,
          {
            ...c,
            methods: c.methods.map((d) => ({
              ...d,
              authorized: !0
            }))
          }
        ])
      ) : void 0,
      messages: e.messages?.map((o) => ({
        ...o,
        authorized: !0
      }))
    }, this._rpcUrl = t, this._chainId = r, this._redirectUrl = s, this._disconnectRedirectUrl = i, this._keychainUrl = a || w, this._apiUrl = n ?? v, this.tryRetrieveFromQueryOrStorage()) {
      const o = localStorage.getItem("sessionSigner");
      if (!o) throw new Error("failed to get sessionSigner");
      const c = JSON.parse(o);
      this._publicKey = c.pubKey, this._sessionKeyGuid = h({
        starknet: { privateKey: u.addHexPrefix(c.privKey) }
      });
    } else {
      const o = p.randomAddress();
      this._publicKey = m.starkCurve.getStarkKey(o), localStorage.setItem(
        "sessionSigner",
        JSON.stringify({
          privKey: o,
          pubKey: this._publicKey
        })
      ), this._sessionKeyGuid = h({
        starknet: { privateKey: u.addHexPrefix(o) }
      });
    }
    typeof window < "u" && (window.starknet_controller_session = this);
  }
  validatePoliciesSubset(t, r) {
    if (t.contracts) {
      if (!r.contracts) return !1;
      for (const [e, s] of Object.entries(t.contracts)) {
        const i = r.contracts[e];
        if (!i) return !1;
        for (const a of s.methods) {
          const n = i.methods.find(
            (l) => l.entrypoint === a.entrypoint
          );
          if (!n || !n.authorized) return !1;
        }
      }
    }
    if (t.messages) {
      if (!r.messages) return !1;
      for (const e of t.messages) {
        const s = r.messages.find(
          (i) => JSON.stringify(i.domain) === JSON.stringify(e.domain) && JSON.stringify(i.types) === JSON.stringify(e.types)
        );
        if (!s || !s.authorized) return !1;
      }
    }
    return !0;
  }
  async username() {
    return await this.tryRetrieveFromQueryOrStorage(), this._username;
  }
  async probe() {
    return this.account ? this.account : (this.account = this.tryRetrieveFromQueryOrStorage(), this.account);
  }
  async connect() {
    if (this.account)
      return this.account;
    if (this.account = this.tryRetrieveFromQueryOrStorage(), this.account)
      return this.account;
    localStorage.setItem("sessionPolicies", JSON.stringify(this._policies)), localStorage.setItem("lastUsedConnector", this.id);
    try {
      if (this.reopenBrowser) {
        const s = p.randomAddress();
        this._publicKey = m.starkCurve.getStarkKey(s), localStorage.setItem(
          "sessionSigner",
          JSON.stringify({
            privKey: s,
            pubKey: this._publicKey
          })
        ), this._sessionKeyGuid = h({
          starknet: { privateKey: u.addHexPrefix(s) }
        });
        const i = `${this._keychainUrl}/session?public_key=${this._publicKey}&redirect_uri=${this._redirectUrl}&redirect_query_name=startapp&policies=${JSON.stringify(
          this._policies
        )}&rpc_url=${this._rpcUrl}`;
        window.open(i, "_blank");
      }
      const t = await g(
        this._sessionKeyGuid,
        this._apiUrl
      ), r = t.authorization[1], e = {
        username: t.controller.accountID,
        address: t.controller.address,
        ownerGuid: r,
        expiresAt: t.expiresAt,
        guardianKeyGuid: "0x0",
        metadataHash: "0x0",
        sessionKeyGuid: this._sessionKeyGuid
      };
      return localStorage.setItem("session", JSON.stringify(e)), this.tryRetrieveFromQueryOrStorage(), this.account;
    } catch (t) {
      throw console.log(t), t;
    }
  }
  switchStarknetChain(t) {
    throw new Error("switchStarknetChain not implemented");
  }
  addStarknetChain(t) {
    throw new Error("addStarknetChain not implemented");
  }
  disconnect() {
    localStorage.removeItem("sessionSigner"), localStorage.removeItem("session"), localStorage.removeItem("sessionPolicies"), this.account = void 0, this._username = void 0;
    const t = new URL(`${this._keychainUrl}`);
    t.pathname = "disconnect", this._disconnectRedirectUrl && t.searchParams.append(
      "redirect_url",
      this._disconnectRedirectUrl
    );
    const r = window.open(t);
    if (r === null) return Promise.resolve();
    const { resolve: e, promise: s } = Promise.withResolvers();
    function i() {
      r?.closed && (e(), clearInterval(a));
    }
    const a = setInterval(i, 500);
    return s;
  }
  tryRetrieveFromQueryOrStorage() {
    if (this.account)
      return this.account;
    const t = localStorage.getItem("sessionSigner"), r = t ? JSON.parse(t) : null;
    let e = null;
    const s = localStorage.getItem("session");
    if (s && (e = JSON.parse(s)), window.location.search.includes("startapp")) {
      const n = new URLSearchParams(window.location.search), l = n.get("startapp");
      if (l) {
        const o = JSON.parse(
          atob(l)
        );
        Number(o.expiresAt) !== Number(e?.expiresAt) && (e = o, localStorage.setItem("session", JSON.stringify(e))), n.delete("startapp");
        const c = window.location.pathname + (n.toString() ? `?${n.toString()}` : "") + window.location.hash;
        window.history.replaceState({}, document.title, c);
      }
    }
    if (!e || !r)
      return;
    const i = parseInt(e.expiresAt) * 1e3;
    if (Date.now() >= i) {
      this.clearStoredSession();
      return;
    }
    const a = localStorage.getItem("sessionPolicies");
    if (a) {
      const n = JSON.parse(
        a
      );
      if (!this.validatePoliciesSubset(
        this._policies,
        n
      )) {
        this.clearStoredSession();
        return;
      }
    }
    return this._username = e.username, this.account = new O(this, {
      rpcUrl: this._rpcUrl,
      privateKey: r.privKey,
      address: e.address,
      ownerGuid: e.ownerGuid,
      chainId: this._chainId,
      expiresAt: parseInt(e.expiresAt),
      policies: I(this._policies),
      guardianKeyGuid: e.guardianKeyGuid,
      metadataHash: e.metadataHash,
      sessionKeyGuid: e.sessionKeyGuid
    }), this.account;
  }
  clearStoredSession() {
    localStorage.removeItem("sessionSigner"), localStorage.removeItem("session"), localStorage.removeItem("sessionPolicies");
  }
}
export {
  C as ALL_AUTH_OPTIONS,
  G as EMBEDDED_WALLETS,
  J as FeeSource,
  E as IMPLEMENTED_AUTH_OPTIONS,
  T as NotReadyToConnect,
  H as ResponseCodes,
  L as StarterPackItemType,
  x as default
};
