// @argent/x-ui/dist/src/components/CompoundDialogBottomSheet/components/RowList.js
import { jsxs as i, jsx as e, Fragment as w } from "react/jsx-runtime";
import { Flex as m, Tabs as S, Spinner as v, TabPanels as F, TabPanel as I } from "@chakra-ui/react";
import { isObject as L } from "lodash-es";
import { useRef as R, useState as D } from "react";
import { useTranslation as P } from "react-i18next";
import { Virtuoso as k } from "react-virtuoso";
import "../../Typography.js";
import { WebSearchInput as C } from "../../WebSearchInput/index.js";
import { H5 as b, P4 as M } from "../../typography.generated.js";
function O(t) {
  return t && L(t[0]) && "title" in t[0] && "rows" in t[0];
}
function G({ rows: t, RowComponent: y, NoResultsComponent: s, loading: u, allowSearch: x = !0, searchRows: f }) {
  const { t: h } = P(), d = O(t), l = {}, o = R([
    ...d ? t.flatMap(({ title: n, rows: a }, r) => {
      let p = 0;
      return r !== 0 && (p = Object.keys(l).reduce((T, j) => Number(j) + Number(T), t[r - 1].rows.length)), l[p] = n, a;
    }) : t
  ]), [c, g] = D(o.current);
  return /* @__PURE__ */ i(m, { flexDirection: "column", gap: "6", children: [
    x && f && /* @__PURE__ */ e(C, { onChange: (n) => {
      g(f(n.target.value, o.current));
    }, onReset: () => g(o.current) }),
    /* @__PURE__ */ i(
      S,
      {
        variant: "unstyled",
        position: "relative",
        isFitted: !0,
        h: { base: "full", sm: "92" },
        overflowY: "auto",
        children: [
          !u && c.length === 0 && /* @__PURE__ */ i(m, { h: "full", w: "full", flexDirection: "column", justifyContent: "center", alignItems: "center", gap: "3", py: "16", children: [
            s && /* @__PURE__ */ e(s, {}),
            !s && /* @__PURE__ */ e(b, { children: h("rowList.nothingToSelect") })
          ] }),
          u && /* @__PURE__ */ i(m, { h: "full", w: "full", flexDirection: "column", justifyContent: "center", alignItems: "center", gap: "3", py: "16", children: [
            /* @__PURE__ */ e(v, {}),
            /* @__PURE__ */ e(b, { children: h("rowList.loading") })
          ] }),
          !u && c.length > 0 && /* @__PURE__ */ e(F, { children: /* @__PURE__ */ e(I, { children: /* @__PURE__ */ e(k, { style: { height: 368 }, data: c, itemContent: (n, a) => {
            var r;
            return /* @__PURE__ */ i(w, { children: [
              d && c.length === ((r = o.current) == null ? void 0 : r.length) && l[n] && /* @__PURE__ */ e(m, { mt: n === 0 ? "0" : "4", mb: "2", children: /* @__PURE__ */ e(M, { fontWeight: "600", color: "text-secondary", mb: "2", children: l[n] }) }),
              /* @__PURE__ */ e(y, { ...a })
            ] });
          } }) }) })
        ]
      }
    )
  ] });
}
export {
  G as RowList
};
