// @argent/x-ui/dist/src/components/transactions/TransactionReview/FeeEstimationBox.js
import { jsx as o, jsxs as i } from "react/jsx-runtime";
import { Flex as r } from "@chakra-ui/react";
import { useTranslation as d } from "react-i18next";
import "../../Typography.js";
import { Label2Bold as s } from "../../typography.generated.js";
const m = (e) => /* @__PURE__ */ o(r, { borderRadius: "xl", backgroundColor: "white", border: "1px", borderColor: "transparent", pl: 3, pr: 2, py: 2, _dark: {
  backgroundColor: "neutrals.900",
  borderColor: "neutrals.500",
  boxShadow: "menu"
}, ...e }), b = ({ children: e }) => {
  const { t } = d();
  return /* @__PURE__ */ i(r, { borderRadius: "xl", border: "1px", flexDir: "column", borderColor: "text-subtle", children: [
    /* @__PURE__ */ o(r, { px: 3, py: 2, children: e }),
    /* @__PURE__ */ o(r, { backgroundColor: "stroke-default-web", justifyContent: "center", alignItems: "center", borderBottomRadius: "xl", p: "1", children: t("feeEstimationBox.includesOneTimeActivationFee") })
  ] });
}, p = ({ children: e, userClickedAddFunds: t }) => {
  const { t: n } = d();
  return /* @__PURE__ */ i(r, { borderRadius: "xl", border: "1px", flexDir: "column", borderColor: "surface-danger-vibrant", children: [
    /* @__PURE__ */ o(r, { px: 3, py: 2, children: e }),
    /* @__PURE__ */ o(r, { backgroundColor: "surface-danger-default", justifyContent: "center", alignItems: "center", borderBottomRadius: "xl", p: "1", children: /* @__PURE__ */ o(s, { color: "text-danger", children: n(t ? "feeEstimationBox.waitingForFunds" : "feeEstimationBox.insufficientFundsToPayFee") }) })
  ] });
};
export {
  m as FeeEstimationBox,
  b as FeeEstimationBoxWithDeploy,
  p as FeeEstimationBoxWithInsufficientFunds
};
