// @argent/x-ui/dist/src/components/charts/TokenDetailsChart.js
import { jsxs as C, jsx as d } from "react/jsx-runtime";
import j from "../../features/icons/EmptyFaceSecondaryIcon.js";
import { useColorModeValue as L, Box as E, Flex as M, Skeleton as g } from "@chakra-ui/react";
import { createChart as A } from "lightweight-charts";
import { isEmpty as B, noop as V } from "lodash-es";
import { useRef as p, useCallback as u, useEffect as R } from "react";
import { useTranslation as X } from "react-i18next";
import "../Typography.js";
import { buildTokenPrices as Y } from "./utils.js";
import { P3 as $ } from "../typography.generated.js";
const a = ({ tokenGraphData: i, isLoading: f, chartOptions: m, getSeriesConfig: b, markerColor: w, asset: q, onCrosshairMove: c = V, children: F, ...P }) => {
  const { t: I } = X(), r = p(null), o = p(null), t = p(null), S = L("neutrals.400", "neutrals.300"), z = !f && B(i == null ? void 0 : i.prices), H = `${m.height}px`, T = u(() => {
    var e;
    (e = o.current) == null || e.applyOptions({ color: "#FFFFFF4D" });
  }, []), y = u(() => {
    var n, s;
    if (!o.current)
      return;
    const { prices: e } = Y({
      tokenGraphPrices: i == null ? void 0 : i.prices.map((l) => ({
        time: l.date,
        value: l.ccyValue
      }))
    });
    (n = o.current) == null || n.setData(e ?? []), (s = r.current) == null || s.timeScale().fitContent();
  }, [i == null ? void 0 : i.prices]), h = u((e) => {
    if (!t.current)
      return;
    const n = t.current.getBoundingClientRect();
    e.clientX >= n.left && e.clientX <= n.right && e.clientY >= n.top && e.clientY <= n.bottom || c(void 0);
  }, [c]);
  R(() => (window == null || window.addEventListener("mousemove", h), () => {
    window == null || window.removeEventListener("mousemove", h);
  }), [h]);
  const x = u((e) => {
    if (!(!t.current || !o.current))
      if (e.point === void 0 || e.point.x < 0 || e.point.x > t.current.clientWidth || e.point.y < 0 || e.point.y > t.current.clientHeight)
        c(void 0);
      else {
        const n = e.seriesData.get(o.current), s = n == null ? void 0 : n.value, l = e.time;
        s && l && c({
          x: e.point.x,
          y: e.point.y,
          time: l,
          price: s,
          width: t.current.clientWidth,
          height: t.current.clientHeight
        });
      }
  }, [c]), W = u((e) => {
    t.current = e, t.current && (r.current && (r.current.unsubscribeCrosshairMove(x), r.current.remove()), r.current = A(t.current, m), o.current = r.current.addLineSeries(b(w)), r.current.subscribeCrosshairMove(x), y());
  }, [
    m,
    b,
    y,
    w,
    x
  ]), v = u(() => {
    var e;
    r.current && (r.current.applyOptions({
      width: (e = t.current) == null ? void 0 : e.clientWidth
    }), r.current.timeScale().fitContent());
  }, []);
  return R(() => (window == null || window.addEventListener("resize", v), () => {
    window == null || window.removeEventListener("resize", v);
  }), [v]), /* @__PURE__ */ C(E, { minHeight: H, position: "relative", onTouchEnd: (e) => {
    e.preventDefault(), c(void 0);
  }, sx: {
    touchAction: "none"
  }, onMouseEnter: T, ...P, children: [
    !f && /* @__PURE__ */ d(E, { ref: W, mx: "-4" }),
    z && /* @__PURE__ */ C(M, { flexDir: "column", justifyContent: "center", alignItems: "center", h: "full", children: [
      /* @__PURE__ */ d(g, { h: "90%", mx: 5, borderRadius: "xl" }),
      " ",
      /* @__PURE__ */ d(j, { color: S, w: "5", h: "5", gap: "2" }),
      /* @__PURE__ */ d($, { fontSize: "sm", fontWeight: "semibold", children: I("tokenDetailsChart.noPriceAvailable") })
    ] }),
    f && /* @__PURE__ */ d(g, { h: "90%", mx: 5, borderRadius: "xl" }),
    F
  ] });
};
export {
  a as TokenDetailsChart
};
