// @tronweb3/walletconnect-tron/lib/cjs/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NETWORK_MAP = exports.shastaTestnet = exports.nileTestnet = exports.mainnet = exports.WalletConnectChainID = void 0;
const networks_1 = require("@reown/appkit/networks");
var WalletConnectChainID;
(function (WalletConnectChainID) {
    WalletConnectChainID["Mainnet"] = "tron:0x2b6653dc";
    WalletConnectChainID["Shasta"] = "tron:0x94a9059e";
    WalletConnectChainID["Nile"] = "tron:0xcd8690dc";
})(WalletConnectChainID = exports.WalletConnectChainID || (exports.WalletConnectChainID = {}));
exports.mainnet = (0, networks_1.defineChain)({
    id: '0x2b6653dc',
    caipNetworkId: 'tron:0x2b6653dc',
    chainNamespace: 'tron',
    name: 'Tron Mainnet',
    nativeCurrency: {
        decimals: 6,
        name: 'TRX',
        symbol: 'TRX'
    },
    rpcUrls: {
        default: {
            http: ['https://api.trongrid.io/jsonrpc'],
            webSocket: ['']
        }
    },
    blockExplorers: {
        default: { name: 'Tron BlockChain Explorer', url: 'https://tronscan.org/' }
    },
    contracts: {}
});
exports.nileTestnet = (0, networks_1.defineChain)({
    id: '0xcd8690dc',
    caipNetworkId: 'tron:0xcd8690dc',
    chainNamespace: 'tron',
    name: 'Tron Nile Testnet',
    nativeCurrency: {
        decimals: 6,
        name: 'TRX',
        symbol: 'TRX'
    },
    rpcUrls: {
        default: {
            http: [''],
            webSocket: ['']
        }
    },
    blockExplorers: {
        default: { name: 'Tron BlockChain Explorer', url: 'https://nile.tronscan.org/' }
    },
    contracts: {}
});
exports.shastaTestnet = (0, networks_1.defineChain)({
    id: '0x94a9059e',
    caipNetworkId: 'tron:0x94a9059e',
    chainNamespace: 'tron',
    name: 'Tron Shasta Testnet',
    nativeCurrency: {
        decimals: 6,
        name: 'TRX',
        symbol: 'TRX'
    },
    rpcUrls: {
        default: {
            http: [''],
            webSocket: ['']
        }
    },
    blockExplorers: {
        default: { name: 'Tron BlockChain Explorer', url: 'https://shasta.tronscan.org/' }
    },
    contracts: {}
});
exports.NETWORK_MAP = new Map([
    [WalletConnectChainID.Mainnet, exports.mainnet],
    [WalletConnectChainID.Nile, exports.nileTestnet],
    [WalletConnectChainID.Shasta, exports.shastaTestnet]
]);
