// @hot-labs/near-connect/build/popups/IframeWalletPopup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IframeWalletPopup = void 0;
const html_1 = require("../helpers/html");
const Popup_1 = require("./Popup");
class IframeWalletPopup extends Popup_1.Popup {
    delegate;
    constructor(delegate) {
        super(delegate);
        this.delegate = delegate;
    }
    handlers() {
        super.handlers();
        this.addListener("button", "click", () => this.delegate.onApprove());
    }
    create() {
        super.create({ show: false });
        const modalBody = this.root.querySelector(".modal-body");
        modalBody.appendChild(this.delegate.iframe);
        this.delegate.iframe.style.width = "100%";
        this.delegate.iframe.style.height = "720px";
        this.delegate.iframe.style.border = "none";
    }
    get footer() {
        if (!this.delegate.footer)
            return "";
        const { icon, heading, link, linkText } = this.delegate.footer;
        return (0, html_1.html) `
      <div class="footer">
        <img src="${icon}" alt="${heading}" />
        <p>${heading}</p>
        <a class="get-wallet-link" href="${link}" target="_blank">${linkText}</a>
      </div>
    `;
    }
    get dom() {
        return (0, html_1.html) ` <div class="modal-container">
      <div class="modal-content">
        <div class="modal-body" style="padding: 0; overflow: auto;"></div>
        <div class="footer">
          <img src="https://tgapp.herewallet.app/images/hot/hot-icon.png" alt="HOT Connector" />
          <p>HOT Connector</p>
          <p class="get-wallet-link">Don't have a wallet?</p>
        </div>
      </div>
    </div>`;
    }
}
exports.IframeWalletPopup = IframeWalletPopup;
