// @tronweb3/tronwallet-adapter-metamask-tron/lib/cjs/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.chainIdToScope = chainIdToScope;
exports.scopeToChainId = scopeToChainId;
exports.getAddressFromCaipAccountId = getAddressFromCaipAccountId;
exports.isAccountChangedEvent = isAccountChangedEvent;
exports.isSessionChangedEvent = isSessionChangedEvent;
exports.scopeToNetworkType = scopeToNetworkType;
const tronwallet_abstract_adapter_1 = require("@tronweb3/tronwallet-abstract-adapter");
const types_js_1 = require("./types.js");
/**
 * Converts a chain ID to its corresponding Tron scope.
 * @param chainId - The chain ID string.
 * @returns The corresponding TronScope.
 * @throws Error if the chain ID is unsupported.
 */
function chainIdToScope(chainId) {
    switch (chainId) {
        case '0x2b6653dc': // Tron mainnet
            return types_js_1.Scope.MAINNET;
        case '0xcd8690dc': // Tron nile testnet
            return types_js_1.Scope.NILE;
        case '0x94a9059e': // Tron shasta testnet
            return types_js_1.Scope.SHASTA;
        default:
            throw new Error(`Could not determine scope for unsupported chainId: ${chainId}`);
    }
}
/**
 * Converts a Tron scope to its corresponding chain ID.
 * @param scope - The TronScope.
 * @returns The corresponding chain ID string.
 * @throws Error if the scope is unsupported.
 */
function scopeToChainId(scope) {
    switch (scope) {
        case types_js_1.Scope.MAINNET:
            return '0x2b6653dc';
        case types_js_1.Scope.NILE:
            return '0xcd8690dc';
        case types_js_1.Scope.SHASTA:
            return '0x94a9059e';
        default:
            throw new Error(`Could not determine chainId for unsupported scope: ${scope}`);
    }
}
/**
 * Extracts the address from a CAIP account ID.
 * @param caipAccountId - The CAIP account ID string (e.g., 'tron:mainnet:address').
 * @returns The extracted address.
 * @throws Error if the CAIP account ID is invalid.
 */
function getAddressFromCaipAccountId(caipAccountId) {
    const [, , address] = caipAccountId.split(':');
    if (!address) {
        throw new Error(`Invalid CAIP account ID: ${caipAccountId}`);
    }
    return address;
}
/**
 * Checks if the given data represents an accountsChanged event.
 * @param data - The event data.
 * @returns True if it's an accountsChanged event, false otherwise.
 */
function isAccountChangedEvent(event) {
    var _a, _b;
    return (event === null || event === void 0 ? void 0 : event.method) === 'wallet_notify' && ((_b = (_a = event === null || event === void 0 ? void 0 : event.params) === null || _a === void 0 ? void 0 : _a.notification) === null || _b === void 0 ? void 0 : _b.method) === 'metamask_accountsChanged';
}
function isSessionChangedEvent(event) {
    return (event === null || event === void 0 ? void 0 : event.method) === 'wallet_sessionChanged';
}
/**
 * Converts a Tron scope to its corresponding NetworkType.
 * @param scope - The Tron scope string.
 * @returns The corresponding NetworkType.
 * @throws Error if the scope is unsupported.
 */
function scopeToNetworkType(scope) {
    switch (scope) {
        case types_js_1.Scope.MAINNET:
            return tronwallet_abstract_adapter_1.NetworkType.Mainnet;
        case types_js_1.Scope.NILE:
            return tronwallet_abstract_adapter_1.NetworkType.Nile;
        case types_js_1.Scope.SHASTA:
            return tronwallet_abstract_adapter_1.NetworkType.Shasta;
        default:
            throw new Error(`Could not determine network type for unsupported scope: ${scope}`);
    }
}
