// @tronweb3/abstract-adapter-evm/lib/cjs/adapter.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Adapter = exports.WalletReadyState = exports.EventEmitter = void 0;
const eventemitter3_1 = __importDefault(require("eventemitter3"));
exports.EventEmitter = eventemitter3_1.default;
const errors_js_1 = require("./errors.js");
/**
 * Wallet ready state.
 */
var WalletReadyState;
(function (WalletReadyState) {
    /**
     * Adapter will start to check if wallet exists after adapter instance is created.
     */
    WalletReadyState["Loading"] = "Loading";
    /**
     * When checking ends and wallet is not found, readyState will be NotFound.
     */
    WalletReadyState["NotFound"] = "NotFound";
    /**
     * When checking ends and wallet is found, readyState will be Found.
     */
    WalletReadyState["Found"] = "Found";
})(WalletReadyState || (exports.WalletReadyState = WalletReadyState = {}));
class Adapter extends eventemitter3_1.default {
    constructor() {
        super(...arguments);
        this.connecting = false;
    }
    get connected() {
        return !!this.address;
    }
    network() {
        return __awaiter(this, void 0, void 0, function* () {
            const provider = yield this.prepareProvider();
            return provider.request({
                method: 'eth_chainId',
                params: [],
            });
        });
    }
    signMessage(_a) {
        return __awaiter(this, arguments, void 0, function* ({ message, address }) {
            const provider = yield this.prepareProvider();
            if (!this.connected) {
                throw new errors_js_1.WalletDisconnectedError();
            }
            return provider.request({
                method: 'personal_sign',
                params: [message, address || this.address],
            });
        });
    }
    signTypedData(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const provider = yield this.prepareProvider();
            if (!this.connected) {
                throw new errors_js_1.WalletDisconnectedError();
            }
            return provider.request({
                method: 'eth_signTypedData_v4',
                params: [params.address || this.address, params.typedData],
            });
        });
    }
    sendTransaction(transaction) {
        return __awaiter(this, void 0, void 0, function* () {
            const provider = yield this.prepareProvider();
            if (!this.connected) {
                throw new errors_js_1.WalletDisconnectedError();
            }
            return provider.request({
                method: 'eth_sendTransaction',
                params: [transaction],
            });
        });
    }
    addChain(chainInfo) {
        return __awaiter(this, void 0, void 0, function* () {
            const provider = yield this.prepareProvider();
            return provider.request({
                method: 'wallet_addEthereumChain',
                params: [chainInfo],
            });
        });
    }
    switchChain(chainId) {
        return __awaiter(this, void 0, void 0, function* () {
            const provider = yield this.prepareProvider();
            return provider.request({
                method: 'wallet_switchEthereumChain',
                params: [{ chainId }],
            });
        });
    }
    watchAsset(asset) {
        return __awaiter(this, void 0, void 0, function* () {
            const provider = yield this.prepareProvider();
            return provider.request({
                method: 'wallet_watchAsset',
                params: asset,
            });
        });
    }
    prepareProvider() {
        return __awaiter(this, void 0, void 0, function* () {
            return (yield this.getProvider());
        });
    }
}
exports.Adapter = Adapter;
