// @tronweb3/abstract-adapter-evm/lib/cjs/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletConnectionError = exports.WalletDisconnectedError = exports.WalletNotFoundError = exports.WalletError = void 0;
class WalletError extends Error {
    constructor(message, error) {
        super(message);
        this.error = error;
    }
}
exports.WalletError = WalletError;
/**
 * Occurs when wallet is not installed.
 */
class WalletNotFoundError extends WalletError {
    constructor() {
        super(...arguments);
        this.name = 'WalletNotFoundError';
        this.message = 'The wallet is not found.';
    }
}
exports.WalletNotFoundError = WalletNotFoundError;
/**
 * Occurs when try to sign but is not connected to wallet.
 */
class WalletDisconnectedError extends WalletError {
    constructor() {
        super(...arguments);
        this.name = 'WalletDisconnectedError';
        this.message = 'The wallet is disconnected. Please connect the wallet first.';
    }
}
exports.WalletDisconnectedError = WalletDisconnectedError;
/**
 * Occurs when try to connect a wallet.
 */
class WalletConnectionError extends WalletError {
    constructor() {
        super(...arguments);
        this.name = 'WalletConnectionError';
    }
}
exports.WalletConnectionError = WalletConnectionError;
