// starknetkit/dist/connector-4bc69b15.js
var w = Object.defineProperty;
var O = (s, a, o) => a in s ? w(s, a, { enumerable: !0, configurable: !0, writable: !0, value: o }) : s[a] = o;
var h = (s, a, o) => (O(s, typeof a != "symbol" ? a + "" : a, o), o);
import "starknet";
class U extends Error {
  constructor() {
    super(...arguments);
    h(this, "name", "ConnectorNotAvailableError");
    h(this, "message", "Connector not available");
  }
}
class R extends Error {
  constructor() {
    super(...arguments);
    h(this, "name", "ConnectorNotConnectedError");
    h(this, "message", "Connector not connected");
  }
}
class T extends Error {
  constructor() {
    super(...arguments);
    h(this, "name", "ConnectorNotFoundError");
    h(this, "message", "Connector not found");
  }
}
class k extends Error {
  constructor() {
    super(...arguments);
    h(this, "name", "UserRejectedRequestError");
    h(this, "message", "User rejected request");
  }
}
class q extends Error {
  constructor() {
    super(...arguments);
    h(this, "name", "UserNotConnectedError");
    h(this, "message", "User not connected");
  }
}
var F = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function N(s) {
  return s && s.__esModule && Object.prototype.hasOwnProperty.call(s, "default") ? s.default : s;
}
function M(s) {
  if (s.__esModule)
    return s;
  var a = s.default;
  if (typeof a == "function") {
    var o = function v() {
      return this instanceof v ? Reflect.construct(a, arguments, this.constructor) : a.apply(this, arguments);
    };
    o.prototype = a.prototype;
  } else
    o = {};
  return Object.defineProperty(o, "__esModule", { value: !0 }), Object.keys(s).forEach(function(v) {
    var g = Object.getOwnPropertyDescriptor(s, v);
    Object.defineProperty(o, v, g.get ? g : {
      enumerable: !0,
      get: function() {
        return s[v];
      }
    });
  }), o;
}
var C = { exports: {} };
(function(s) {
  var a = Object.prototype.hasOwnProperty, o = "~";
  function v() {
  }
  Object.create && (v.prototype = /* @__PURE__ */ Object.create(null), new v().__proto__ || (o = !1));
  function g(c, t, n) {
    this.fn = c, this.context = t, this.once = n || !1;
  }
  function b(c, t, n, r, p) {
    if (typeof n != "function")
      throw new TypeError("The listener must be a function");
    var l = new g(n, r || c, p), i = o ? o + t : t;
    return c._events[i] ? c._events[i].fn ? c._events[i] = [c._events[i], l] : c._events[i].push(l) : (c._events[i] = l, c._eventsCount++), c;
  }
  function E(c, t) {
    --c._eventsCount === 0 ? c._events = new v() : delete c._events[t];
  }
  function u() {
    this._events = new v(), this._eventsCount = 0;
  }
  u.prototype.eventNames = function() {
    var t = [], n, r;
    if (this._eventsCount === 0)
      return t;
    for (r in n = this._events)
      a.call(n, r) && t.push(o ? r.slice(1) : r);
    return Object.getOwnPropertySymbols ? t.concat(Object.getOwnPropertySymbols(n)) : t;
  }, u.prototype.listeners = function(t) {
    var n = o ? o + t : t, r = this._events[n];
    if (!r)
      return [];
    if (r.fn)
      return [r.fn];
    for (var p = 0, l = r.length, i = new Array(l); p < l; p++)
      i[p] = r[p].fn;
    return i;
  }, u.prototype.listenerCount = function(t) {
    var n = o ? o + t : t, r = this._events[n];
    return r ? r.fn ? 1 : r.length : 0;
  }, u.prototype.emit = function(t, n, r, p, l, i) {
    var d = o ? o + t : t;
    if (!this._events[d])
      return !1;
    var e = this._events[d], y = arguments.length, m, f;
    if (e.fn) {
      switch (e.once && this.removeListener(t, e.fn, void 0, !0), y) {
        case 1:
          return e.fn.call(e.context), !0;
        case 2:
          return e.fn.call(e.context, n), !0;
        case 3:
          return e.fn.call(e.context, n, r), !0;
        case 4:
          return e.fn.call(e.context, n, r, p), !0;
        case 5:
          return e.fn.call(e.context, n, r, p, l), !0;
        case 6:
          return e.fn.call(e.context, n, r, p, l, i), !0;
      }
      for (f = 1, m = new Array(y - 1); f < y; f++)
        m[f - 1] = arguments[f];
      e.fn.apply(e.context, m);
    } else {
      var x = e.length, _;
      for (f = 0; f < x; f++)
        switch (e[f].once && this.removeListener(t, e[f].fn, void 0, !0), y) {
          case 1:
            e[f].fn.call(e[f].context);
            break;
          case 2:
            e[f].fn.call(e[f].context, n);
            break;
          case 3:
            e[f].fn.call(e[f].context, n, r);
            break;
          case 4:
            e[f].fn.call(e[f].context, n, r, p);
            break;
          default:
            if (!m)
              for (_ = 1, m = new Array(y - 1); _ < y; _++)
                m[_ - 1] = arguments[_];
            e[f].fn.apply(e[f].context, m);
        }
    }
    return !0;
  }, u.prototype.on = function(t, n, r) {
    return b(this, t, n, r, !1);
  }, u.prototype.once = function(t, n, r) {
    return b(this, t, n, r, !0);
  }, u.prototype.removeListener = function(t, n, r, p) {
    var l = o ? o + t : t;
    if (!this._events[l])
      return this;
    if (!n)
      return E(this, l), this;
    var i = this._events[l];
    if (i.fn)
      i.fn === n && (!p || i.once) && (!r || i.context === r) && E(this, l);
    else {
      for (var d = 0, e = [], y = i.length; d < y; d++)
        (i[d].fn !== n || p && !i[d].once || r && i[d].context !== r) && e.push(i[d]);
      e.length ? this._events[l] = e.length === 1 ? e[0] : e : E(this, l);
    }
    return this;
  }, u.prototype.removeAllListeners = function(t) {
    var n;
    return t ? (n = o ? o + t : t, this._events[n] && E(this, n)) : (this._events = new v(), this._eventsCount = 0), this;
  }, u.prototype.off = u.prototype.removeListener, u.prototype.addListener = u.prototype.on, u.prefixed = o, u.EventEmitter = u, s.exports = u;
})(C);
var j = C.exports;
const L = /* @__PURE__ */ N(j);
class D extends L {
  /** Handle starknet-react type compatibility */
  eventNames() {
    return ["change", "connect", "disconnect"];
  }
}
export {
  D as C,
  k as U,
  R as a,
  T as b,
  F as c,
  U as d,
  q as e,
  N as f,
  M as g
};
