// starknetkit/dist/keplrMobile.js
var l = Object.defineProperty;
var i = (n, e, r) => e in n ? l(n, e, { enumerable: !0, configurable: !0, writable: !0, value: r }) : n[e] = r;
var t = (n, e, r) => (i(n, typeof e != "symbol" ? e + "" : e, r), r);
import "./core-7c9d487b.js";
import { C as s } from "./connector-4bc69b15.js";
import { Keplr as p } from "./keplr.js";
import "starknet";
import "./index-40f0fb49.js";
import "./lastConnected-c0cae438.js";
const o = `<svg width="32" height="32" viewBox="0 0 128 128" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M0 64C0 41.5979 0 30.3969 4.35974 21.8404C8.19467 14.3139 14.3139 8.19467 21.8404 4.35974C30.3969 0 41.5979 0 64 0C86.4021 0 97.6031 0 106.16 4.35974C113.686 8.19467 119.805 14.3139 123.64 21.8404C128 30.3969 128 41.5979 128 64C128 86.4021 128 97.6031 123.64 106.16C119.805 113.686 113.686 119.805 106.16 123.64C97.6031 128 86.4021 128 64 128C41.5979 128 30.3969 128 21.8404 123.64C14.3139 119.805 8.19467 113.686 4.35974 106.16C0 97.6031 0 86.4021 0 64Z" fill="#14AFEB"/>
  <path d="M66.6672 41.3438C67.3936 48.7672 67.7565 52.4791 69.7022 55.201C70.5359 56.3673 71.5642 57.3817 72.7418 58.1994C75.4899 60.1076 79.2063 60.4202 86.6387 61.0451L89.7812 61.3091V66.6909L86.6387 66.9549C79.2063 67.5798 75.4899 67.8924 72.7418 69.8006C71.5642 70.6183 70.5359 71.6327 69.7022 72.799C67.7565 75.5209 67.3936 79.2328 66.6672 86.6562H61.3328C60.6064 79.2328 60.2435 75.5209 58.2978 72.799C57.4641 71.6327 56.4358 70.6183 55.2582 69.8006C52.5101 67.8924 48.7937 67.5798 41.3613 66.9549L38.2187 66.6909V61.3091L41.3613 61.0451C48.7937 60.4202 52.5101 60.1076 55.2582 58.1994C56.4358 57.3817 57.4641 56.3673 58.2978 55.201C60.2435 52.4791 60.6064 48.7672 61.3328 41.3438H66.6672Z" fill="white"/>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M64 14C105.175 14 114 22.825 114 64C114 105.175 105.175 114 64 114C22.825 114 14 105.175 14 64C14 22.825 22.825 14 64 14ZM96.6782 31.2798C84.5687 19.1705 60.1124 23.9936 42.0533 42.0525C23.9941 60.1117 19.171 84.5687 31.2806 96.6782C43.3902 108.788 67.8464 103.965 85.9055 85.9055C103.965 67.8463 108.788 43.3894 96.6782 31.2798Z" fill="white"/>
</svg>`, a = () => {
  if (typeof window > "u")
    return !1;
  const e = navigator.userAgent.includes("KeplrWalletMobile");
  return e || !1;
};
class w extends s {
  constructor() {
    super();
    t(this, "_wallet", null);
  }
  available() {
    return !0;
  }
  async ready() {
    return !0;
  }
  get id() {
    return "keplrMobile";
  }
  get name() {
    return "Keplr (mobile)";
  }
  get icon() {
    return {
      dark: o,
      light: o
    };
  }
  get wallet() {
    throw new Error("not implemented");
  }
  async connect(r = {}) {
    throw this.ensureWallet(), new Error("Please continue in Keplr mobile app");
  }
  async disconnect() {
    throw new Error("not implemented");
  }
  async account(r) {
    throw new Error("not implemented");
  }
  async chainId() {
    throw new Error("not implemented");
  }
  async request(r) {
    throw new Error("not implemented");
  }
  // needed, methods required by starknet-react. Otherwise an exception is throwd
  async initEventListener(r) {
    throw new Error("not implemented");
  }
  // needed, methods required by starknet-react. Otherwise an exception is throwd
  async removeEventListener(r) {
    throw new Error("not implemented");
  }
  async ensureWallet() {
    window.open(
      `https://deeplink.keplr.app/web-browser?url=${encodeURIComponent(window.origin)}`,
      "_blank"
    );
  }
}
class f {
  static init(e) {
    const { inAppBrowserOptions: r } = e || {};
    return a() ? new p(r) : new w();
  }
}
export {
  w as KeplrMobileBaseConnector,
  f as KeplrMobileConnector,
  a as isInKeplrMobileAppBrowser
};
