// starknetkit/dist/trpc-d914d822.js
import { constants as Re } from "starknet";
import { g as cr, c as Ee } from "./connector-4bc69b15.js";
import { P as qt } from "./lastConnected-c0cae438.js";
const Wa = (t) => t === Re.StarknetChainId.SN_MAIN ? Re.StarknetChainId.SN_MAIN : Re.StarknetChainId.SN_SEPOLIA, an = "https://web.ready.co", Ha = `<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
  <rect x="2" y="2" width="28" height="28" rx="5.6" fill="black"/>
  <g clip-path="url(#clip0_9_981)">
    <path d="M10.5 20.9999H21.5C21.9166 20.9999 22.2736 20.7452 22.424 20.383L18.1419 16.3631L16.3739 18.0018C16.163 18.1973 15.837 18.1973 15.6261 18.0018L13.8581 16.3631L9.57603 20.383C9.72637 20.7452 10.0834 20.9999 10.5 20.9999Z" fill="white"/>
    <path d="M18.8771 15.6817L22.5 19.0828V12.3239L18.8771 15.6817Z" fill="white"/>
    <path d="M13.1229 15.6817L9.5 12.3239V19.0828L13.1229 15.6817Z" fill="white"/>
    <path d="M22 10.9999H10C9.8752 10.9999 9.76108 11.0456 9.67348 11.1212L15.7734 16.7748C15.9012 16.8933 16.0988 16.8933 16.2266 16.7748L22.3265 11.1212C22.2389 11.0456 22.1248 10.9999 22 10.9999Z" fill="white"/>
  </g>
  <rect x="20" y="20" width="12" height="12" rx="2.4" fill="#FF875B"/>
  <path d="M26.9611 22.6222H25.0397C24.9755 22.6222 24.924 22.675 24.9226 22.7406C24.8838 24.5841 23.9398 26.3339 22.3148 27.5732C22.2633 27.6126 22.2515 27.6865 22.2892 27.7397L23.4134 29.3279C23.4517 29.3819 23.5262 29.3941 23.5786 29.3544C24.5947 28.5845 25.4119 27.6558 26.0004 26.6264C26.5889 27.6558 27.4062 28.5845 28.4222 29.3544C28.4746 29.3941 28.5492 29.3819 28.5875 29.3279L29.7117 27.7397C29.7493 27.6865 29.7376 27.6126 29.686 27.5732C28.0611 26.3339 27.117 24.5841 27.0782 22.7406C27.0768 22.675 27.0253 22.6222 26.9611 22.6222Z" fill="white"/>
  <defs>
  <clipPath id="clip0_9_981">
    <rect width="16" height="16" fill="white" transform="translate(8 7.99991)"/>
  </clipPath>
  </defs>
</svg>`, Ja = "https://static.hydrogen.argent47.net/webwallet/iframe_whitelist_testnet.json", Ya = "https://static.argent.net/webwallet/iframe_whitelist_mainnet.json", J = Re.NetworkName, on = J.SN_SEPOLIA;
function Ga(t) {
  try {
    const { origin: e } = new URL(t);
    if (e.includes("localhost") || e.includes("127.0.0.1"))
      return on;
    if (e.includes("hydrogen") || e.includes("sepolia-web.staging"))
      return J.SN_SEPOLIA;
    if (e.includes("staging"))
      return J.SN_MAIN;
    if (e.includes("dev") || e.includes("sepolia-web.ready.co"))
      return J.SN_SEPOLIA;
    if (e.includes("ready.co"))
      return J.SN_MAIN;
  } catch {
    console.warn(
      "Could not determine network from target URL, defaulting to mainnet-alpha"
    );
  }
  return J.SN_MAIN;
}
function cn(t) {
  return t;
}
function un(t) {
  return t.length === 0 ? cn : t.length === 1 ? t[0] : function(r) {
    return t.reduce((n, s) => s(n), r);
  };
}
function dn(t) {
  return typeof t == "object" && t !== null && "subscribe" in t;
}
function Ie(t) {
  const e = {
    subscribe(r) {
      let n = null, s = !1, a = !1, i = !1;
      function o() {
        if (n === null) {
          i = !0;
          return;
        }
        a || (a = !0, typeof n == "function" ? n() : n && n.unsubscribe());
      }
      return n = t({
        next(c) {
          s || r.next?.(c);
        },
        error(c) {
          s || (s = !0, r.error?.(c), o());
        },
        complete() {
          s || (s = !0, r.complete?.(), o());
        }
      }), i && o(), {
        unsubscribe: o
      };
    },
    pipe(...r) {
      return un(r)(e);
    }
  };
  return e;
}
function ur(t) {
  return (e) => {
    let r = 0, n = null;
    const s = [];
    function a() {
      n || (n = e.subscribe({
        next(o) {
          for (const c of s)
            c.next?.(o);
        },
        error(o) {
          for (const c of s)
            c.error?.(o);
        },
        complete() {
          for (const o of s)
            o.complete?.();
        }
      }));
    }
    function i() {
      if (r === 0 && n) {
        const o = n;
        n = null, o.unsubscribe();
      }
    }
    return {
      subscribe(o) {
        return r++, s.push(o), a(), {
          unsubscribe() {
            r--, i();
            const c = s.findIndex((u) => u === o);
            c > -1 && s.splice(c, 1);
          }
        };
      }
    };
  };
}
function ln(t) {
  return (e) => ({
    subscribe(r) {
      let n = 0;
      return e.subscribe({
        next(a) {
          r.next?.(t(a, n++));
        },
        error(a) {
          r.error?.(a);
        },
        complete() {
          r.complete?.();
        }
      });
    }
  });
}
function dr(t) {
  return (e) => ({
    subscribe(r) {
      return e.subscribe({
        next(n) {
          t.next?.(n), r.next?.(n);
        },
        error(n) {
          t.error?.(n), r.error?.(n);
        },
        complete() {
          t.complete?.(), r.complete?.();
        }
      });
    }
  });
}
let fn = class lr extends Error {
  constructor(e) {
    super(e), this.name = "ObservableAbortError", Object.setPrototypeOf(this, lr.prototype);
  }
};
function fr(t) {
  let e;
  return {
    promise: new Promise((n, s) => {
      let a = !1;
      function i() {
        a || (a = !0, s(new fn("This operation was aborted.")), o.unsubscribe());
      }
      const o = t.subscribe({
        next(c) {
          a = !0, n(c), i();
        },
        error(c) {
          a = !0, s(c), i();
        },
        complete() {
          a = !0, i();
        }
      });
      e = i;
    }),
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    abort: e
  };
}
const pn = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  isObservable: dn,
  map: ln,
  observable: Ie,
  observableToPromise: fr,
  share: ur,
  tap: dr
}, Symbol.toStringTag, { value: "Module" }));
function pr(t) {
  return Ie((e) => {
    function r(s = 0, a = t.op) {
      const i = t.links[s];
      if (!i)
        throw new Error("No more links to execute - did you forget to add an ending link?");
      return i({
        op: a,
        next(c) {
          return r(s + 1, c);
        }
      });
    }
    return r().subscribe(e);
  });
}
function zt(t) {
  return Array.isArray(t) ? t : [
    t
  ];
}
function hn(t) {
  return (e) => {
    const r = zt(t.true).map((s) => s(e)), n = zt(t.false).map((s) => s(e));
    return (s) => Ie((a) => {
      const i = t.condition(s.op) ? r : n;
      return pr({
        op: s.op,
        links: i
      }).subscribe(a);
    });
  };
}
function hr(t) {
  const e = /* @__PURE__ */ Object.create(null);
  for (const r in t) {
    const n = t[r];
    e[n] = r;
  }
  return e;
}
const $e = {
  /**
  * Invalid JSON was received by the server.
  * An error occurred on the server while parsing the JSON text.
  */
  PARSE_ERROR: -32700,
  /**
  * The JSON sent is not a valid Request object.
  */
  BAD_REQUEST: -32600,
  // Internal JSON-RPC error
  INTERNAL_SERVER_ERROR: -32603,
  NOT_IMPLEMENTED: -32603,
  // Implementation specific errors
  UNAUTHORIZED: -32001,
  FORBIDDEN: -32003,
  NOT_FOUND: -32004,
  METHOD_NOT_SUPPORTED: -32005,
  TIMEOUT: -32008,
  CONFLICT: -32009,
  PRECONDITION_FAILED: -32012,
  PAYLOAD_TOO_LARGE: -32013,
  UNPROCESSABLE_CONTENT: -32022,
  TOO_MANY_REQUESTS: -32029,
  CLIENT_CLOSED_REQUEST: -32099
};
hr($e);
hr($e);
const mn = {
  PARSE_ERROR: 400,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
  FORBIDDEN: 403,
  METHOD_NOT_SUPPORTED: 405,
  TIMEOUT: 408,
  CONFLICT: 409,
  PRECONDITION_FAILED: 412,
  PAYLOAD_TOO_LARGE: 413,
  UNPROCESSABLE_CONTENT: 422,
  TOO_MANY_REQUESTS: 429,
  CLIENT_CLOSED_REQUEST: 499,
  INTERNAL_SERVER_ERROR: 500,
  NOT_IMPLEMENTED: 501
};
function yn(t) {
  return mn[t] ?? 500;
}
function mr(t) {
  return yn(t.code);
}
const yr = () => {
};
function gr(t, e) {
  return new Proxy(yr, {
    get(n, s) {
      if (!(typeof s != "string" || s === "then"))
        return gr(t, [
          ...e,
          s
        ]);
    },
    apply(n, s, a) {
      const i = e[e.length - 1] === "apply";
      return t({
        args: i ? a.length >= 2 ? a[1] : [] : a,
        path: i ? e.slice(0, -1) : e
      });
    }
  });
}
const Et = (t) => gr(t, []), Rt = (t) => new Proxy(yr, {
  get(e, r) {
    if (!(typeof r != "string" || r === "then"))
      return t(r);
  }
});
function gn(t) {
  const { path: e, error: r, config: n } = t, { code: s } = t.error, a = {
    message: r.message,
    code: $e[s],
    data: {
      code: s,
      httpStatus: mr(r)
    }
  };
  return n.isDev && typeof t.error.stack == "string" && (a.data.stack = t.error.stack), typeof e == "string" && (a.data.path = e), n.errorFormatter({
    ...t,
    shape: a
  });
}
function Bt(t, e) {
  return "error" in e ? {
    ...e,
    error: t.transformer.output.serialize(e.error)
  } : "data" in e.result ? {
    ...e,
    result: {
      ...e.result,
      data: t.transformer.output.serialize(e.result.data)
    }
  } : e;
}
function _n(t, e) {
  return Array.isArray(e) ? e.map((r) => Bt(t, r)) : Bt(t, e);
}
function vn(t) {
  return !!t && !Array.isArray(t) && typeof t == "object";
}
class bn extends Error {
}
function Ot(t) {
  if (t instanceof Error)
    return t;
  const e = typeof t;
  if (!(e === "undefined" || e === "function" || t === null)) {
    if (e !== "object")
      return new Error(String(t));
    if (vn(t)) {
      const r = new bn();
      for (const n in t)
        r[n] = t[n];
      return r;
    }
  }
}
const wn = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  createFlatProxy: Rt,
  createRecursiveProxy: Et,
  getCauseFromUnknown: Ot,
  getErrorShape: gn,
  transformTRPCResponse: _n
}, Symbol.toStringTag, { value: "Module" }));
function Ft(t) {
  return !!t && !Array.isArray(t) && typeof t == "object";
}
function xn(t) {
  return t instanceof _r || /**
  * @deprecated
  * Delete in next major
  */
  t instanceof Error && t.name === "TRPCClientError";
}
function kn(t) {
  return Ft(t) && Ft(t.error) && typeof t.error.code == "number" && typeof t.error.message == "string";
}
let _r = class _e extends Error {
  static from(e, r = {}) {
    const n = e;
    return xn(n) ? (r.meta && (n.meta = {
      ...n.meta,
      ...r.meta
    }), n) : kn(n) ? new _e(n.error.message, {
      ...r,
      result: n
    }) : n instanceof Error ? new _e(n.message, {
      ...r,
      cause: Ot(n)
    }) : new _e("Unknown error", {
      ...r,
      cause: n
    });
  }
  constructor(e, r) {
    const n = r?.cause;
    super(e, {
      cause: n
    }), this.meta = r?.meta, this.cause = n, this.shape = r?.result?.error, this.data = r?.result?.error.data, this.name = "TRPCClientError", Object.setPrototypeOf(this, _e.prototype);
  }
};
function Tn(t) {
  return typeof FormData > "u" ? !1 : t instanceof FormData;
}
const Ve = {
  css: {
    query: [
      "72e3ff",
      "3fb0d8"
    ],
    mutation: [
      "c5a3fc",
      "904dfc"
    ],
    subscription: [
      "ff49e1",
      "d83fbe"
    ]
  },
  ansi: {
    regular: {
      // Cyan background, black and white text respectively
      query: [
        "\x1B[30;46m",
        "\x1B[97;46m"
      ],
      // Magenta background, black and white text respectively
      mutation: [
        "\x1B[30;45m",
        "\x1B[97;45m"
      ],
      // Green background, black and white text respectively
      subscription: [
        "\x1B[30;42m",
        "\x1B[97;42m"
      ]
    },
    bold: {
      query: [
        "\x1B[1;30;46m",
        "\x1B[1;97;46m"
      ],
      mutation: [
        "\x1B[1;30;45m",
        "\x1B[1;97;45m"
      ],
      subscription: [
        "\x1B[1;30;42m",
        "\x1B[1;97;42m"
      ]
    }
  }
};
function Cn(t) {
  const { direction: e, type: r, path: n, id: s, input: a } = t, i = [], o = [];
  if (t.colorMode === "ansi") {
    const [h, y] = Ve.ansi.regular[r], [k, C] = Ve.ansi.bold[r], A = "\x1B[0m";
    return i.push(e === "up" ? h : y, e === "up" ? ">>" : "<<", r, e === "up" ? k : C, `#${s}`, n, A), e === "up" ? o.push({
      input: t.input
    }) : o.push({
      input: t.input,
      // strip context from result cause it's too noisy in terminal wihtout collapse mode
      result: "result" in t.result ? t.result.result : t.result,
      elapsedMs: t.elapsedMs
    }), {
      parts: i,
      args: o
    };
  }
  const [c, u] = Ve.css[r], d = `
    background-color: #${e === "up" ? c : u}; 
    color: ${e === "up" ? "black" : "white"};
    padding: 2px;
  `;
  return i.push("%c", e === "up" ? ">>" : "<<", r, `#${s}`, `%c${n}%c`, "%O"), o.push(d, `${d}; font-weight: bold;`, `${d}; font-weight: normal;`), e === "up" ? o.push({
    input: a,
    context: t.context
  }) : o.push({
    input: a,
    result: t.result,
    elapsedMs: t.elapsedMs,
    context: t.context
  }), {
    parts: i,
    args: o
  };
}
const En = ({ c: t = console, colorMode: e = "css" }) => (r) => {
  const n = r.input, s = Tn(n) ? Object.fromEntries(n) : n, { parts: a, args: i } = Cn({
    ...r,
    colorMode: e,
    input: s
  }), o = r.direction === "down" && r.result && (r.result instanceof Error || "error" in r.result.result) ? "error" : "log";
  t[o].apply(null, [
    a.join(" ")
  ].concat(i));
};
function Rn(t = {}) {
  const { enabled: e = () => !0 } = t, r = t.colorMode ?? (typeof window > "u" ? "ansi" : "css"), { logger: n = En({
    c: t.console,
    colorMode: r
  }) } = t;
  return () => ({ op: s, next: a }) => Ie((i) => {
    e({
      ...s,
      direction: "up"
    }) && n({
      ...s,
      direction: "up"
    });
    const o = Date.now();
    function c(u) {
      const d = Date.now() - o;
      e({
        ...s,
        direction: "down",
        result: u
      }) && n({
        ...s,
        direction: "down",
        elapsedMs: d,
        result: u
      });
    }
    return a(s).pipe(dr({
      next(u) {
        c(u);
      },
      error(u) {
        c(u);
      }
    })).subscribe(i);
  });
}
let On = class {
  $request({ type: e, input: r, path: n, context: s = {} }) {
    return pr({
      links: this.links,
      op: {
        id: ++this.requestId,
        type: e,
        path: n,
        input: r,
        context: s
      }
    }).pipe(ur());
  }
  requestAsPromise(e) {
    const r = this.$request(e), { promise: n, abort: s } = fr(r);
    return new Promise((i, o) => {
      e.signal?.addEventListener("abort", s), n.then((c) => {
        i(c.result.data);
      }).catch((c) => {
        o(_r.from(c));
      });
    });
  }
  query(e, r, n) {
    return this.requestAsPromise({
      type: "query",
      path: e,
      input: r,
      context: n?.context,
      signal: n?.signal
    });
  }
  mutation(e, r, n) {
    return this.requestAsPromise({
      type: "mutation",
      path: e,
      input: r,
      context: n?.context,
      signal: n?.signal
    });
  }
  subscription(e, r, n) {
    return this.$request({
      type: "subscription",
      path: e,
      input: r,
      context: n?.context
    }).subscribe({
      next(a) {
        a.result.type === "started" ? n.onStarted?.() : a.result.type === "stopped" ? n.onStopped?.() : n.onData?.(a.result.data);
      },
      error(a) {
        n.onError?.(a);
      },
      complete() {
        n.onComplete?.();
      }
    });
  }
  constructor(e) {
    this.requestId = 0;
    const r = (() => {
      const n = e.transformer;
      return n ? "input" in n ? e.transformer : {
        input: n,
        output: n
      } : {
        input: {
          serialize: (s) => s,
          deserialize: (s) => s
        },
        output: {
          serialize: (s) => s,
          deserialize: (s) => s
        }
      };
    })();
    this.runtime = {
      transformer: {
        serialize: (n) => r.input.serialize(n),
        deserialize: (n) => r.output.deserialize(n)
      },
      combinedTransformer: r
    }, this.links = e.links.map((n) => n(this.runtime));
  }
};
const Sn = {
  query: "query",
  mutate: "mutation",
  subscribe: "subscription"
}, Pn = (t) => Sn[t];
function Nn(t) {
  return Rt((e) => t.hasOwnProperty(e) ? t[e] : e === "__untypedClient" ? t : Et(({ path: r, args: n }) => {
    const s = [
      e,
      ...r
    ], a = Pn(s.pop()), i = s.join(".");
    return t[a](i, ...n);
  }));
}
function An(t) {
  const e = new On(t);
  return Nn(e);
}
function Ln(t) {
  if (t instanceof oe || t instanceof Error && t.name === "TRPCError")
    return t;
  const e = new oe({
    code: "INTERNAL_SERVER_ERROR",
    cause: t
  });
  return t instanceof Error && t.stack && (e.stack = t.stack), e;
}
class oe extends Error {
  constructor(e) {
    const r = Ot(e.cause), n = e.message ?? r?.message ?? e.code;
    super(n, {
      cause: r
    }), this.code = e.code, this.name = "TRPCError", this.cause || (this.cause = r);
  }
}
function In(t) {
  return "input" in t ? t : {
    input: t,
    output: t
  };
}
const we = {
  _default: !0,
  input: {
    serialize: (t) => t,
    deserialize: (t) => t
  },
  output: {
    serialize: (t) => t,
    deserialize: (t) => t
  }
}, xe = ({ shape: t }) => t;
function $n(t) {
  return Object.assign(/* @__PURE__ */ Object.create(null), t);
}
const jn = [
  "query",
  "mutation",
  "subscription"
];
function Mn(t) {
  return "router" in t._def;
}
const Dn = {
  _ctx: null,
  _errorShape: null,
  _meta: null,
  queries: {},
  mutations: {},
  subscriptions: {},
  errorFormatter: xe,
  transformer: we
}, Zn = [
  /**
  * Then is a reserved word because otherwise we can't return a promise that returns a Proxy
  * since JS will think that `.then` is something that exists
  */
  "then"
];
function vr(t) {
  return function(r) {
    const n = new Set(Object.keys(r).filter((c) => Zn.includes(c)));
    if (n.size > 0)
      throw new Error("Reserved words used in `router({})` call: " + Array.from(n).join(", "));
    const s = $n({});
    function a(c, u = "") {
      for (const [d, h] of Object.entries(c ?? {})) {
        const y = `${u}${d}`;
        if (Mn(h)) {
          a(h._def.procedures, `${y}.`);
          continue;
        }
        if (s[y])
          throw new Error(`Duplicate key: ${y}`);
        s[y] = h;
      }
    }
    a(r);
    const i = {
      _config: t,
      router: !0,
      procedures: s,
      ...Dn,
      record: r,
      queries: Object.entries(s).filter((c) => c[1]._def.query).reduce((c, [u, d]) => ({
        ...c,
        [u]: d
      }), {}),
      mutations: Object.entries(s).filter((c) => c[1]._def.mutation).reduce((c, [u, d]) => ({
        ...c,
        [u]: d
      }), {}),
      subscriptions: Object.entries(s).filter((c) => c[1]._def.subscription).reduce((c, [u, d]) => ({
        ...c,
        [u]: d
      }), {})
    }, o = {
      ...r,
      _def: i,
      createCaller(c) {
        return br()(o)(c);
      },
      getErrorShape(c) {
        const { path: u, error: d } = c, { code: h } = c.error, y = {
          message: d.message,
          code: $e[h],
          data: {
            code: h,
            httpStatus: mr(d)
          }
        };
        return t.isDev && typeof c.error.stack == "string" && (y.data.stack = c.error.stack), typeof u == "string" && (y.data.path = u), this._def._config.errorFormatter({
          ...c,
          shape: y
        });
      }
    };
    return o;
  };
}
function Un(t) {
  const { type: e, path: r } = t;
  if (!(r in t.procedures) || !t.procedures[r]?._def[e])
    throw new oe({
      code: "NOT_FOUND",
      message: `No "${e}"-procedure on path "${r}"`
    });
  const n = t.procedures[r];
  return n(t);
}
function br() {
  return function(e) {
    const r = e._def;
    return function(s) {
      return Et(({ path: i, args: o }) => {
        if (i.length === 1 && jn.includes(i[0]))
          return Un({
            procedures: r.procedures,
            path: o[0],
            rawInput: o[1],
            ctx: s,
            type: i[0]
          });
        const c = i.join("."), u = r.procedures[c];
        let d = "query";
        return u._def.mutation ? d = "mutation" : u._def.subscription && (d = "subscription"), u({
          path: c,
          rawInput: o[0],
          ctx: s,
          type: d
        });
      });
    };
  };
}
const Vt = typeof window > "u" || "Deno" in window || globalThis.process?.env?.NODE_ENV === "test" || !!globalThis.process?.env?.JEST_WORKER_ID || !!globalThis.process?.env?.VITEST_WORKER_ID;
function Wt(t) {
  const e = t;
  if (typeof e == "function")
    return e;
  if (typeof e.parseAsync == "function")
    return e.parseAsync.bind(e);
  if (typeof e.parse == "function")
    return e.parse.bind(e);
  if (typeof e.validateSync == "function")
    return e.validateSync.bind(e);
  if (typeof e.create == "function")
    return e.create.bind(e);
  if (typeof e.assert == "function")
    return (r) => (e.assert(r), r);
  throw new Error("Could not find a validator fn");
}
function wr(t, ...e) {
  const r = Object.assign(/* @__PURE__ */ Object.create(null), t);
  for (const n of e)
    for (const s in n) {
      if (s in r && r[s] !== n[s])
        throw new Error(`Duplicate key ${s}`);
      r[s] = n[s];
    }
  return r;
}
function qn() {
  function t(r) {
    return {
      _middlewares: r,
      unstable_pipe(n) {
        const s = "_middlewares" in n ? n._middlewares : [
          n
        ];
        return t([
          ...r,
          ...s
        ]);
      }
    };
  }
  function e(r) {
    return t([
      r
    ]);
  }
  return e;
}
function Ht(t) {
  return t && typeof t == "object" && !Array.isArray(t);
}
function zn(t) {
  const e = async ({ next: r, rawInput: n, input: s }) => {
    let a;
    try {
      a = await t(n);
    } catch (o) {
      throw new oe({
        code: "BAD_REQUEST",
        cause: o
      });
    }
    const i = Ht(s) && Ht(a) ? {
      ...s,
      ...a
    } : a;
    return r({
      input: i
    });
  };
  return e._type = "input", e;
}
function Bn(t) {
  const e = async ({ next: r }) => {
    const n = await r();
    if (!n.ok)
      return n;
    try {
      const s = await t(n.data);
      return {
        ...n,
        data: s
      };
    } catch (s) {
      throw new oe({
        message: "Output validation failed",
        code: "INTERNAL_SERVER_ERROR",
        cause: s
      });
    }
  };
  return e._type = "output", e;
}
const xr = "middlewareMarker";
function ne(t, e) {
  const { middlewares: r = [], inputs: n, meta: s, ...a } = e;
  return kr({
    ...wr(t, a),
    inputs: [
      ...t.inputs,
      ...n ?? []
    ],
    middlewares: [
      ...t.middlewares,
      ...r
    ],
    meta: t.meta && s ? {
      ...t.meta,
      ...s
    } : s ?? t.meta
  });
}
function kr(t = {}) {
  const e = {
    inputs: [],
    middlewares: [],
    ...t
  };
  return {
    _def: e,
    input(r) {
      const n = Wt(r);
      return ne(e, {
        inputs: [
          r
        ],
        middlewares: [
          zn(n)
        ]
      });
    },
    output(r) {
      const n = Wt(r);
      return ne(e, {
        output: r,
        middlewares: [
          Bn(n)
        ]
      });
    },
    meta(r) {
      return ne(e, {
        meta: r
      });
    },
    /**
    * @deprecated
    * This functionality is deprecated and will be removed in the next major version.
    */
    unstable_concat(r) {
      return ne(e, r._def);
    },
    use(r) {
      const n = "_middlewares" in r ? r._middlewares : [
        r
      ];
      return ne(e, {
        middlewares: n
      });
    },
    query(r) {
      return We({
        ...e,
        query: !0
      }, r);
    },
    mutation(r) {
      return We({
        ...e,
        mutation: !0
      }, r);
    },
    subscription(r) {
      return We({
        ...e,
        subscription: !0
      }, r);
    }
  };
}
function We(t, e) {
  const r = ne(t, {
    resolver: e,
    middlewares: [
      async function(s) {
        const a = await e(s);
        return {
          marker: xr,
          ok: !0,
          data: a,
          ctx: s.ctx
        };
      }
    ]
  });
  return Vn(r._def);
}
const Fn = `
This is a client-only function.
If you want to call this function on the server, see https://trpc.io/docs/server/server-side-calls
`.trim();
function Vn(t) {
  const e = async function(n) {
    if (!n || !("rawInput" in n))
      throw new Error(Fn);
    const s = async (i = {
      index: 0,
      ctx: n.ctx
    }) => {
      try {
        const o = t.middlewares[i.index];
        return await o({
          ctx: i.ctx,
          type: n.type,
          path: n.path,
          rawInput: i.rawInput ?? n.rawInput,
          meta: t.meta,
          input: i.input,
          next(u) {
            const d = u;
            return s({
              index: i.index + 1,
              ctx: d && "ctx" in d ? {
                ...i.ctx,
                ...d.ctx
              } : i.ctx,
              input: d && "input" in d ? d.input : i.input,
              rawInput: d && "rawInput" in d ? d.rawInput : i.rawInput
            });
          }
        });
      } catch (o) {
        return {
          ok: !1,
          error: Ln(o),
          marker: xr
        };
      }
    }, a = await s();
    if (!a)
      throw new oe({
        code: "INTERNAL_SERVER_ERROR",
        message: "No result from middlewares - did you forget to `return next()`?"
      });
    if (!a.ok)
      throw a.error;
    return a.data;
  };
  return e._def = t, e.meta = t.meta, e;
}
function Wn(...t) {
  const e = wr({}, ...t.map((a) => a._def.record)), r = t.reduce((a, i) => {
    if (i._def._config.errorFormatter && i._def._config.errorFormatter !== xe) {
      if (a !== xe && a !== i._def._config.errorFormatter)
        throw new Error("You seem to have several error formatters");
      return i._def._config.errorFormatter;
    }
    return a;
  }, xe), n = t.reduce((a, i) => {
    if (i._def._config.transformer && i._def._config.transformer !== we) {
      if (a !== we && a !== i._def._config.transformer)
        throw new Error("You seem to have several transformers");
      return i._def._config.transformer;
    }
    return a;
  }, we);
  return vr({
    errorFormatter: r,
    transformer: n,
    isDev: t.some((a) => a._def._config.isDev),
    allowOutsideOfServer: t.some((a) => a._def._config.allowOutsideOfServer),
    isServer: t.some((a) => a._def._config.isServer),
    $types: t[0]?._def._config.$types
  })(e);
}
class Oe {
  context() {
    return new Oe();
  }
  meta() {
    return new Oe();
  }
  create(e) {
    return Jn()(e);
  }
}
const Hn = new Oe();
function Jn() {
  return function(e) {
    const r = e?.errorFormatter ?? xe, s = {
      transformer: In(e?.transformer ?? we),
      isDev: e?.isDev ?? globalThis.process?.env?.NODE_ENV !== "production",
      allowOutsideOfServer: e?.allowOutsideOfServer ?? !1,
      errorFormatter: r,
      isServer: e?.isServer ?? Vt,
      /**
      * @internal
      */
      $types: Rt((a) => {
        throw new Error(`Tried to access "$types.${a}" which is not available at runtime`);
      })
    };
    if (!(e?.isServer ?? Vt) && e?.allowOutsideOfServer !== !0)
      throw new Error("You're trying to use @trpc/server in a non-server environment. This is not supported by default.");
    return {
      /**
      * These are just types, they can't be used
      * @internal
      */
      _config: s,
      /**
      * Builder object for creating procedures
      * @see https://trpc.io/docs/server/procedures
      */
      procedure: kr({
        meta: e?.defaultMeta
      }),
      /**
      * Create reusable middlewares
      * @see https://trpc.io/docs/server/middlewares
      */
      middleware: qn(),
      /**
      * Create a router
      * @see https://trpc.io/docs/server/routers
      */
      router: vr(s),
      /**
      * Merge Routers
      * @see https://trpc.io/docs/server/merging-routers
      */
      mergeRouters: Wn,
      /**
      * Create a server-side caller for a router
      * @see https://trpc.io/docs/server/server-side-calls
      */
      createCallerFactory: br()
    };
  };
}
var tt = {}, je = {}, he = {}, P = {};
const me = /* @__PURE__ */ cr(pn);
var St = {}, Tr = me;
function Cr(t) {
  return Tr.observable((e) => {
    function r(s = 0, a = t.op) {
      const i = t.links[s];
      if (!i)
        throw new Error("No more links to execute - did you forget to add an ending link?");
      return i({
        op: a,
        next(c) {
          return r(s + 1, c);
        }
      });
    }
    return r().subscribe(e);
  });
}
function Jt(t) {
  return Array.isArray(t) ? t : [
    t
  ];
}
function Yn(t) {
  return (e) => {
    const r = Jt(t.true).map((s) => s(e)), n = Jt(t.false).map((s) => s(e));
    return (s) => Tr.observable((a) => {
      const i = t.condition(s.op) ? r : n;
      return Cr({
        op: s.op,
        links: i
      }).subscribe(a);
    });
  };
}
St.createChain = Cr;
St.splitLink = Yn;
var ye = {};
const Er = /* @__PURE__ */ cr(wn);
var ge = {};
function rt(t) {
  return !!t && !Array.isArray(t) && typeof t == "object";
}
function Gn(t, e) {
  if ("error" in t) {
    const n = e.transformer.deserialize(t.error);
    return {
      ok: !1,
      error: {
        ...t,
        error: n
      }
    };
  }
  return {
    ok: !0,
    result: {
      ...t.result,
      ...(!t.result.type || t.result.type === "data") && {
        type: "data",
        data: e.transformer.deserialize(t.result.data)
      }
    }
  };
}
class He extends Error {
  constructor() {
    super("Unable to transform response from server");
  }
}
function Qn(t, e) {
  let r;
  try {
    r = Gn(t, e);
  } catch {
    throw new He();
  }
  if (!r.ok && (!rt(r.error.error) || typeof r.error.error.code != "number"))
    throw new He();
  if (r.ok && !rt(r.result))
    throw new He();
  return r;
}
ge.isObject = rt;
ge.transformResult = Qn;
var Xn = Er, Yt = ge;
function Kn(t) {
  return t instanceof Rr || /**
  * @deprecated
  * Delete in next major
  */
  t instanceof Error && t.name === "TRPCClientError";
}
function es(t) {
  return Yt.isObject(t) && Yt.isObject(t.error) && typeof t.error.code == "number" && typeof t.error.message == "string";
}
let Rr = class ve extends Error {
  static from(e, r = {}) {
    const n = e;
    return Kn(n) ? (r.meta && (n.meta = {
      ...n.meta,
      ...r.meta
    }), n) : es(n) ? new ve(n.error.message, {
      ...r,
      result: n
    }) : n instanceof Error ? new ve(n.message, {
      ...r,
      cause: Xn.getCauseFromUnknown(n)
    }) : new ve("Unknown error", {
      ...r,
      cause: n
    });
  }
  constructor(e, r) {
    const n = r?.cause;
    super(e, {
      cause: n
    }), this.meta = r?.meta, this.cause = n, this.shape = r?.result?.error, this.data = r?.result?.error.data, this.name = "TRPCClientError", Object.setPrototypeOf(this, ve.prototype);
  }
};
ye.TRPCClientError = Rr;
var B = {}, ts = ye;
const Gt = (t) => typeof t == "function";
function Or(t) {
  if (t)
    return t;
  if (typeof window < "u" && Gt(window.fetch))
    return window.fetch;
  if (typeof globalThis < "u" && Gt(globalThis.fetch))
    return globalThis.fetch;
  throw new Error("No fetch implementation found");
}
function rs(t) {
  return t || (typeof window < "u" && window.AbortController ? window.AbortController : typeof globalThis < "u" && globalThis.AbortController ? globalThis.AbortController : null);
}
function ns(t) {
  return {
    url: t.url.toString().replace(/\/$/, ""),
    fetch: t.fetch,
    AbortController: rs(t.AbortController)
  };
}
function ss(t) {
  const e = {};
  for (let r = 0; r < t.length; r++) {
    const n = t[r];
    e[r] = n;
  }
  return e;
}
const as = {
  query: "GET",
  mutation: "POST"
};
function Sr(t) {
  return "input" in t ? t.runtime.transformer.serialize(t.input) : ss(t.inputs.map((e) => t.runtime.transformer.serialize(e)));
}
const Pr = (t) => {
  let e = t.url + "/" + t.path;
  const r = [];
  if ("inputs" in t && r.push("batch=1"), t.type === "query") {
    const n = Sr(t);
    n !== void 0 && r.push(`input=${encodeURIComponent(JSON.stringify(n))}`);
  }
  return r.length && (e += "?" + r.join("&")), e;
}, Nr = (t) => {
  if (t.type === "query")
    return;
  const e = Sr(t);
  return e !== void 0 ? JSON.stringify(e) : void 0;
}, is = (t) => Lr({
  ...t,
  contentTypeHeader: "application/json",
  getUrl: Pr,
  getBody: Nr
});
async function Ar(t, e) {
  const r = t.getUrl(t), n = t.getBody(t), { type: s } = t, a = await t.headers();
  /* istanbul ignore if -- @preserve */
  if (s === "subscription")
    throw new Error("Subscriptions should use wsLink");
  const i = {
    ...t.contentTypeHeader ? {
      "content-type": t.contentTypeHeader
    } : {},
    ...t.batchModeHeader ? {
      "trpc-batch-mode": t.batchModeHeader
    } : {},
    ...a
  };
  return Or(t.fetch)(r, {
    method: as[s],
    signal: e?.signal,
    body: n,
    headers: i
  });
}
function Lr(t) {
  const e = t.AbortController ? new t.AbortController() : null, r = {};
  let n = !1;
  return {
    promise: new Promise((i, o) => {
      Ar(t, e).then((c) => (r.response = c, n = !0, c.json())).then((c) => {
        r.responseJSON = c, i({
          json: c,
          meta: r
        });
      }).catch((c) => {
        n = !0, o(ts.TRPCClientError.from(c, {
          meta: r
        }));
      });
    }),
    cancel: () => {
      n || e?.abort();
    }
  };
}
B.fetchHTTPResponse = Ar;
B.getBody = Nr;
B.getFetch = Or;
B.getUrl = Pr;
B.httpRequest = Lr;
B.jsonHttpRequester = is;
B.resolveHTTPLinkOptions = ns;
var Pt = {}, os = me, cs = ge, Qt = ye, nt = B;
const Je = () => {
  throw new Error("Something went wrong. Please submit an issue at https://github.com/trpc/trpc/issues/new");
};
function Ye(t) {
  let e = null, r = null;
  const n = () => {
    clearTimeout(r), r = null, e = null;
  };
  function s(o) {
    const c = [
      []
    ];
    let u = 0;
    for (; ; ) {
      const d = o[u];
      if (!d)
        break;
      const h = c[c.length - 1];
      if (d.aborted) {
        d.reject?.(new Error("Aborted")), u++;
        continue;
      }
      if (t.validate(h.concat(d).map((k) => k.key))) {
        h.push(d), u++;
        continue;
      }
      if (h.length === 0) {
        d.reject?.(new Error("Input is too big for a single dispatch")), u++;
        continue;
      }
      c.push([]);
    }
    return c;
  }
  function a() {
    const o = s(e);
    n();
    for (const c of o) {
      if (!c.length)
        continue;
      const u = {
        items: c,
        cancel: Je
      };
      for (const k of c)
        k.batch = u;
      const d = (k, C) => {
        const A = u.items[k];
        A.resolve?.(C), A.batch = null, A.reject = null, A.resolve = null;
      }, { promise: h, cancel: y } = t.fetch(u.items.map((k) => k.key), d);
      u.cancel = y, h.then((k) => {
        for (let C = 0; C < k.length; C++) {
          const A = k[C];
          d(C, A);
        }
        for (const C of u.items)
          C.reject?.(new Error("Missing result")), C.batch = null;
      }).catch((k) => {
        for (const C of u.items)
          C.reject?.(k), C.batch = null;
      });
    }
  }
  function i(o) {
    const c = {
      aborted: !1,
      key: o,
      batch: null,
      resolve: Je,
      reject: Je
    }, u = new Promise((h, y) => {
      c.reject = y, c.resolve = h, e || (e = []), e.push(c);
    });
    return r || (r = setTimeout(a)), {
      promise: u,
      cancel: () => {
        c.aborted = !0, c.batch?.items.every((h) => h.aborted) && (c.batch.cancel(), c.batch = null);
      }
    };
  }
  return {
    load: i
  };
}
function Ir(t) {
  return function(r) {
    const n = nt.resolveHTTPLinkOptions(r), s = r.maxURLLength ?? 1 / 0;
    return (a) => {
      const i = (h) => {
        const y = (C) => {
          if (s === 1 / 0)
            return !0;
          const A = C.map((F) => F.path).join(","), te = C.map((F) => F.input);
          return nt.getUrl({
            ...n,
            runtime: a,
            type: h,
            path: A,
            inputs: te
          }).length <= s;
        }, k = t({
          ...n,
          runtime: a,
          type: h,
          opts: r
        });
        return {
          validate: y,
          fetch: k
        };
      }, o = Ye(i("query")), c = Ye(i("mutation")), u = Ye(i("subscription")), d = {
        query: o,
        subscription: u,
        mutation: c
      };
      return ({ op: h }) => os.observable((y) => {
        const k = d[h.type], { promise: C, cancel: A } = k.load(h);
        let te;
        return C.then((j) => {
          te = j;
          const F = cs.transformResult(j.json, a);
          if (!F.ok) {
            y.error(Qt.TRPCClientError.from(F.error, {
              meta: j.meta
            }));
            return;
          }
          y.next({
            context: j.meta,
            result: F.result
          }), y.complete();
        }).catch((j) => {
          y.error(Qt.TRPCClientError.from(j, {
            meta: te?.meta
          }));
        }), () => {
          A();
        };
      });
    };
  };
}
const us = (t) => (e) => {
  const r = e.map((i) => i.path).join(","), n = e.map((i) => i.input), { promise: s, cancel: a } = nt.jsonHttpRequester({
    ...t,
    path: r,
    inputs: n,
    headers() {
      return t.opts.headers ? typeof t.opts.headers == "function" ? t.opts.headers({
        opList: e
      }) : t.opts.headers : {};
    }
  });
  return {
    promise: s.then((i) => (Array.isArray(i.json) ? i.json : e.map(() => i.json)).map((u) => ({
      meta: i.meta,
      json: u
    }))),
    cancel: a
  };
}, ds = Ir(us);
Pt.createHTTPBatchLink = Ir;
Pt.httpBatchLink = ds;
var Me = {};
Object.defineProperty(Me, "__esModule", { value: !0 });
var ls = me, fs = ge, Xt = ye, $r = B;
function jr(t) {
  return (e) => {
    const r = $r.resolveHTTPLinkOptions(e);
    return (n) => ({ op: s }) => ls.observable((a) => {
      const { path: i, input: o, type: c } = s, { promise: u, cancel: d } = t.requester({
        ...r,
        runtime: n,
        type: c,
        path: i,
        input: o,
        headers() {
          return e.headers ? typeof e.headers == "function" ? e.headers({
            op: s
          }) : e.headers : {};
        }
      });
      let h;
      return u.then((y) => {
        h = y.meta;
        const k = fs.transformResult(y.json, n);
        if (!k.ok) {
          a.error(Xt.TRPCClientError.from(k.error, {
            meta: h
          }));
          return;
        }
        a.next({
          context: y.meta,
          result: k.result
        }), a.complete();
      }).catch((y) => {
        a.error(Xt.TRPCClientError.from(y, {
          meta: h
        }));
      }), () => {
        d();
      };
    });
  };
}
const ps = jr({
  requester: $r.jsonHttpRequester
});
Me.httpLink = ps;
Me.httpLinkFactory = jr;
var Nt = {};
Object.defineProperty(Nt, "__esModule", { value: !0 });
var Kt = me;
function hs(t) {
  return typeof FormData > "u" ? !1 : t instanceof FormData;
}
const Ge = {
  css: {
    query: [
      "72e3ff",
      "3fb0d8"
    ],
    mutation: [
      "c5a3fc",
      "904dfc"
    ],
    subscription: [
      "ff49e1",
      "d83fbe"
    ]
  },
  ansi: {
    regular: {
      // Cyan background, black and white text respectively
      query: [
        "\x1B[30;46m",
        "\x1B[97;46m"
      ],
      // Magenta background, black and white text respectively
      mutation: [
        "\x1B[30;45m",
        "\x1B[97;45m"
      ],
      // Green background, black and white text respectively
      subscription: [
        "\x1B[30;42m",
        "\x1B[97;42m"
      ]
    },
    bold: {
      query: [
        "\x1B[1;30;46m",
        "\x1B[1;97;46m"
      ],
      mutation: [
        "\x1B[1;30;45m",
        "\x1B[1;97;45m"
      ],
      subscription: [
        "\x1B[1;30;42m",
        "\x1B[1;97;42m"
      ]
    }
  }
};
function ms(t) {
  const { direction: e, type: r, path: n, id: s, input: a } = t, i = [], o = [];
  if (t.colorMode === "ansi") {
    const [h, y] = Ge.ansi.regular[r], [k, C] = Ge.ansi.bold[r], A = "\x1B[0m";
    return i.push(e === "up" ? h : y, e === "up" ? ">>" : "<<", r, e === "up" ? k : C, `#${s}`, n, A), e === "up" ? o.push({
      input: t.input
    }) : o.push({
      input: t.input,
      // strip context from result cause it's too noisy in terminal wihtout collapse mode
      result: "result" in t.result ? t.result.result : t.result,
      elapsedMs: t.elapsedMs
    }), {
      parts: i,
      args: o
    };
  }
  const [c, u] = Ge.css[r], d = `
    background-color: #${e === "up" ? c : u}; 
    color: ${e === "up" ? "black" : "white"};
    padding: 2px;
  `;
  return i.push("%c", e === "up" ? ">>" : "<<", r, `#${s}`, `%c${n}%c`, "%O"), o.push(d, `${d}; font-weight: bold;`, `${d}; font-weight: normal;`), e === "up" ? o.push({
    input: a,
    context: t.context
  }) : o.push({
    input: a,
    result: t.result,
    elapsedMs: t.elapsedMs,
    context: t.context
  }), {
    parts: i,
    args: o
  };
}
const ys = ({ c: t = console, colorMode: e = "css" }) => (r) => {
  const n = r.input, s = hs(n) ? Object.fromEntries(n) : n, { parts: a, args: i } = ms({
    ...r,
    colorMode: e,
    input: s
  }), o = r.direction === "down" && r.result && (r.result instanceof Error || "error" in r.result.result) ? "error" : "log";
  t[o].apply(null, [
    a.join(" ")
  ].concat(i));
};
function gs(t = {}) {
  const { enabled: e = () => !0 } = t, r = t.colorMode ?? (typeof window > "u" ? "ansi" : "css"), { logger: n = ys({
    c: t.console,
    colorMode: r
  }) } = t;
  return () => ({ op: s, next: a }) => Kt.observable((i) => {
    e({
      ...s,
      direction: "up"
    }) && n({
      ...s,
      direction: "up"
    });
    const o = Date.now();
    function c(u) {
      const d = Date.now() - o;
      e({
        ...s,
        direction: "down",
        result: u
      }) && n({
        ...s,
        direction: "down",
        elapsedMs: d,
        result: u
      });
    }
    return a(s).pipe(Kt.tap({
      next(u) {
        c(u);
      },
      error(u) {
        c(u);
      }
    })).subscribe(i);
  });
}
Nt.loggerLink = gs;
var De = {};
Object.defineProperty(De, "__esModule", { value: !0 });
var _s = me, vs = ge, Mr = ye;
/* istanbul ignore next -- @preserve */
const bs = (t) => t === 0 ? 0 : Math.min(1e3 * 2 ** t, 3e4);
function ws(t) {
  const { url: e, WebSocket: r = WebSocket, retryDelayMs: n = bs, onOpen: s, onClose: a } = t;
  /* istanbul ignore next -- @preserve */
  if (!r)
    throw new Error("No WebSocket implementation found - you probably don't want to use this on the server, but if you do you need to pass a `WebSocket`-ponyfill");
  let i = [];
  const o = /* @__PURE__ */ Object.create(null);
  let c = 0, u = null, d = null, h = Zt(), y = "connecting";
  function k() {
    y !== "open" || u || (u = setTimeout(() => {
      u = null, i.length === 1 ? h.send(JSON.stringify(i.pop())) : h.send(JSON.stringify(i)), i = [];
    }));
  }
  function C() {
    if (d !== null || y === "closed")
      return;
    const O = n(c++);
    te(O);
  }
  function A() {
    y = "connecting";
    const O = h;
    h = Zt(), j(O);
  }
  function te(O) {
    d || (y = "connecting", d = setTimeout(A, O));
  }
  function j(O) {
    Object.values(o).some((re) => re.ws === O) || O.close();
  }
  function F() {
    Object.values(o).forEach((O) => {
      O.type === "subscription" && O.callbacks.complete();
    });
  }
  function Dt(O) {
    i.some((N) => N.id === O.op.id) || Ut(O.op, O.callbacks);
  }
  function Zt() {
    const O = typeof e == "function" ? e() : e, N = new r(O);
    clearTimeout(d), d = null, N.addEventListener("open", () => {
      /* istanbul ignore next -- @preserve */
      N === h && (c = 0, y = "open", s?.(), k());
    }), N.addEventListener("error", () => {
      N === h && C();
    });
    const re = ($) => {
      if ($.method === "reconnect" && N === h) {
        y === "open" && a?.(), A();
        for (const E of Object.values(o))
          E.type === "subscription" && Dt(E);
      }
    }, Fe = ($) => {
      const E = $.id !== null && o[$.id];
      if (E) {
        if (E.callbacks.next?.($), E.ws !== h && N === h) {
          const V = E.ws;
          E.ws = h, j(V);
        }
        "result" in $ && $.result.type === "stopped" && N === h && E.callbacks.complete();
      }
    };
    return N.addEventListener("message", ({ data: $ }) => {
      const E = JSON.parse($);
      "method" in E ? re(E) : Fe(E), (N !== h || y === "closed") && j(N);
    }), N.addEventListener("close", ({ code: $ }) => {
      y === "open" && a?.({
        code: $
      }), h === N && C();
      for (const [E, V] of Object.entries(o))
        if (V.ws === N) {
          if (y === "closed") {
            delete o[E], V.callbacks.complete?.();
            continue;
          }
          V.type === "subscription" ? Dt(V) : (delete o[E], V.callbacks.error?.(Mr.TRPCClientError.from(new At("WebSocket closed prematurely"))));
        }
    }), N;
  }
  function Ut(O, N) {
    const { type: re, input: Fe, path: $, id: E } = O, V = {
      id: E,
      method: re,
      params: {
        input: Fe,
        path: $
      }
    };
    return o[E] = {
      ws: h,
      type: re,
      callbacks: N,
      op: O
    }, i.push(V), k(), () => {
      const nn = o[E]?.callbacks;
      delete o[E], i = i.filter((sn) => sn.id !== E), nn?.complete?.(), h.readyState === r.OPEN && O.type === "subscription" && (i.push({
        id: E,
        method: "subscription.stop"
      }), k());
    };
  }
  return {
    close: () => {
      y = "closed", a?.(), F(), j(h), clearTimeout(d), d = null;
    },
    request: Ut,
    getConnection() {
      return h;
    }
  };
}
class At extends Error {
  constructor(e) {
    super(e), this.name = "TRPCWebSocketClosedError", Object.setPrototypeOf(this, At.prototype);
  }
}
function xs(t) {
  return (e) => {
    const { client: r } = t;
    return ({ op: n }) => _s.observable((s) => {
      const { type: a, path: i, id: o, context: c } = n, u = e.transformer.serialize(n.input), d = r.request({
        type: a,
        path: i,
        input: u,
        id: o,
        context: c
      }, {
        error(h) {
          s.error(h), d();
        },
        complete() {
          s.complete();
        },
        next(h) {
          const y = vs.transformResult(h, e);
          if (!y.ok) {
            s.error(Mr.TRPCClientError.from(y.error));
            return;
          }
          s.next({
            result: y.result
          }), n.type !== "subscription" && (d(), s.complete());
        }
      });
      return () => {
        d();
      };
    });
  };
}
De.createWSClient = ws;
De.wsLink = xs;
Object.defineProperty(P, "__esModule", { value: !0 });
var er = me, Dr = St, Zr = ye, tr = Er, ke = B, Ur = Pt, Lt = Me, ks = Nt, qr = De;
class Ze {
  $request({ type: e, input: r, path: n, context: s = {} }) {
    return Dr.createChain({
      links: this.links,
      op: {
        id: ++this.requestId,
        type: e,
        path: n,
        input: r,
        context: s
      }
    }).pipe(er.share());
  }
  requestAsPromise(e) {
    const r = this.$request(e), { promise: n, abort: s } = er.observableToPromise(r);
    return new Promise((i, o) => {
      e.signal?.addEventListener("abort", s), n.then((c) => {
        i(c.result.data);
      }).catch((c) => {
        o(Zr.TRPCClientError.from(c));
      });
    });
  }
  query(e, r, n) {
    return this.requestAsPromise({
      type: "query",
      path: e,
      input: r,
      context: n?.context,
      signal: n?.signal
    });
  }
  mutation(e, r, n) {
    return this.requestAsPromise({
      type: "mutation",
      path: e,
      input: r,
      context: n?.context,
      signal: n?.signal
    });
  }
  subscription(e, r, n) {
    return this.$request({
      type: "subscription",
      path: e,
      input: r,
      context: n?.context
    }).subscribe({
      next(a) {
        a.result.type === "started" ? n.onStarted?.() : a.result.type === "stopped" ? n.onStopped?.() : n.onData?.(a.result.data);
      },
      error(a) {
        n.onError?.(a);
      },
      complete() {
        n.onComplete?.();
      }
    });
  }
  constructor(e) {
    this.requestId = 0;
    const r = (() => {
      const n = e.transformer;
      return n ? "input" in n ? e.transformer : {
        input: n,
        output: n
      } : {
        input: {
          serialize: (s) => s,
          deserialize: (s) => s
        },
        output: {
          serialize: (s) => s,
          deserialize: (s) => s
        }
      };
    })();
    this.runtime = {
      transformer: {
        serialize: (n) => r.input.serialize(n),
        deserialize: (n) => r.output.deserialize(n)
      },
      combinedTransformer: r
    }, this.links = e.links.map((n) => n(this.runtime));
  }
}
function Ts(t) {
  return new Ze(t);
}
function Cs(t) {
  return new Ze(t);
}
const Es = {
  query: "query",
  mutate: "mutation",
  subscribe: "subscription"
}, zr = (t) => Es[t];
function Br(t) {
  return tr.createFlatProxy((e) => t.hasOwnProperty(e) ? t[e] : e === "__untypedClient" ? t : tr.createRecursiveProxy(({ path: r, args: n }) => {
    const s = [
      e,
      ...r
    ], a = zr(s.pop()), i = s.join(".");
    return t[a](i, ...n);
  }));
}
function Rs(t) {
  const e = new Ze(t);
  return Br(e);
}
function Os(t) {
  return t.__untypedClient;
}
function Ss(t) {
  if (t)
    return t;
  if (typeof window < "u" && window.TextDecoder)
    return new window.TextDecoder();
  if (typeof globalThis < "u" && globalThis.TextDecoder)
    return new globalThis.TextDecoder();
  throw new Error("No TextDecoder implementation found");
}
async function Ps(t) {
  const e = t.parse ?? JSON.parse, r = (n) => {
    if (t.signal?.aborted || !n || n === "}")
      return;
    const s = n.indexOf(":"), a = n.substring(2, s - 1), i = n.substring(s + 1);
    t.onSingle(Number(a), e(i));
  };
  await Ns(t.readableStream, r, t.textDecoder);
}
async function Ns(t, e, r) {
  let n = "";
  const s = (a) => {
    const o = r.decode(a).split(`
`);
    if (o.length === 1)
      n += o[0];
    else if (o.length > 1) {
      e(n + o[0]);
      for (let c = 1; c < o.length - 1; c++)
        e(o[c]);
      n = o[o.length - 1];
    }
  };
  "getReader" in t ? await Ls(t, s) : await As(t, s), e(n);
}
function As(t, e) {
  return new Promise((r) => {
    t.on("data", e), t.on("end", r);
  });
}
async function Ls(t, e) {
  const r = t.getReader();
  let n = await r.read();
  for (; !n.done; )
    e(n.value), n = await r.read();
}
const Is = (t, e) => {
  const r = t.AbortController ? new t.AbortController() : null, n = ke.fetchHTTPResponse({
    ...t,
    contentTypeHeader: "application/json",
    batchModeHeader: "stream",
    getUrl: ke.getUrl,
    getBody: ke.getBody
  }, r), s = () => r?.abort(), a = n.then(async (i) => {
    if (!i.body)
      throw new Error("Received response without body");
    const o = {
      response: i
    };
    return Ps({
      readableStream: i.body,
      onSingle: e,
      parse: (c) => ({
        json: JSON.parse(c),
        meta: o
      }),
      signal: r?.signal,
      textDecoder: t.textDecoder
    });
  });
  return {
    cancel: s,
    promise: a
  };
}, $s = (t) => {
  const e = Ss(t.opts.textDecoder);
  return (r, n) => {
    const s = r.map((c) => c.path).join(","), a = r.map((c) => c.input), { cancel: i, promise: o } = Is({
      ...t,
      textDecoder: e,
      path: s,
      inputs: a,
      headers() {
        return t.opts.headers ? typeof t.opts.headers == "function" ? t.opts.headers({
          opList: r
        }) : t.opts.headers : {};
      }
    }, (c, u) => {
      n(c, u);
    });
    return {
      /**
      * return an empty array because the batchLoader expects an array of results
      * but we've already called the `unitResolver` for each of them, there's
      * nothing left to do here.
      */
      promise: o.then(() => []),
      cancel: i
    };
  };
}, js = Ur.createHTTPBatchLink($s), Ms = (t) => {
  if ("input" in t) {
    if (!(t.input instanceof FormData))
      throw new Error("Input is not FormData");
    return t.input;
  }
}, Ds = (t) => {
  if (t.type !== "mutation")
    throw new Error("We only handle mutations with formdata");
  return ke.httpRequest({
    ...t,
    getUrl() {
      return `${t.url}/${t.path}`;
    },
    getBody: Ms
  });
}, Zs = Lt.httpLinkFactory({
  requester: Ds
});
P.splitLink = Dr.splitLink;
P.TRPCClientError = Zr.TRPCClientError;
P.getFetch = ke.getFetch;
P.httpBatchLink = Ur.httpBatchLink;
P.httpLink = Lt.httpLink;
P.httpLinkFactory = Lt.httpLinkFactory;
P.loggerLink = ks.loggerLink;
P.createWSClient = qr.createWSClient;
P.wsLink = qr.wsLink;
P.TRPCUntypedClient = Ze;
P.clientCallTypeToProcedureType = zr;
P.createTRPCClient = Cs;
P.createTRPCClientProxy = Br;
P.createTRPCProxyClient = Rs;
P.createTRPCUntypedClient = Ts;
P.experimental_formDataLink = Zs;
P.getUntypedClient = Os;
P.unstable_httpBatchStreamLink = js;
var K = {}, It = {};
function Us(t) {
  return t;
}
function qs(t) {
  return t.length === 0 ? Us : t.length === 1 ? t[0] : function(r) {
    return t.reduce((n, s) => s(n), r);
  };
}
function zs(t) {
  return typeof t == "object" && t !== null && "subscribe" in t;
}
function Bs(t) {
  const e = {
    subscribe(r) {
      let n = null, s = !1, a = !1, i = !1;
      function o() {
        if (n === null) {
          i = !0;
          return;
        }
        a || (a = !0, typeof n == "function" ? n() : n && n.unsubscribe());
      }
      return n = t({
        next(c) {
          s || r.next?.(c);
        },
        error(c) {
          s || (s = !0, r.error?.(c), o());
        },
        complete() {
          s || (s = !0, r.complete?.(), o());
        }
      }), i && o(), {
        unsubscribe: o
      };
    },
    pipe(...r) {
      return qs(r)(e);
    }
  };
  return e;
}
It.isObservable = zs;
It.observable = Bs;
Object.defineProperty(K, "__esModule", { value: !0 });
var Fr = It;
function Fs(t) {
  return (e) => {
    let r = 0, n = null;
    const s = [];
    function a() {
      n || (n = e.subscribe({
        next(o) {
          for (const c of s)
            c.next?.(o);
        },
        error(o) {
          for (const c of s)
            c.error?.(o);
        },
        complete() {
          for (const o of s)
            o.complete?.();
        }
      }));
    }
    function i() {
      if (r === 0 && n) {
        const o = n;
        n = null, o.unsubscribe();
      }
    }
    return {
      subscribe(o) {
        return r++, s.push(o), a(), {
          unsubscribe() {
            r--, i();
            const c = s.findIndex((u) => u === o);
            c > -1 && s.splice(c, 1);
          }
        };
      }
    };
  };
}
function Vs(t) {
  return (e) => ({
    subscribe(r) {
      let n = 0;
      return e.subscribe({
        next(a) {
          r.next?.(t(a, n++));
        },
        error(a) {
          r.error?.(a);
        },
        complete() {
          r.complete?.();
        }
      });
    }
  });
}
function Ws(t) {
  return (e) => ({
    subscribe(r) {
      return e.subscribe({
        next(n) {
          t.next?.(n), r.next?.(n);
        },
        error(n) {
          t.error?.(n), r.error?.(n);
        },
        complete() {
          t.complete?.(), r.complete?.();
        }
      });
    }
  });
}
class $t extends Error {
  constructor(e) {
    super(e), this.name = "ObservableAbortError", Object.setPrototypeOf(this, $t.prototype);
  }
}
function Hs(t) {
  let e;
  return {
    promise: new Promise((n, s) => {
      let a = !1;
      function i() {
        a || (a = !0, s(new $t("This operation was aborted.")), o.unsubscribe());
      }
      const o = t.subscribe({
        next(c) {
          a = !0, n(c), i();
        },
        error(c) {
          a = !0, s(c), i();
        },
        complete() {
          a = !0, i();
        }
      });
      e = i;
    }),
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    abort: e
  };
}
K.isObservable = Fr.isObservable;
K.observable = Fr.observable;
K.map = Vs;
K.observableToPromise = Hs;
K.share = Fs;
K.tap = Ws;
var U = {};
Object.defineProperty(U, "__esModule", { value: !0 });
U.isTRPCRequestWithId = U.isTRPCRequest = U.isTRPCResponse = U.isTRPCMessage = void 0;
function rr(t) {
  return typeof t == "object" && t !== null && !Array.isArray(t);
}
function Js(t) {
  return t == null;
}
function Vr(t) {
  return !!(rr(t) && "trpc" in t && rr(t.trpc));
}
U.isTRPCMessage = Vr;
function jt(t) {
  return Vr(t) && "id" in t.trpc && !Js(t.trpc.id);
}
function Ys(t) {
  return jt(t) && ("error" in t.trpc || "result" in t.trpc);
}
U.isTRPCResponse = Ys;
function Wr(t) {
  return jt(t) && "method" in t.trpc;
}
U.isTRPCRequest = Wr;
function Gs(t) {
  return Wr(t) && jt(t);
}
U.isTRPCRequestWithId = Gs;
Object.defineProperty(he, "__esModule", { value: !0 });
he.createBaseLink = void 0;
const Qe = P, Qs = K, Xs = U, Ks = (t) => (e) => ({ op: r }) => (0, Qs.observable)((n) => {
  const s = [], { id: a, type: i, path: o } = r;
  try {
    const c = e.transformer.serialize(r.input), u = () => {
      n.error(new Qe.TRPCClientError("Port disconnected prematurely"));
    };
    t.addCloseListener(u), s.push(() => t.removeCloseListener(u));
    const d = (h) => {
      if (!(0, Xs.isTRPCResponse)(h))
        return;
      const { trpc: y } = h;
      if (a === y.id) {
        if ("error" in y)
          return n.error(Qe.TRPCClientError.from(y));
        n.next({
          result: Object.assign(Object.assign({}, y.result), (!y.result.type || y.result.type === "data") && {
            type: "data",
            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
            data: e.transformer.deserialize(y.result.data)
          })
        }), (i !== "subscription" || y.result.type === "stopped") && n.complete();
      }
    };
    t.addMessageListener(d), s.push(() => t.removeMessageListener(d)), t.postMessage({
      trpc: {
        id: a,
        jsonrpc: void 0,
        method: i,
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        params: { path: o, input: c }
      }
    });
  } catch (c) {
    n.error(new Qe.TRPCClientError(c instanceof Error ? c.message : "Unknown error"));
  }
  return () => {
    i === "subscription" && t.postMessage({
      trpc: {
        id: a,
        jsonrpc: void 0,
        method: "subscription.stop"
      }
    }), s.forEach((c) => c());
  };
});
he.createBaseLink = Ks;
Object.defineProperty(je, "__esModule", { value: !0 });
je.chromeLink = void 0;
const ea = he, ta = (t) => (0, ea.createBaseLink)({
  postMessage(e) {
    t.port.postMessage(e);
  },
  addMessageListener(e) {
    t.port.onMessage.addListener(e);
  },
  removeMessageListener(e) {
    t.port.onMessage.removeListener(e);
  },
  addCloseListener(e) {
    t.port.onDisconnect.addListener(e);
  },
  removeCloseListener(e) {
    t.port.onDisconnect.removeListener(e);
  }
});
je.chromeLink = ta;
var Ue = {};
Object.defineProperty(Ue, "__esModule", { value: !0 });
Ue.windowLink = void 0;
const ra = he, na = (t) => {
  var e;
  const r = /* @__PURE__ */ new Map(), n = t.window, s = (e = t.postWindow) !== null && e !== void 0 ? e : n, a = (i, o, c) => {
    try {
      n[`${i}EventListener`](o, c);
    } catch (u) {
      console.error(`Failed to ${i} ${o} listener:`, u);
    }
  };
  return (0, ra.createBaseLink)({
    postMessage(i) {
      s.postMessage(i, {
        targetOrigin: t.postOrigin
      });
    },
    addMessageListener(i) {
      const o = (c) => {
        i(c.data);
      };
      r.set(i, o), a("add", "message", o);
    },
    removeMessageListener(i) {
      const o = r.get(i);
      o && a("remove", "message", o);
    },
    addCloseListener(i) {
      a("add", "beforeunload", i);
    },
    removeCloseListener(i) {
      a("remove", "beforeunload", i);
    }
  });
};
Ue.windowLink = na;
var qe = {}, ze = {};
Object.defineProperty(ze, "__esModule", { value: !0 });
ze.TRPC_BROWSER_LOADED_EVENT = void 0;
ze.TRPC_BROWSER_LOADED_EVENT = "TRPC_BROWSER::POPUP_LOADED";
Object.defineProperty(qe, "__esModule", { value: !0 });
qe.popupLink = void 0;
const sa = ze, aa = he, ia = (t) => {
  const e = /* @__PURE__ */ new Map(), r = /* @__PURE__ */ new Set();
  let n = null;
  async function s(a) {
    if (!n || n.closed) {
      n = t.createPopup(), await Promise.race([
        // wait til window is loaded (same origin)
        new Promise((i) => {
          var o;
          try {
            (o = n?.addEventListener) === null || o === void 0 || o.call(n, "load", i);
          } catch {
          }
        }),
        // this is needed for cross-origin popups as they don't have a load event
        new Promise((i) => {
          a.addEventListener("message", (o) => {
            o.data === sa.TRPC_BROWSER_LOADED_EVENT && i();
          });
        }),
        // expect the popup to load after 15s max, in case non of the above events fire
        new Promise((i) => {
          console.warn("Could not detect if popup loading succeeded after 15s timeout, continuing anyway"), setTimeout(i, 15e3);
        })
      ]);
      try {
        if (!n.addEventListener)
          throw new Error("popupWindow.addEventListener is not a function");
        n.addEventListener("beforeunload", () => {
          n = null;
        });
      } catch {
        const o = setInterval(() => {
          n && n.closed && (n = null, r.forEach((c) => {
            c();
          }), clearInterval(o));
        }, 1e3);
      }
    }
    return n;
  }
  return (0, aa.createBaseLink)({
    async postMessage(a) {
      return (await s(t.listenWindow)).postMessage(a, {
        targetOrigin: t.postOrigin
      });
    },
    addMessageListener(a) {
      const i = (o) => {
        a(o.data);
      };
      e.set(a, i), t.listenWindow.addEventListener("message", i);
    },
    removeMessageListener(a) {
      const i = e.get(a);
      i && t.listenWindow.removeEventListener("message", i);
    },
    addCloseListener(a) {
      t.listenWindow.addEventListener("beforeunload", a), r.add(a);
    },
    removeCloseListener(a) {
      t.listenWindow.removeEventListener("beforeunload", a), r.delete(a);
    }
  });
};
qe.popupLink = ia;
(function(t) {
  var e = Ee && Ee.__createBinding || (Object.create ? function(n, s, a, i) {
    i === void 0 && (i = a);
    var o = Object.getOwnPropertyDescriptor(s, a);
    (!o || ("get" in o ? !s.__esModule : o.writable || o.configurable)) && (o = { enumerable: !0, get: function() {
      return s[a];
    } }), Object.defineProperty(n, i, o);
  } : function(n, s, a, i) {
    i === void 0 && (i = a), n[i] = s[a];
  }), r = Ee && Ee.__exportStar || function(n, s) {
    for (var a in n)
      a !== "default" && !Object.prototype.hasOwnProperty.call(s, a) && e(s, n, a);
  };
  Object.defineProperty(t, "__esModule", { value: !0 }), r(je, t), r(Ue, t), r(qe, t);
})(tt);
var x;
(function(t) {
  t.assertEqual = (s) => {
  };
  function e(s) {
  }
  t.assertIs = e;
  function r(s) {
    throw new Error();
  }
  t.assertNever = r, t.arrayToEnum = (s) => {
    const a = {};
    for (const i of s)
      a[i] = i;
    return a;
  }, t.getValidEnumValues = (s) => {
    const a = t.objectKeys(s).filter((o) => typeof s[s[o]] != "number"), i = {};
    for (const o of a)
      i[o] = s[o];
    return t.objectValues(i);
  }, t.objectValues = (s) => t.objectKeys(s).map(function(a) {
    return s[a];
  }), t.objectKeys = typeof Object.keys == "function" ? (s) => Object.keys(s) : (s) => {
    const a = [];
    for (const i in s)
      Object.prototype.hasOwnProperty.call(s, i) && a.push(i);
    return a;
  }, t.find = (s, a) => {
    for (const i of s)
      if (a(i))
        return i;
  }, t.isInteger = typeof Number.isInteger == "function" ? (s) => Number.isInteger(s) : (s) => typeof s == "number" && Number.isFinite(s) && Math.floor(s) === s;
  function n(s, a = " | ") {
    return s.map((i) => typeof i == "string" ? `'${i}'` : i).join(a);
  }
  t.joinValues = n, t.jsonStringifyReplacer = (s, a) => typeof a == "bigint" ? a.toString() : a;
})(x || (x = {}));
var nr;
(function(t) {
  t.mergeShapes = (e, r) => ({
    ...e,
    ...r
    // second overwrites first
  });
})(nr || (nr = {}));
const p = x.arrayToEnum([
  "string",
  "nan",
  "number",
  "integer",
  "float",
  "boolean",
  "date",
  "bigint",
  "symbol",
  "function",
  "undefined",
  "null",
  "array",
  "object",
  "unknown",
  "promise",
  "void",
  "never",
  "map",
  "set"
]), Y = (t) => {
  switch (typeof t) {
    case "undefined":
      return p.undefined;
    case "string":
      return p.string;
    case "number":
      return Number.isNaN(t) ? p.nan : p.number;
    case "boolean":
      return p.boolean;
    case "function":
      return p.function;
    case "bigint":
      return p.bigint;
    case "symbol":
      return p.symbol;
    case "object":
      return Array.isArray(t) ? p.array : t === null ? p.null : t.then && typeof t.then == "function" && t.catch && typeof t.catch == "function" ? p.promise : typeof Map < "u" && t instanceof Map ? p.map : typeof Set < "u" && t instanceof Set ? p.set : typeof Date < "u" && t instanceof Date ? p.date : p.object;
    default:
      return p.unknown;
  }
}, l = x.arrayToEnum([
  "invalid_type",
  "invalid_literal",
  "custom",
  "invalid_union",
  "invalid_union_discriminator",
  "invalid_enum_value",
  "unrecognized_keys",
  "invalid_arguments",
  "invalid_return_type",
  "invalid_date",
  "invalid_string",
  "too_small",
  "too_big",
  "invalid_intersection_types",
  "not_multiple_of",
  "not_finite"
]);
class H extends Error {
  get errors() {
    return this.issues;
  }
  constructor(e) {
    super(), this.issues = [], this.addIssue = (n) => {
      this.issues = [...this.issues, n];
    }, this.addIssues = (n = []) => {
      this.issues = [...this.issues, ...n];
    };
    const r = new.target.prototype;
    Object.setPrototypeOf ? Object.setPrototypeOf(this, r) : this.__proto__ = r, this.name = "ZodError", this.issues = e;
  }
  format(e) {
    const r = e || function(a) {
      return a.message;
    }, n = { _errors: [] }, s = (a) => {
      for (const i of a.issues)
        if (i.code === "invalid_union")
          i.unionErrors.map(s);
        else if (i.code === "invalid_return_type")
          s(i.returnTypeError);
        else if (i.code === "invalid_arguments")
          s(i.argumentsError);
        else if (i.path.length === 0)
          n._errors.push(r(i));
        else {
          let o = n, c = 0;
          for (; c < i.path.length; ) {
            const u = i.path[c];
            c === i.path.length - 1 ? (o[u] = o[u] || { _errors: [] }, o[u]._errors.push(r(i))) : o[u] = o[u] || { _errors: [] }, o = o[u], c++;
          }
        }
    };
    return s(this), n;
  }
  static assert(e) {
    if (!(e instanceof H))
      throw new Error(`Not a ZodError: ${e}`);
  }
  toString() {
    return this.message;
  }
  get message() {
    return JSON.stringify(this.issues, x.jsonStringifyReplacer, 2);
  }
  get isEmpty() {
    return this.issues.length === 0;
  }
  flatten(e = (r) => r.message) {
    const r = {}, n = [];
    for (const s of this.issues)
      if (s.path.length > 0) {
        const a = s.path[0];
        r[a] = r[a] || [], r[a].push(e(s));
      } else
        n.push(e(s));
    return { formErrors: n, fieldErrors: r };
  }
  get formErrors() {
    return this.flatten();
  }
}
H.create = (t) => new H(t);
const oa = (t, e) => {
  let r;
  switch (t.code) {
    case l.invalid_type:
      t.received === p.undefined ? r = "Required" : r = `Expected ${t.expected}, received ${t.received}`;
      break;
    case l.invalid_literal:
      r = `Invalid literal value, expected ${JSON.stringify(t.expected, x.jsonStringifyReplacer)}`;
      break;
    case l.unrecognized_keys:
      r = `Unrecognized key(s) in object: ${x.joinValues(t.keys, ", ")}`;
      break;
    case l.invalid_union:
      r = "Invalid input";
      break;
    case l.invalid_union_discriminator:
      r = `Invalid discriminator value. Expected ${x.joinValues(t.options)}`;
      break;
    case l.invalid_enum_value:
      r = `Invalid enum value. Expected ${x.joinValues(t.options)}, received '${t.received}'`;
      break;
    case l.invalid_arguments:
      r = "Invalid function arguments";
      break;
    case l.invalid_return_type:
      r = "Invalid function return type";
      break;
    case l.invalid_date:
      r = "Invalid date";
      break;
    case l.invalid_string:
      typeof t.validation == "object" ? "includes" in t.validation ? (r = `Invalid input: must include "${t.validation.includes}"`, typeof t.validation.position == "number" && (r = `${r} at one or more positions greater than or equal to ${t.validation.position}`)) : "startsWith" in t.validation ? r = `Invalid input: must start with "${t.validation.startsWith}"` : "endsWith" in t.validation ? r = `Invalid input: must end with "${t.validation.endsWith}"` : x.assertNever(t.validation) : t.validation !== "regex" ? r = `Invalid ${t.validation}` : r = "Invalid";
      break;
    case l.too_small:
      t.type === "array" ? r = `Array must contain ${t.exact ? "exactly" : t.inclusive ? "at least" : "more than"} ${t.minimum} element(s)` : t.type === "string" ? r = `String must contain ${t.exact ? "exactly" : t.inclusive ? "at least" : "over"} ${t.minimum} character(s)` : t.type === "number" ? r = `Number must be ${t.exact ? "exactly equal to " : t.inclusive ? "greater than or equal to " : "greater than "}${t.minimum}` : t.type === "bigint" ? r = `Number must be ${t.exact ? "exactly equal to " : t.inclusive ? "greater than or equal to " : "greater than "}${t.minimum}` : t.type === "date" ? r = `Date must be ${t.exact ? "exactly equal to " : t.inclusive ? "greater than or equal to " : "greater than "}${new Date(Number(t.minimum))}` : r = "Invalid input";
      break;
    case l.too_big:
      t.type === "array" ? r = `Array must contain ${t.exact ? "exactly" : t.inclusive ? "at most" : "less than"} ${t.maximum} element(s)` : t.type === "string" ? r = `String must contain ${t.exact ? "exactly" : t.inclusive ? "at most" : "under"} ${t.maximum} character(s)` : t.type === "number" ? r = `Number must be ${t.exact ? "exactly" : t.inclusive ? "less than or equal to" : "less than"} ${t.maximum}` : t.type === "bigint" ? r = `BigInt must be ${t.exact ? "exactly" : t.inclusive ? "less than or equal to" : "less than"} ${t.maximum}` : t.type === "date" ? r = `Date must be ${t.exact ? "exactly" : t.inclusive ? "smaller than or equal to" : "smaller than"} ${new Date(Number(t.maximum))}` : r = "Invalid input";
      break;
    case l.custom:
      r = "Invalid input";
      break;
    case l.invalid_intersection_types:
      r = "Intersection results could not be merged";
      break;
    case l.not_multiple_of:
      r = `Number must be a multiple of ${t.multipleOf}`;
      break;
    case l.not_finite:
      r = "Number must be finite";
      break;
    default:
      r = e.defaultError, x.assertNever(t);
  }
  return { message: r };
}, st = oa;
let ca = st;
function ua() {
  return ca;
}
const da = (t) => {
  const { data: e, path: r, errorMaps: n, issueData: s } = t, a = [...r, ...s.path || []], i = {
    ...s,
    path: a
  };
  if (s.message !== void 0)
    return {
      ...s,
      path: a,
      message: s.message
    };
  let o = "";
  const c = n.filter((u) => !!u).slice().reverse();
  for (const u of c)
    o = u(i, { data: e, defaultError: o }).message;
  return {
    ...s,
    path: a,
    message: o
  };
};
function f(t, e) {
  const r = ua(), n = da({
    issueData: e,
    data: t.data,
    path: t.path,
    errorMaps: [
      t.common.contextualErrorMap,
      // contextual error map is first priority
      t.schemaErrorMap,
      // then schema-bound map if available
      r,
      // then global override map
      r === st ? void 0 : st
      // then global default map
    ].filter((s) => !!s)
  });
  t.common.issues.push(n);
}
class I {
  constructor() {
    this.value = "valid";
  }
  dirty() {
    this.value === "valid" && (this.value = "dirty");
  }
  abort() {
    this.value !== "aborted" && (this.value = "aborted");
  }
  static mergeArray(e, r) {
    const n = [];
    for (const s of r) {
      if (s.status === "aborted")
        return v;
      s.status === "dirty" && e.dirty(), n.push(s.value);
    }
    return { status: e.value, value: n };
  }
  static async mergeObjectAsync(e, r) {
    const n = [];
    for (const s of r) {
      const a = await s.key, i = await s.value;
      n.push({
        key: a,
        value: i
      });
    }
    return I.mergeObjectSync(e, n);
  }
  static mergeObjectSync(e, r) {
    const n = {};
    for (const s of r) {
      const { key: a, value: i } = s;
      if (a.status === "aborted" || i.status === "aborted")
        return v;
      a.status === "dirty" && e.dirty(), i.status === "dirty" && e.dirty(), a.value !== "__proto__" && (typeof i.value < "u" || s.alwaysSet) && (n[a.value] = i.value);
    }
    return { status: e.value, value: n };
  }
}
const v = Object.freeze({
  status: "aborted"
}), be = (t) => ({ status: "dirty", value: t }), D = (t) => ({ status: "valid", value: t }), sr = (t) => t.status === "aborted", ar = (t) => t.status === "dirty", ce = (t) => t.status === "valid", Se = (t) => typeof Promise < "u" && t instanceof Promise;
var m;
(function(t) {
  t.errToObj = (e) => typeof e == "string" ? { message: e } : e || {}, t.toString = (e) => typeof e == "string" ? e : e?.message;
})(m || (m = {}));
class z {
  constructor(e, r, n, s) {
    this._cachedPath = [], this.parent = e, this.data = r, this._path = n, this._key = s;
  }
  get path() {
    return this._cachedPath.length || (Array.isArray(this._key) ? this._cachedPath.push(...this._path, ...this._key) : this._cachedPath.push(...this._path, this._key)), this._cachedPath;
  }
}
const ir = (t, e) => {
  if (ce(e))
    return { success: !0, data: e.value };
  if (!t.common.issues.length)
    throw new Error("Validation failed but no issues detected.");
  return {
    success: !1,
    get error() {
      if (this._error)
        return this._error;
      const r = new H(t.common.issues);
      return this._error = r, this._error;
    }
  };
};
function b(t) {
  if (!t)
    return {};
  const { errorMap: e, invalid_type_error: r, required_error: n, description: s } = t;
  if (e && (r || n))
    throw new Error(`Can't use "invalid_type_error" or "required_error" in conjunction with custom error map.`);
  return e ? { errorMap: e, description: s } : { errorMap: (i, o) => {
    const { message: c } = t;
    return i.code === "invalid_enum_value" ? { message: c ?? o.defaultError } : typeof o.data > "u" ? { message: c ?? n ?? o.defaultError } : i.code !== "invalid_type" ? { message: o.defaultError } : { message: c ?? r ?? o.defaultError };
  }, description: s };
}
class w {
  get description() {
    return this._def.description;
  }
  _getType(e) {
    return Y(e.data);
  }
  _getOrReturnCtx(e, r) {
    return r || {
      common: e.parent.common,
      data: e.data,
      parsedType: Y(e.data),
      schemaErrorMap: this._def.errorMap,
      path: e.path,
      parent: e.parent
    };
  }
  _processInputParams(e) {
    return {
      status: new I(),
      ctx: {
        common: e.parent.common,
        data: e.data,
        parsedType: Y(e.data),
        schemaErrorMap: this._def.errorMap,
        path: e.path,
        parent: e.parent
      }
    };
  }
  _parseSync(e) {
    const r = this._parse(e);
    if (Se(r))
      throw new Error("Synchronous parse encountered promise.");
    return r;
  }
  _parseAsync(e) {
    const r = this._parse(e);
    return Promise.resolve(r);
  }
  parse(e, r) {
    const n = this.safeParse(e, r);
    if (n.success)
      return n.data;
    throw n.error;
  }
  safeParse(e, r) {
    const n = {
      common: {
        issues: [],
        async: r?.async ?? !1,
        contextualErrorMap: r?.errorMap
      },
      path: r?.path || [],
      schemaErrorMap: this._def.errorMap,
      parent: null,
      data: e,
      parsedType: Y(e)
    }, s = this._parseSync({ data: e, path: n.path, parent: n });
    return ir(n, s);
  }
  "~validate"(e) {
    const r = {
      common: {
        issues: [],
        async: !!this["~standard"].async
      },
      path: [],
      schemaErrorMap: this._def.errorMap,
      parent: null,
      data: e,
      parsedType: Y(e)
    };
    if (!this["~standard"].async)
      try {
        const n = this._parseSync({ data: e, path: [], parent: r });
        return ce(n) ? {
          value: n.value
        } : {
          issues: r.common.issues
        };
      } catch (n) {
        n?.message?.toLowerCase()?.includes("encountered") && (this["~standard"].async = !0), r.common = {
          issues: [],
          async: !0
        };
      }
    return this._parseAsync({ data: e, path: [], parent: r }).then((n) => ce(n) ? {
      value: n.value
    } : {
      issues: r.common.issues
    });
  }
  async parseAsync(e, r) {
    const n = await this.safeParseAsync(e, r);
    if (n.success)
      return n.data;
    throw n.error;
  }
  async safeParseAsync(e, r) {
    const n = {
      common: {
        issues: [],
        contextualErrorMap: r?.errorMap,
        async: !0
      },
      path: r?.path || [],
      schemaErrorMap: this._def.errorMap,
      parent: null,
      data: e,
      parsedType: Y(e)
    }, s = this._parse({ data: e, path: n.path, parent: n }), a = await (Se(s) ? s : Promise.resolve(s));
    return ir(n, a);
  }
  refine(e, r) {
    const n = (s) => typeof r == "string" || typeof r > "u" ? { message: r } : typeof r == "function" ? r(s) : r;
    return this._refinement((s, a) => {
      const i = e(s), o = () => a.addIssue({
        code: l.custom,
        ...n(s)
      });
      return typeof Promise < "u" && i instanceof Promise ? i.then((c) => c ? !0 : (o(), !1)) : i ? !0 : (o(), !1);
    });
  }
  refinement(e, r) {
    return this._refinement((n, s) => e(n) ? !0 : (s.addIssue(typeof r == "function" ? r(n, s) : r), !1));
  }
  _refinement(e) {
    return new X({
      schema: this,
      typeName: _.ZodEffects,
      effect: { type: "refinement", refinement: e }
    });
  }
  superRefine(e) {
    return this._refinement(e);
  }
  constructor(e) {
    this.spa = this.safeParseAsync, this._def = e, this.parse = this.parse.bind(this), this.safeParse = this.safeParse.bind(this), this.parseAsync = this.parseAsync.bind(this), this.safeParseAsync = this.safeParseAsync.bind(this), this.spa = this.spa.bind(this), this.refine = this.refine.bind(this), this.refinement = this.refinement.bind(this), this.superRefine = this.superRefine.bind(this), this.optional = this.optional.bind(this), this.nullable = this.nullable.bind(this), this.nullish = this.nullish.bind(this), this.array = this.array.bind(this), this.promise = this.promise.bind(this), this.or = this.or.bind(this), this.and = this.and.bind(this), this.transform = this.transform.bind(this), this.brand = this.brand.bind(this), this.default = this.default.bind(this), this.catch = this.catch.bind(this), this.describe = this.describe.bind(this), this.pipe = this.pipe.bind(this), this.readonly = this.readonly.bind(this), this.isNullable = this.isNullable.bind(this), this.isOptional = this.isOptional.bind(this), this["~standard"] = {
      version: 1,
      vendor: "zod",
      validate: (r) => this["~validate"](r)
    };
  }
  optional() {
    return G.create(this, this._def);
  }
  nullable() {
    return pe.create(this, this._def);
  }
  nullish() {
    return this.nullable().optional();
  }
  array() {
    return q.create(this);
  }
  promise() {
    return Le.create(this, this._def);
  }
  or(e) {
    return Pe.create([this, e], this._def);
  }
  and(e) {
    return Ne.create(this, e, this._def);
  }
  transform(e) {
    return new X({
      ...b(this._def),
      schema: this,
      typeName: _.ZodEffects,
      effect: { type: "transform", transform: e }
    });
  }
  default(e) {
    const r = typeof e == "function" ? e : () => e;
    return new gt({
      ...b(this._def),
      innerType: this,
      defaultValue: r,
      typeName: _.ZodDefault
    });
  }
  brand() {
    return new La({
      typeName: _.ZodBranded,
      type: this,
      ...b(this._def)
    });
  }
  catch(e) {
    const r = typeof e == "function" ? e : () => e;
    return new _t({
      ...b(this._def),
      innerType: this,
      catchValue: r,
      typeName: _.ZodCatch
    });
  }
  describe(e) {
    const r = this.constructor;
    return new r({
      ...this._def,
      description: e
    });
  }
  pipe(e) {
    return Mt.create(this, e);
  }
  readonly() {
    return bt.create(this);
  }
  isOptional() {
    return this.safeParse(void 0).success;
  }
  isNullable() {
    return this.safeParse(null).success;
  }
}
const la = /^c[^\s-]{8,}$/i, fa = /^[0-9a-z]+$/, pa = /^[0-9A-HJKMNP-TV-Z]{26}$/i, ha = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/i, ma = /^[a-z0-9_-]{21}$/i, ya = /^[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]*$/, ga = /^[-+]?P(?!$)(?:(?:[-+]?\d+Y)|(?:[-+]?\d+[.,]\d+Y$))?(?:(?:[-+]?\d+M)|(?:[-+]?\d+[.,]\d+M$))?(?:(?:[-+]?\d+W)|(?:[-+]?\d+[.,]\d+W$))?(?:(?:[-+]?\d+D)|(?:[-+]?\d+[.,]\d+D$))?(?:T(?=[\d+-])(?:(?:[-+]?\d+H)|(?:[-+]?\d+[.,]\d+H$))?(?:(?:[-+]?\d+M)|(?:[-+]?\d+[.,]\d+M$))?(?:[-+]?\d+(?:[.,]\d+)?S)?)??$/, _a = /^(?!\.)(?!.*\.\.)([A-Z0-9_'+\-\.]*)[A-Z0-9_+-]@([A-Z0-9][A-Z0-9\-]*\.)+[A-Z]{2,}$/i, va = "^(\\p{Extended_Pictographic}|\\p{Emoji_Component})+$";
let Xe;
const ba = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])$/, wa = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\/(3[0-2]|[12]?[0-9])$/, xa = /^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))$/, ka = /^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))\/(12[0-8]|1[01][0-9]|[1-9]?[0-9])$/, Ta = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/, Ca = /^([0-9a-zA-Z-_]{4})*(([0-9a-zA-Z-_]{2}(==)?)|([0-9a-zA-Z-_]{3}(=)?))?$/, Hr = "((\\d\\d[2468][048]|\\d\\d[13579][26]|\\d\\d0[48]|[02468][048]00|[13579][26]00)-02-29|\\d{4}-((0[13578]|1[02])-(0[1-9]|[12]\\d|3[01])|(0[469]|11)-(0[1-9]|[12]\\d|30)|(02)-(0[1-9]|1\\d|2[0-8])))", Ea = new RegExp(`^${Hr}$`);
function Jr(t) {
  let e = "[0-5]\\d";
  t.precision ? e = `${e}\\.\\d{${t.precision}}` : t.precision == null && (e = `${e}(\\.\\d+)?`);
  const r = t.precision ? "+" : "?";
  return `([01]\\d|2[0-3]):[0-5]\\d(:${e})${r}`;
}
function Ra(t) {
  return new RegExp(`^${Jr(t)}$`);
}
function Oa(t) {
  let e = `${Hr}T${Jr(t)}`;
  const r = [];
  return r.push(t.local ? "Z?" : "Z"), t.offset && r.push("([+-]\\d{2}:?\\d{2})"), e = `${e}(${r.join("|")})`, new RegExp(`^${e}$`);
}
function Sa(t, e) {
  return !!((e === "v4" || !e) && ba.test(t) || (e === "v6" || !e) && xa.test(t));
}
function Pa(t, e) {
  if (!ya.test(t))
    return !1;
  try {
    const [r] = t.split(".");
    if (!r)
      return !1;
    const n = r.replace(/-/g, "+").replace(/_/g, "/").padEnd(r.length + (4 - r.length % 4) % 4, "="), s = JSON.parse(atob(n));
    return !(typeof s != "object" || s === null || "typ" in s && s?.typ !== "JWT" || !s.alg || e && s.alg !== e);
  } catch {
    return !1;
  }
}
function Na(t, e) {
  return !!((e === "v4" || !e) && wa.test(t) || (e === "v6" || !e) && ka.test(t));
}
class W extends w {
  _parse(e) {
    if (this._def.coerce && (e.data = String(e.data)), this._getType(e) !== p.string) {
      const a = this._getOrReturnCtx(e);
      return f(a, {
        code: l.invalid_type,
        expected: p.string,
        received: a.parsedType
      }), v;
    }
    const n = new I();
    let s;
    for (const a of this._def.checks)
      if (a.kind === "min")
        e.data.length < a.value && (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.too_small,
          minimum: a.value,
          type: "string",
          inclusive: !0,
          exact: !1,
          message: a.message
        }), n.dirty());
      else if (a.kind === "max")
        e.data.length > a.value && (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.too_big,
          maximum: a.value,
          type: "string",
          inclusive: !0,
          exact: !1,
          message: a.message
        }), n.dirty());
      else if (a.kind === "length") {
        const i = e.data.length > a.value, o = e.data.length < a.value;
        (i || o) && (s = this._getOrReturnCtx(e, s), i ? f(s, {
          code: l.too_big,
          maximum: a.value,
          type: "string",
          inclusive: !0,
          exact: !0,
          message: a.message
        }) : o && f(s, {
          code: l.too_small,
          minimum: a.value,
          type: "string",
          inclusive: !0,
          exact: !0,
          message: a.message
        }), n.dirty());
      } else if (a.kind === "email")
        _a.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "email",
          code: l.invalid_string,
          message: a.message
        }), n.dirty());
      else if (a.kind === "emoji")
        Xe || (Xe = new RegExp(va, "u")), Xe.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "emoji",
          code: l.invalid_string,
          message: a.message
        }), n.dirty());
      else if (a.kind === "uuid")
        ha.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "uuid",
          code: l.invalid_string,
          message: a.message
        }), n.dirty());
      else if (a.kind === "nanoid")
        ma.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "nanoid",
          code: l.invalid_string,
          message: a.message
        }), n.dirty());
      else if (a.kind === "cuid")
        la.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "cuid",
          code: l.invalid_string,
          message: a.message
        }), n.dirty());
      else if (a.kind === "cuid2")
        fa.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "cuid2",
          code: l.invalid_string,
          message: a.message
        }), n.dirty());
      else if (a.kind === "ulid")
        pa.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "ulid",
          code: l.invalid_string,
          message: a.message
        }), n.dirty());
      else if (a.kind === "url")
        try {
          new URL(e.data);
        } catch {
          s = this._getOrReturnCtx(e, s), f(s, {
            validation: "url",
            code: l.invalid_string,
            message: a.message
          }), n.dirty();
        }
      else
        a.kind === "regex" ? (a.regex.lastIndex = 0, a.regex.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "regex",
          code: l.invalid_string,
          message: a.message
        }), n.dirty())) : a.kind === "trim" ? e.data = e.data.trim() : a.kind === "includes" ? e.data.includes(a.value, a.position) || (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.invalid_string,
          validation: { includes: a.value, position: a.position },
          message: a.message
        }), n.dirty()) : a.kind === "toLowerCase" ? e.data = e.data.toLowerCase() : a.kind === "toUpperCase" ? e.data = e.data.toUpperCase() : a.kind === "startsWith" ? e.data.startsWith(a.value) || (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.invalid_string,
          validation: { startsWith: a.value },
          message: a.message
        }), n.dirty()) : a.kind === "endsWith" ? e.data.endsWith(a.value) || (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.invalid_string,
          validation: { endsWith: a.value },
          message: a.message
        }), n.dirty()) : a.kind === "datetime" ? Oa(a).test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.invalid_string,
          validation: "datetime",
          message: a.message
        }), n.dirty()) : a.kind === "date" ? Ea.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.invalid_string,
          validation: "date",
          message: a.message
        }), n.dirty()) : a.kind === "time" ? Ra(a).test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          code: l.invalid_string,
          validation: "time",
          message: a.message
        }), n.dirty()) : a.kind === "duration" ? ga.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "duration",
          code: l.invalid_string,
          message: a.message
        }), n.dirty()) : a.kind === "ip" ? Sa(e.data, a.version) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "ip",
          code: l.invalid_string,
          message: a.message
        }), n.dirty()) : a.kind === "jwt" ? Pa(e.data, a.alg) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "jwt",
          code: l.invalid_string,
          message: a.message
        }), n.dirty()) : a.kind === "cidr" ? Na(e.data, a.version) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "cidr",
          code: l.invalid_string,
          message: a.message
        }), n.dirty()) : a.kind === "base64" ? Ta.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "base64",
          code: l.invalid_string,
          message: a.message
        }), n.dirty()) : a.kind === "base64url" ? Ca.test(e.data) || (s = this._getOrReturnCtx(e, s), f(s, {
          validation: "base64url",
          code: l.invalid_string,
          message: a.message
        }), n.dirty()) : x.assertNever(a);
    return { status: n.value, value: e.data };
  }
  _regex(e, r, n) {
    return this.refinement((s) => e.test(s), {
      validation: r,
      code: l.invalid_string,
      ...m.errToObj(n)
    });
  }
  _addCheck(e) {
    return new W({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  email(e) {
    return this._addCheck({ kind: "email", ...m.errToObj(e) });
  }
  url(e) {
    return this._addCheck({ kind: "url", ...m.errToObj(e) });
  }
  emoji(e) {
    return this._addCheck({ kind: "emoji", ...m.errToObj(e) });
  }
  uuid(e) {
    return this._addCheck({ kind: "uuid", ...m.errToObj(e) });
  }
  nanoid(e) {
    return this._addCheck({ kind: "nanoid", ...m.errToObj(e) });
  }
  cuid(e) {
    return this._addCheck({ kind: "cuid", ...m.errToObj(e) });
  }
  cuid2(e) {
    return this._addCheck({ kind: "cuid2", ...m.errToObj(e) });
  }
  ulid(e) {
    return this._addCheck({ kind: "ulid", ...m.errToObj(e) });
  }
  base64(e) {
    return this._addCheck({ kind: "base64", ...m.errToObj(e) });
  }
  base64url(e) {
    return this._addCheck({
      kind: "base64url",
      ...m.errToObj(e)
    });
  }
  jwt(e) {
    return this._addCheck({ kind: "jwt", ...m.errToObj(e) });
  }
  ip(e) {
    return this._addCheck({ kind: "ip", ...m.errToObj(e) });
  }
  cidr(e) {
    return this._addCheck({ kind: "cidr", ...m.errToObj(e) });
  }
  datetime(e) {
    return typeof e == "string" ? this._addCheck({
      kind: "datetime",
      precision: null,
      offset: !1,
      local: !1,
      message: e
    }) : this._addCheck({
      kind: "datetime",
      precision: typeof e?.precision > "u" ? null : e?.precision,
      offset: e?.offset ?? !1,
      local: e?.local ?? !1,
      ...m.errToObj(e?.message)
    });
  }
  date(e) {
    return this._addCheck({ kind: "date", message: e });
  }
  time(e) {
    return typeof e == "string" ? this._addCheck({
      kind: "time",
      precision: null,
      message: e
    }) : this._addCheck({
      kind: "time",
      precision: typeof e?.precision > "u" ? null : e?.precision,
      ...m.errToObj(e?.message)
    });
  }
  duration(e) {
    return this._addCheck({ kind: "duration", ...m.errToObj(e) });
  }
  regex(e, r) {
    return this._addCheck({
      kind: "regex",
      regex: e,
      ...m.errToObj(r)
    });
  }
  includes(e, r) {
    return this._addCheck({
      kind: "includes",
      value: e,
      position: r?.position,
      ...m.errToObj(r?.message)
    });
  }
  startsWith(e, r) {
    return this._addCheck({
      kind: "startsWith",
      value: e,
      ...m.errToObj(r)
    });
  }
  endsWith(e, r) {
    return this._addCheck({
      kind: "endsWith",
      value: e,
      ...m.errToObj(r)
    });
  }
  min(e, r) {
    return this._addCheck({
      kind: "min",
      value: e,
      ...m.errToObj(r)
    });
  }
  max(e, r) {
    return this._addCheck({
      kind: "max",
      value: e,
      ...m.errToObj(r)
    });
  }
  length(e, r) {
    return this._addCheck({
      kind: "length",
      value: e,
      ...m.errToObj(r)
    });
  }
  /**
   * Equivalent to `.min(1)`
   */
  nonempty(e) {
    return this.min(1, m.errToObj(e));
  }
  trim() {
    return new W({
      ...this._def,
      checks: [...this._def.checks, { kind: "trim" }]
    });
  }
  toLowerCase() {
    return new W({
      ...this._def,
      checks: [...this._def.checks, { kind: "toLowerCase" }]
    });
  }
  toUpperCase() {
    return new W({
      ...this._def,
      checks: [...this._def.checks, { kind: "toUpperCase" }]
    });
  }
  get isDatetime() {
    return !!this._def.checks.find((e) => e.kind === "datetime");
  }
  get isDate() {
    return !!this._def.checks.find((e) => e.kind === "date");
  }
  get isTime() {
    return !!this._def.checks.find((e) => e.kind === "time");
  }
  get isDuration() {
    return !!this._def.checks.find((e) => e.kind === "duration");
  }
  get isEmail() {
    return !!this._def.checks.find((e) => e.kind === "email");
  }
  get isURL() {
    return !!this._def.checks.find((e) => e.kind === "url");
  }
  get isEmoji() {
    return !!this._def.checks.find((e) => e.kind === "emoji");
  }
  get isUUID() {
    return !!this._def.checks.find((e) => e.kind === "uuid");
  }
  get isNANOID() {
    return !!this._def.checks.find((e) => e.kind === "nanoid");
  }
  get isCUID() {
    return !!this._def.checks.find((e) => e.kind === "cuid");
  }
  get isCUID2() {
    return !!this._def.checks.find((e) => e.kind === "cuid2");
  }
  get isULID() {
    return !!this._def.checks.find((e) => e.kind === "ulid");
  }
  get isIP() {
    return !!this._def.checks.find((e) => e.kind === "ip");
  }
  get isCIDR() {
    return !!this._def.checks.find((e) => e.kind === "cidr");
  }
  get isBase64() {
    return !!this._def.checks.find((e) => e.kind === "base64");
  }
  get isBase64url() {
    return !!this._def.checks.find((e) => e.kind === "base64url");
  }
  get minLength() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "min" && (e === null || r.value > e) && (e = r.value);
    return e;
  }
  get maxLength() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "max" && (e === null || r.value < e) && (e = r.value);
    return e;
  }
}
W.create = (t) => new W({
  checks: [],
  typeName: _.ZodString,
  coerce: t?.coerce ?? !1,
  ...b(t)
});
function Aa(t, e) {
  const r = (t.toString().split(".")[1] || "").length, n = (e.toString().split(".")[1] || "").length, s = r > n ? r : n, a = Number.parseInt(t.toFixed(s).replace(".", "")), i = Number.parseInt(e.toFixed(s).replace(".", ""));
  return a % i / 10 ** s;
}
class ue extends w {
  constructor() {
    super(...arguments), this.min = this.gte, this.max = this.lte, this.step = this.multipleOf;
  }
  _parse(e) {
    if (this._def.coerce && (e.data = Number(e.data)), this._getType(e) !== p.number) {
      const a = this._getOrReturnCtx(e);
      return f(a, {
        code: l.invalid_type,
        expected: p.number,
        received: a.parsedType
      }), v;
    }
    let n;
    const s = new I();
    for (const a of this._def.checks)
      a.kind === "int" ? x.isInteger(e.data) || (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.invalid_type,
        expected: "integer",
        received: "float",
        message: a.message
      }), s.dirty()) : a.kind === "min" ? (a.inclusive ? e.data < a.value : e.data <= a.value) && (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.too_small,
        minimum: a.value,
        type: "number",
        inclusive: a.inclusive,
        exact: !1,
        message: a.message
      }), s.dirty()) : a.kind === "max" ? (a.inclusive ? e.data > a.value : e.data >= a.value) && (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.too_big,
        maximum: a.value,
        type: "number",
        inclusive: a.inclusive,
        exact: !1,
        message: a.message
      }), s.dirty()) : a.kind === "multipleOf" ? Aa(e.data, a.value) !== 0 && (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.not_multiple_of,
        multipleOf: a.value,
        message: a.message
      }), s.dirty()) : a.kind === "finite" ? Number.isFinite(e.data) || (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.not_finite,
        message: a.message
      }), s.dirty()) : x.assertNever(a);
    return { status: s.value, value: e.data };
  }
  gte(e, r) {
    return this.setLimit("min", e, !0, m.toString(r));
  }
  gt(e, r) {
    return this.setLimit("min", e, !1, m.toString(r));
  }
  lte(e, r) {
    return this.setLimit("max", e, !0, m.toString(r));
  }
  lt(e, r) {
    return this.setLimit("max", e, !1, m.toString(r));
  }
  setLimit(e, r, n, s) {
    return new ue({
      ...this._def,
      checks: [
        ...this._def.checks,
        {
          kind: e,
          value: r,
          inclusive: n,
          message: m.toString(s)
        }
      ]
    });
  }
  _addCheck(e) {
    return new ue({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  int(e) {
    return this._addCheck({
      kind: "int",
      message: m.toString(e)
    });
  }
  positive(e) {
    return this._addCheck({
      kind: "min",
      value: 0,
      inclusive: !1,
      message: m.toString(e)
    });
  }
  negative(e) {
    return this._addCheck({
      kind: "max",
      value: 0,
      inclusive: !1,
      message: m.toString(e)
    });
  }
  nonpositive(e) {
    return this._addCheck({
      kind: "max",
      value: 0,
      inclusive: !0,
      message: m.toString(e)
    });
  }
  nonnegative(e) {
    return this._addCheck({
      kind: "min",
      value: 0,
      inclusive: !0,
      message: m.toString(e)
    });
  }
  multipleOf(e, r) {
    return this._addCheck({
      kind: "multipleOf",
      value: e,
      message: m.toString(r)
    });
  }
  finite(e) {
    return this._addCheck({
      kind: "finite",
      message: m.toString(e)
    });
  }
  safe(e) {
    return this._addCheck({
      kind: "min",
      inclusive: !0,
      value: Number.MIN_SAFE_INTEGER,
      message: m.toString(e)
    })._addCheck({
      kind: "max",
      inclusive: !0,
      value: Number.MAX_SAFE_INTEGER,
      message: m.toString(e)
    });
  }
  get minValue() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "min" && (e === null || r.value > e) && (e = r.value);
    return e;
  }
  get maxValue() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "max" && (e === null || r.value < e) && (e = r.value);
    return e;
  }
  get isInt() {
    return !!this._def.checks.find((e) => e.kind === "int" || e.kind === "multipleOf" && x.isInteger(e.value));
  }
  get isFinite() {
    let e = null, r = null;
    for (const n of this._def.checks) {
      if (n.kind === "finite" || n.kind === "int" || n.kind === "multipleOf")
        return !0;
      n.kind === "min" ? (r === null || n.value > r) && (r = n.value) : n.kind === "max" && (e === null || n.value < e) && (e = n.value);
    }
    return Number.isFinite(r) && Number.isFinite(e);
  }
}
ue.create = (t) => new ue({
  checks: [],
  typeName: _.ZodNumber,
  coerce: t?.coerce || !1,
  ...b(t)
});
class de extends w {
  constructor() {
    super(...arguments), this.min = this.gte, this.max = this.lte;
  }
  _parse(e) {
    if (this._def.coerce)
      try {
        e.data = BigInt(e.data);
      } catch {
        return this._getInvalidInput(e);
      }
    if (this._getType(e) !== p.bigint)
      return this._getInvalidInput(e);
    let n;
    const s = new I();
    for (const a of this._def.checks)
      a.kind === "min" ? (a.inclusive ? e.data < a.value : e.data <= a.value) && (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.too_small,
        type: "bigint",
        minimum: a.value,
        inclusive: a.inclusive,
        message: a.message
      }), s.dirty()) : a.kind === "max" ? (a.inclusive ? e.data > a.value : e.data >= a.value) && (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.too_big,
        type: "bigint",
        maximum: a.value,
        inclusive: a.inclusive,
        message: a.message
      }), s.dirty()) : a.kind === "multipleOf" ? e.data % a.value !== BigInt(0) && (n = this._getOrReturnCtx(e, n), f(n, {
        code: l.not_multiple_of,
        multipleOf: a.value,
        message: a.message
      }), s.dirty()) : x.assertNever(a);
    return { status: s.value, value: e.data };
  }
  _getInvalidInput(e) {
    const r = this._getOrReturnCtx(e);
    return f(r, {
      code: l.invalid_type,
      expected: p.bigint,
      received: r.parsedType
    }), v;
  }
  gte(e, r) {
    return this.setLimit("min", e, !0, m.toString(r));
  }
  gt(e, r) {
    return this.setLimit("min", e, !1, m.toString(r));
  }
  lte(e, r) {
    return this.setLimit("max", e, !0, m.toString(r));
  }
  lt(e, r) {
    return this.setLimit("max", e, !1, m.toString(r));
  }
  setLimit(e, r, n, s) {
    return new de({
      ...this._def,
      checks: [
        ...this._def.checks,
        {
          kind: e,
          value: r,
          inclusive: n,
          message: m.toString(s)
        }
      ]
    });
  }
  _addCheck(e) {
    return new de({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  positive(e) {
    return this._addCheck({
      kind: "min",
      value: BigInt(0),
      inclusive: !1,
      message: m.toString(e)
    });
  }
  negative(e) {
    return this._addCheck({
      kind: "max",
      value: BigInt(0),
      inclusive: !1,
      message: m.toString(e)
    });
  }
  nonpositive(e) {
    return this._addCheck({
      kind: "max",
      value: BigInt(0),
      inclusive: !0,
      message: m.toString(e)
    });
  }
  nonnegative(e) {
    return this._addCheck({
      kind: "min",
      value: BigInt(0),
      inclusive: !0,
      message: m.toString(e)
    });
  }
  multipleOf(e, r) {
    return this._addCheck({
      kind: "multipleOf",
      value: e,
      message: m.toString(r)
    });
  }
  get minValue() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "min" && (e === null || r.value > e) && (e = r.value);
    return e;
  }
  get maxValue() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "max" && (e === null || r.value < e) && (e = r.value);
    return e;
  }
}
de.create = (t) => new de({
  checks: [],
  typeName: _.ZodBigInt,
  coerce: t?.coerce ?? !1,
  ...b(t)
});
class at extends w {
  _parse(e) {
    if (this._def.coerce && (e.data = !!e.data), this._getType(e) !== p.boolean) {
      const n = this._getOrReturnCtx(e);
      return f(n, {
        code: l.invalid_type,
        expected: p.boolean,
        received: n.parsedType
      }), v;
    }
    return D(e.data);
  }
}
at.create = (t) => new at({
  typeName: _.ZodBoolean,
  coerce: t?.coerce || !1,
  ...b(t)
});
class Te extends w {
  _parse(e) {
    if (this._def.coerce && (e.data = new Date(e.data)), this._getType(e) !== p.date) {
      const a = this._getOrReturnCtx(e);
      return f(a, {
        code: l.invalid_type,
        expected: p.date,
        received: a.parsedType
      }), v;
    }
    if (Number.isNaN(e.data.getTime())) {
      const a = this._getOrReturnCtx(e);
      return f(a, {
        code: l.invalid_date
      }), v;
    }
    const n = new I();
    let s;
    for (const a of this._def.checks)
      a.kind === "min" ? e.data.getTime() < a.value && (s = this._getOrReturnCtx(e, s), f(s, {
        code: l.too_small,
        message: a.message,
        inclusive: !0,
        exact: !1,
        minimum: a.value,
        type: "date"
      }), n.dirty()) : a.kind === "max" ? e.data.getTime() > a.value && (s = this._getOrReturnCtx(e, s), f(s, {
        code: l.too_big,
        message: a.message,
        inclusive: !0,
        exact: !1,
        maximum: a.value,
        type: "date"
      }), n.dirty()) : x.assertNever(a);
    return {
      status: n.value,
      value: new Date(e.data.getTime())
    };
  }
  _addCheck(e) {
    return new Te({
      ...this._def,
      checks: [...this._def.checks, e]
    });
  }
  min(e, r) {
    return this._addCheck({
      kind: "min",
      value: e.getTime(),
      message: m.toString(r)
    });
  }
  max(e, r) {
    return this._addCheck({
      kind: "max",
      value: e.getTime(),
      message: m.toString(r)
    });
  }
  get minDate() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "min" && (e === null || r.value > e) && (e = r.value);
    return e != null ? new Date(e) : null;
  }
  get maxDate() {
    let e = null;
    for (const r of this._def.checks)
      r.kind === "max" && (e === null || r.value < e) && (e = r.value);
    return e != null ? new Date(e) : null;
  }
}
Te.create = (t) => new Te({
  checks: [],
  coerce: t?.coerce || !1,
  typeName: _.ZodDate,
  ...b(t)
});
class it extends w {
  _parse(e) {
    if (this._getType(e) !== p.symbol) {
      const n = this._getOrReturnCtx(e);
      return f(n, {
        code: l.invalid_type,
        expected: p.symbol,
        received: n.parsedType
      }), v;
    }
    return D(e.data);
  }
}
it.create = (t) => new it({
  typeName: _.ZodSymbol,
  ...b(t)
});
class ot extends w {
  _parse(e) {
    if (this._getType(e) !== p.undefined) {
      const n = this._getOrReturnCtx(e);
      return f(n, {
        code: l.invalid_type,
        expected: p.undefined,
        received: n.parsedType
      }), v;
    }
    return D(e.data);
  }
}
ot.create = (t) => new ot({
  typeName: _.ZodUndefined,
  ...b(t)
});
class ct extends w {
  _parse(e) {
    if (this._getType(e) !== p.null) {
      const n = this._getOrReturnCtx(e);
      return f(n, {
        code: l.invalid_type,
        expected: p.null,
        received: n.parsedType
      }), v;
    }
    return D(e.data);
  }
}
ct.create = (t) => new ct({
  typeName: _.ZodNull,
  ...b(t)
});
class ut extends w {
  constructor() {
    super(...arguments), this._any = !0;
  }
  _parse(e) {
    return D(e.data);
  }
}
ut.create = (t) => new ut({
  typeName: _.ZodAny,
  ...b(t)
});
class dt extends w {
  constructor() {
    super(...arguments), this._unknown = !0;
  }
  _parse(e) {
    return D(e.data);
  }
}
dt.create = (t) => new dt({
  typeName: _.ZodUnknown,
  ...b(t)
});
class Q extends w {
  _parse(e) {
    const r = this._getOrReturnCtx(e);
    return f(r, {
      code: l.invalid_type,
      expected: p.never,
      received: r.parsedType
    }), v;
  }
}
Q.create = (t) => new Q({
  typeName: _.ZodNever,
  ...b(t)
});
class lt extends w {
  _parse(e) {
    if (this._getType(e) !== p.undefined) {
      const n = this._getOrReturnCtx(e);
      return f(n, {
        code: l.invalid_type,
        expected: p.void,
        received: n.parsedType
      }), v;
    }
    return D(e.data);
  }
}
lt.create = (t) => new lt({
  typeName: _.ZodVoid,
  ...b(t)
});
class q extends w {
  _parse(e) {
    const { ctx: r, status: n } = this._processInputParams(e), s = this._def;
    if (r.parsedType !== p.array)
      return f(r, {
        code: l.invalid_type,
        expected: p.array,
        received: r.parsedType
      }), v;
    if (s.exactLength !== null) {
      const i = r.data.length > s.exactLength.value, o = r.data.length < s.exactLength.value;
      (i || o) && (f(r, {
        code: i ? l.too_big : l.too_small,
        minimum: o ? s.exactLength.value : void 0,
        maximum: i ? s.exactLength.value : void 0,
        type: "array",
        inclusive: !0,
        exact: !0,
        message: s.exactLength.message
      }), n.dirty());
    }
    if (s.minLength !== null && r.data.length < s.minLength.value && (f(r, {
      code: l.too_small,
      minimum: s.minLength.value,
      type: "array",
      inclusive: !0,
      exact: !1,
      message: s.minLength.message
    }), n.dirty()), s.maxLength !== null && r.data.length > s.maxLength.value && (f(r, {
      code: l.too_big,
      maximum: s.maxLength.value,
      type: "array",
      inclusive: !0,
      exact: !1,
      message: s.maxLength.message
    }), n.dirty()), r.common.async)
      return Promise.all([...r.data].map((i, o) => s.type._parseAsync(new z(r, i, r.path, o)))).then((i) => I.mergeArray(n, i));
    const a = [...r.data].map((i, o) => s.type._parseSync(new z(r, i, r.path, o)));
    return I.mergeArray(n, a);
  }
  get element() {
    return this._def.type;
  }
  min(e, r) {
    return new q({
      ...this._def,
      minLength: { value: e, message: m.toString(r) }
    });
  }
  max(e, r) {
    return new q({
      ...this._def,
      maxLength: { value: e, message: m.toString(r) }
    });
  }
  length(e, r) {
    return new q({
      ...this._def,
      exactLength: { value: e, message: m.toString(r) }
    });
  }
  nonempty(e) {
    return this.min(1, e);
  }
}
q.create = (t, e) => new q({
  type: t,
  minLength: null,
  maxLength: null,
  exactLength: null,
  typeName: _.ZodArray,
  ...b(e)
});
function se(t) {
  if (t instanceof R) {
    const e = {};
    for (const r in t.shape) {
      const n = t.shape[r];
      e[r] = G.create(se(n));
    }
    return new R({
      ...t._def,
      shape: () => e
    });
  } else
    return t instanceof q ? new q({
      ...t._def,
      type: se(t.element)
    }) : t instanceof G ? G.create(se(t.unwrap())) : t instanceof pe ? pe.create(se(t.unwrap())) : t instanceof ee ? ee.create(t.items.map((e) => se(e))) : t;
}
class R extends w {
  constructor() {
    super(...arguments), this._cached = null, this.nonstrict = this.passthrough, this.augment = this.extend;
  }
  _getCached() {
    if (this._cached !== null)
      return this._cached;
    const e = this._def.shape(), r = x.objectKeys(e);
    return this._cached = { shape: e, keys: r }, this._cached;
  }
  _parse(e) {
    if (this._getType(e) !== p.object) {
      const u = this._getOrReturnCtx(e);
      return f(u, {
        code: l.invalid_type,
        expected: p.object,
        received: u.parsedType
      }), v;
    }
    const { status: n, ctx: s } = this._processInputParams(e), { shape: a, keys: i } = this._getCached(), o = [];
    if (!(this._def.catchall instanceof Q && this._def.unknownKeys === "strip"))
      for (const u in s.data)
        i.includes(u) || o.push(u);
    const c = [];
    for (const u of i) {
      const d = a[u], h = s.data[u];
      c.push({
        key: { status: "valid", value: u },
        value: d._parse(new z(s, h, s.path, u)),
        alwaysSet: u in s.data
      });
    }
    if (this._def.catchall instanceof Q) {
      const u = this._def.unknownKeys;
      if (u === "passthrough")
        for (const d of o)
          c.push({
            key: { status: "valid", value: d },
            value: { status: "valid", value: s.data[d] }
          });
      else if (u === "strict")
        o.length > 0 && (f(s, {
          code: l.unrecognized_keys,
          keys: o
        }), n.dirty());
      else if (u !== "strip")
        throw new Error("Internal ZodObject error: invalid unknownKeys value.");
    } else {
      const u = this._def.catchall;
      for (const d of o) {
        const h = s.data[d];
        c.push({
          key: { status: "valid", value: d },
          value: u._parse(
            new z(s, h, s.path, d)
            //, ctx.child(key), value, getParsedType(value)
          ),
          alwaysSet: d in s.data
        });
      }
    }
    return s.common.async ? Promise.resolve().then(async () => {
      const u = [];
      for (const d of c) {
        const h = await d.key, y = await d.value;
        u.push({
          key: h,
          value: y,
          alwaysSet: d.alwaysSet
        });
      }
      return u;
    }).then((u) => I.mergeObjectSync(n, u)) : I.mergeObjectSync(n, c);
  }
  get shape() {
    return this._def.shape();
  }
  strict(e) {
    return m.errToObj, new R({
      ...this._def,
      unknownKeys: "strict",
      ...e !== void 0 ? {
        errorMap: (r, n) => {
          const s = this._def.errorMap?.(r, n).message ?? n.defaultError;
          return r.code === "unrecognized_keys" ? {
            message: m.errToObj(e).message ?? s
          } : {
            message: s
          };
        }
      } : {}
    });
  }
  strip() {
    return new R({
      ...this._def,
      unknownKeys: "strip"
    });
  }
  passthrough() {
    return new R({
      ...this._def,
      unknownKeys: "passthrough"
    });
  }
  // const AugmentFactory =
  //   <Def extends ZodObjectDef>(def: Def) =>
  //   <Augmentation extends ZodRawShape>(
  //     augmentation: Augmentation
  //   ): ZodObject<
  //     extendShape<ReturnType<Def["shape"]>, Augmentation>,
  //     Def["unknownKeys"],
  //     Def["catchall"]
  //   > => {
  //     return new ZodObject({
  //       ...def,
  //       shape: () => ({
  //         ...def.shape(),
  //         ...augmentation,
  //       }),
  //     }) as any;
  //   };
  extend(e) {
    return new R({
      ...this._def,
      shape: () => ({
        ...this._def.shape(),
        ...e
      })
    });
  }
  /**
   * Prior to zod@1.0.12 there was a bug in the
   * inferred type of merged objects. Please
   * upgrade if you are experiencing issues.
   */
  merge(e) {
    return new R({
      unknownKeys: e._def.unknownKeys,
      catchall: e._def.catchall,
      shape: () => ({
        ...this._def.shape(),
        ...e._def.shape()
      }),
      typeName: _.ZodObject
    });
  }
  // merge<
  //   Incoming extends AnyZodObject,
  //   Augmentation extends Incoming["shape"],
  //   NewOutput extends {
  //     [k in keyof Augmentation | keyof Output]: k extends keyof Augmentation
  //       ? Augmentation[k]["_output"]
  //       : k extends keyof Output
  //       ? Output[k]
  //       : never;
  //   },
  //   NewInput extends {
  //     [k in keyof Augmentation | keyof Input]: k extends keyof Augmentation
  //       ? Augmentation[k]["_input"]
  //       : k extends keyof Input
  //       ? Input[k]
  //       : never;
  //   }
  // >(
  //   merging: Incoming
  // ): ZodObject<
  //   extendShape<T, ReturnType<Incoming["_def"]["shape"]>>,
  //   Incoming["_def"]["unknownKeys"],
  //   Incoming["_def"]["catchall"],
  //   NewOutput,
  //   NewInput
  // > {
  //   const merged: any = new ZodObject({
  //     unknownKeys: merging._def.unknownKeys,
  //     catchall: merging._def.catchall,
  //     shape: () =>
  //       objectUtil.mergeShapes(this._def.shape(), merging._def.shape()),
  //     typeName: ZodFirstPartyTypeKind.ZodObject,
  //   }) as any;
  //   return merged;
  // }
  setKey(e, r) {
    return this.augment({ [e]: r });
  }
  // merge<Incoming extends AnyZodObject>(
  //   merging: Incoming
  // ): //ZodObject<T & Incoming["_shape"], UnknownKeys, Catchall> = (merging) => {
  // ZodObject<
  //   extendShape<T, ReturnType<Incoming["_def"]["shape"]>>,
  //   Incoming["_def"]["unknownKeys"],
  //   Incoming["_def"]["catchall"]
  // > {
  //   // const mergedShape = objectUtil.mergeShapes(
  //   //   this._def.shape(),
  //   //   merging._def.shape()
  //   // );
  //   const merged: any = new ZodObject({
  //     unknownKeys: merging._def.unknownKeys,
  //     catchall: merging._def.catchall,
  //     shape: () =>
  //       objectUtil.mergeShapes(this._def.shape(), merging._def.shape()),
  //     typeName: ZodFirstPartyTypeKind.ZodObject,
  //   }) as any;
  //   return merged;
  // }
  catchall(e) {
    return new R({
      ...this._def,
      catchall: e
    });
  }
  pick(e) {
    const r = {};
    for (const n of x.objectKeys(e))
      e[n] && this.shape[n] && (r[n] = this.shape[n]);
    return new R({
      ...this._def,
      shape: () => r
    });
  }
  omit(e) {
    const r = {};
    for (const n of x.objectKeys(this.shape))
      e[n] || (r[n] = this.shape[n]);
    return new R({
      ...this._def,
      shape: () => r
    });
  }
  /**
   * @deprecated
   */
  deepPartial() {
    return se(this);
  }
  partial(e) {
    const r = {};
    for (const n of x.objectKeys(this.shape)) {
      const s = this.shape[n];
      e && !e[n] ? r[n] = s : r[n] = s.optional();
    }
    return new R({
      ...this._def,
      shape: () => r
    });
  }
  required(e) {
    const r = {};
    for (const n of x.objectKeys(this.shape))
      if (e && !e[n])
        r[n] = this.shape[n];
      else {
        let a = this.shape[n];
        for (; a instanceof G; )
          a = a._def.innerType;
        r[n] = a;
      }
    return new R({
      ...this._def,
      shape: () => r
    });
  }
  keyof() {
    return Yr(x.objectKeys(this.shape));
  }
}
R.create = (t, e) => new R({
  shape: () => t,
  unknownKeys: "strip",
  catchall: Q.create(),
  typeName: _.ZodObject,
  ...b(e)
});
R.strictCreate = (t, e) => new R({
  shape: () => t,
  unknownKeys: "strict",
  catchall: Q.create(),
  typeName: _.ZodObject,
  ...b(e)
});
R.lazycreate = (t, e) => new R({
  shape: t,
  unknownKeys: "strip",
  catchall: Q.create(),
  typeName: _.ZodObject,
  ...b(e)
});
class Pe extends w {
  _parse(e) {
    const { ctx: r } = this._processInputParams(e), n = this._def.options;
    function s(a) {
      for (const o of a)
        if (o.result.status === "valid")
          return o.result;
      for (const o of a)
        if (o.result.status === "dirty")
          return r.common.issues.push(...o.ctx.common.issues), o.result;
      const i = a.map((o) => new H(o.ctx.common.issues));
      return f(r, {
        code: l.invalid_union,
        unionErrors: i
      }), v;
    }
    if (r.common.async)
      return Promise.all(n.map(async (a) => {
        const i = {
          ...r,
          common: {
            ...r.common,
            issues: []
          },
          parent: null
        };
        return {
          result: await a._parseAsync({
            data: r.data,
            path: r.path,
            parent: i
          }),
          ctx: i
        };
      })).then(s);
    {
      let a;
      const i = [];
      for (const c of n) {
        const u = {
          ...r,
          common: {
            ...r.common,
            issues: []
          },
          parent: null
        }, d = c._parseSync({
          data: r.data,
          path: r.path,
          parent: u
        });
        if (d.status === "valid")
          return d;
        d.status === "dirty" && !a && (a = { result: d, ctx: u }), u.common.issues.length && i.push(u.common.issues);
      }
      if (a)
        return r.common.issues.push(...a.ctx.common.issues), a.result;
      const o = i.map((c) => new H(c));
      return f(r, {
        code: l.invalid_union,
        unionErrors: o
      }), v;
    }
  }
  get options() {
    return this._def.options;
  }
}
Pe.create = (t, e) => new Pe({
  options: t,
  typeName: _.ZodUnion,
  ...b(e)
});
function ft(t, e) {
  const r = Y(t), n = Y(e);
  if (t === e)
    return { valid: !0, data: t };
  if (r === p.object && n === p.object) {
    const s = x.objectKeys(e), a = x.objectKeys(t).filter((o) => s.indexOf(o) !== -1), i = { ...t, ...e };
    for (const o of a) {
      const c = ft(t[o], e[o]);
      if (!c.valid)
        return { valid: !1 };
      i[o] = c.data;
    }
    return { valid: !0, data: i };
  } else if (r === p.array && n === p.array) {
    if (t.length !== e.length)
      return { valid: !1 };
    const s = [];
    for (let a = 0; a < t.length; a++) {
      const i = t[a], o = e[a], c = ft(i, o);
      if (!c.valid)
        return { valid: !1 };
      s.push(c.data);
    }
    return { valid: !0, data: s };
  } else
    return r === p.date && n === p.date && +t == +e ? { valid: !0, data: t } : { valid: !1 };
}
class Ne extends w {
  _parse(e) {
    const { status: r, ctx: n } = this._processInputParams(e), s = (a, i) => {
      if (sr(a) || sr(i))
        return v;
      const o = ft(a.value, i.value);
      return o.valid ? ((ar(a) || ar(i)) && r.dirty(), { status: r.value, value: o.data }) : (f(n, {
        code: l.invalid_intersection_types
      }), v);
    };
    return n.common.async ? Promise.all([
      this._def.left._parseAsync({
        data: n.data,
        path: n.path,
        parent: n
      }),
      this._def.right._parseAsync({
        data: n.data,
        path: n.path,
        parent: n
      })
    ]).then(([a, i]) => s(a, i)) : s(this._def.left._parseSync({
      data: n.data,
      path: n.path,
      parent: n
    }), this._def.right._parseSync({
      data: n.data,
      path: n.path,
      parent: n
    }));
  }
}
Ne.create = (t, e, r) => new Ne({
  left: t,
  right: e,
  typeName: _.ZodIntersection,
  ...b(r)
});
class ee extends w {
  _parse(e) {
    const { status: r, ctx: n } = this._processInputParams(e);
    if (n.parsedType !== p.array)
      return f(n, {
        code: l.invalid_type,
        expected: p.array,
        received: n.parsedType
      }), v;
    if (n.data.length < this._def.items.length)
      return f(n, {
        code: l.too_small,
        minimum: this._def.items.length,
        inclusive: !0,
        exact: !1,
        type: "array"
      }), v;
    !this._def.rest && n.data.length > this._def.items.length && (f(n, {
      code: l.too_big,
      maximum: this._def.items.length,
      inclusive: !0,
      exact: !1,
      type: "array"
    }), r.dirty());
    const a = [...n.data].map((i, o) => {
      const c = this._def.items[o] || this._def.rest;
      return c ? c._parse(new z(n, i, n.path, o)) : null;
    }).filter((i) => !!i);
    return n.common.async ? Promise.all(a).then((i) => I.mergeArray(r, i)) : I.mergeArray(r, a);
  }
  get items() {
    return this._def.items;
  }
  rest(e) {
    return new ee({
      ...this._def,
      rest: e
    });
  }
}
ee.create = (t, e) => {
  if (!Array.isArray(t))
    throw new Error("You must pass an array of schemas to z.tuple([ ... ])");
  return new ee({
    items: t,
    typeName: _.ZodTuple,
    rest: null,
    ...b(e)
  });
};
class Ae extends w {
  get keySchema() {
    return this._def.keyType;
  }
  get valueSchema() {
    return this._def.valueType;
  }
  _parse(e) {
    const { status: r, ctx: n } = this._processInputParams(e);
    if (n.parsedType !== p.object)
      return f(n, {
        code: l.invalid_type,
        expected: p.object,
        received: n.parsedType
      }), v;
    const s = [], a = this._def.keyType, i = this._def.valueType;
    for (const o in n.data)
      s.push({
        key: a._parse(new z(n, o, n.path, o)),
        value: i._parse(new z(n, n.data[o], n.path, o)),
        alwaysSet: o in n.data
      });
    return n.common.async ? I.mergeObjectAsync(r, s) : I.mergeObjectSync(r, s);
  }
  get element() {
    return this._def.valueType;
  }
  static create(e, r, n) {
    return r instanceof w ? new Ae({
      keyType: e,
      valueType: r,
      typeName: _.ZodRecord,
      ...b(n)
    }) : new Ae({
      keyType: W.create(),
      valueType: e,
      typeName: _.ZodRecord,
      ...b(r)
    });
  }
}
class pt extends w {
  get keySchema() {
    return this._def.keyType;
  }
  get valueSchema() {
    return this._def.valueType;
  }
  _parse(e) {
    const { status: r, ctx: n } = this._processInputParams(e);
    if (n.parsedType !== p.map)
      return f(n, {
        code: l.invalid_type,
        expected: p.map,
        received: n.parsedType
      }), v;
    const s = this._def.keyType, a = this._def.valueType, i = [...n.data.entries()].map(([o, c], u) => ({
      key: s._parse(new z(n, o, n.path, [u, "key"])),
      value: a._parse(new z(n, c, n.path, [u, "value"]))
    }));
    if (n.common.async) {
      const o = /* @__PURE__ */ new Map();
      return Promise.resolve().then(async () => {
        for (const c of i) {
          const u = await c.key, d = await c.value;
          if (u.status === "aborted" || d.status === "aborted")
            return v;
          (u.status === "dirty" || d.status === "dirty") && r.dirty(), o.set(u.value, d.value);
        }
        return { status: r.value, value: o };
      });
    } else {
      const o = /* @__PURE__ */ new Map();
      for (const c of i) {
        const u = c.key, d = c.value;
        if (u.status === "aborted" || d.status === "aborted")
          return v;
        (u.status === "dirty" || d.status === "dirty") && r.dirty(), o.set(u.value, d.value);
      }
      return { status: r.value, value: o };
    }
  }
}
pt.create = (t, e, r) => new pt({
  valueType: e,
  keyType: t,
  typeName: _.ZodMap,
  ...b(r)
});
class le extends w {
  _parse(e) {
    const { status: r, ctx: n } = this._processInputParams(e);
    if (n.parsedType !== p.set)
      return f(n, {
        code: l.invalid_type,
        expected: p.set,
        received: n.parsedType
      }), v;
    const s = this._def;
    s.minSize !== null && n.data.size < s.minSize.value && (f(n, {
      code: l.too_small,
      minimum: s.minSize.value,
      type: "set",
      inclusive: !0,
      exact: !1,
      message: s.minSize.message
    }), r.dirty()), s.maxSize !== null && n.data.size > s.maxSize.value && (f(n, {
      code: l.too_big,
      maximum: s.maxSize.value,
      type: "set",
      inclusive: !0,
      exact: !1,
      message: s.maxSize.message
    }), r.dirty());
    const a = this._def.valueType;
    function i(c) {
      const u = /* @__PURE__ */ new Set();
      for (const d of c) {
        if (d.status === "aborted")
          return v;
        d.status === "dirty" && r.dirty(), u.add(d.value);
      }
      return { status: r.value, value: u };
    }
    const o = [...n.data.values()].map((c, u) => a._parse(new z(n, c, n.path, u)));
    return n.common.async ? Promise.all(o).then((c) => i(c)) : i(o);
  }
  min(e, r) {
    return new le({
      ...this._def,
      minSize: { value: e, message: m.toString(r) }
    });
  }
  max(e, r) {
    return new le({
      ...this._def,
      maxSize: { value: e, message: m.toString(r) }
    });
  }
  size(e, r) {
    return this.min(e, r).max(e, r);
  }
  nonempty(e) {
    return this.min(1, e);
  }
}
le.create = (t, e) => new le({
  valueType: t,
  minSize: null,
  maxSize: null,
  typeName: _.ZodSet,
  ...b(e)
});
class ht extends w {
  get schema() {
    return this._def.getter();
  }
  _parse(e) {
    const { ctx: r } = this._processInputParams(e);
    return this._def.getter()._parse({ data: r.data, path: r.path, parent: r });
  }
}
ht.create = (t, e) => new ht({
  getter: t,
  typeName: _.ZodLazy,
  ...b(e)
});
class mt extends w {
  _parse(e) {
    if (e.data !== this._def.value) {
      const r = this._getOrReturnCtx(e);
      return f(r, {
        received: r.data,
        code: l.invalid_literal,
        expected: this._def.value
      }), v;
    }
    return { status: "valid", value: e.data };
  }
  get value() {
    return this._def.value;
  }
}
mt.create = (t, e) => new mt({
  value: t,
  typeName: _.ZodLiteral,
  ...b(e)
});
function Yr(t, e) {
  return new fe({
    values: t,
    typeName: _.ZodEnum,
    ...b(e)
  });
}
class fe extends w {
  _parse(e) {
    if (typeof e.data != "string") {
      const r = this._getOrReturnCtx(e), n = this._def.values;
      return f(r, {
        expected: x.joinValues(n),
        received: r.parsedType,
        code: l.invalid_type
      }), v;
    }
    if (this._cache || (this._cache = new Set(this._def.values)), !this._cache.has(e.data)) {
      const r = this._getOrReturnCtx(e), n = this._def.values;
      return f(r, {
        received: r.data,
        code: l.invalid_enum_value,
        options: n
      }), v;
    }
    return D(e.data);
  }
  get options() {
    return this._def.values;
  }
  get enum() {
    const e = {};
    for (const r of this._def.values)
      e[r] = r;
    return e;
  }
  get Values() {
    const e = {};
    for (const r of this._def.values)
      e[r] = r;
    return e;
  }
  get Enum() {
    const e = {};
    for (const r of this._def.values)
      e[r] = r;
    return e;
  }
  extract(e, r = this._def) {
    return fe.create(e, {
      ...this._def,
      ...r
    });
  }
  exclude(e, r = this._def) {
    return fe.create(this.options.filter((n) => !e.includes(n)), {
      ...this._def,
      ...r
    });
  }
}
fe.create = Yr;
class yt extends w {
  _parse(e) {
    const r = x.getValidEnumValues(this._def.values), n = this._getOrReturnCtx(e);
    if (n.parsedType !== p.string && n.parsedType !== p.number) {
      const s = x.objectValues(r);
      return f(n, {
        expected: x.joinValues(s),
        received: n.parsedType,
        code: l.invalid_type
      }), v;
    }
    if (this._cache || (this._cache = new Set(x.getValidEnumValues(this._def.values))), !this._cache.has(e.data)) {
      const s = x.objectValues(r);
      return f(n, {
        received: n.data,
        code: l.invalid_enum_value,
        options: s
      }), v;
    }
    return D(e.data);
  }
  get enum() {
    return this._def.values;
  }
}
yt.create = (t, e) => new yt({
  values: t,
  typeName: _.ZodNativeEnum,
  ...b(e)
});
class Le extends w {
  unwrap() {
    return this._def.type;
  }
  _parse(e) {
    const { ctx: r } = this._processInputParams(e);
    if (r.parsedType !== p.promise && r.common.async === !1)
      return f(r, {
        code: l.invalid_type,
        expected: p.promise,
        received: r.parsedType
      }), v;
    const n = r.parsedType === p.promise ? r.data : Promise.resolve(r.data);
    return D(n.then((s) => this._def.type.parseAsync(s, {
      path: r.path,
      errorMap: r.common.contextualErrorMap
    })));
  }
}
Le.create = (t, e) => new Le({
  type: t,
  typeName: _.ZodPromise,
  ...b(e)
});
class X extends w {
  innerType() {
    return this._def.schema;
  }
  sourceType() {
    return this._def.schema._def.typeName === _.ZodEffects ? this._def.schema.sourceType() : this._def.schema;
  }
  _parse(e) {
    const { status: r, ctx: n } = this._processInputParams(e), s = this._def.effect || null, a = {
      addIssue: (i) => {
        f(n, i), i.fatal ? r.abort() : r.dirty();
      },
      get path() {
        return n.path;
      }
    };
    if (a.addIssue = a.addIssue.bind(a), s.type === "preprocess") {
      const i = s.transform(n.data, a);
      if (n.common.async)
        return Promise.resolve(i).then(async (o) => {
          if (r.value === "aborted")
            return v;
          const c = await this._def.schema._parseAsync({
            data: o,
            path: n.path,
            parent: n
          });
          return c.status === "aborted" ? v : c.status === "dirty" || r.value === "dirty" ? be(c.value) : c;
        });
      {
        if (r.value === "aborted")
          return v;
        const o = this._def.schema._parseSync({
          data: i,
          path: n.path,
          parent: n
        });
        return o.status === "aborted" ? v : o.status === "dirty" || r.value === "dirty" ? be(o.value) : o;
      }
    }
    if (s.type === "refinement") {
      const i = (o) => {
        const c = s.refinement(o, a);
        if (n.common.async)
          return Promise.resolve(c);
        if (c instanceof Promise)
          throw new Error("Async refinement encountered during synchronous parse operation. Use .parseAsync instead.");
        return o;
      };
      if (n.common.async === !1) {
        const o = this._def.schema._parseSync({
          data: n.data,
          path: n.path,
          parent: n
        });
        return o.status === "aborted" ? v : (o.status === "dirty" && r.dirty(), i(o.value), { status: r.value, value: o.value });
      } else
        return this._def.schema._parseAsync({ data: n.data, path: n.path, parent: n }).then((o) => o.status === "aborted" ? v : (o.status === "dirty" && r.dirty(), i(o.value).then(() => ({ status: r.value, value: o.value }))));
    }
    if (s.type === "transform")
      if (n.common.async === !1) {
        const i = this._def.schema._parseSync({
          data: n.data,
          path: n.path,
          parent: n
        });
        if (!ce(i))
          return v;
        const o = s.transform(i.value, a);
        if (o instanceof Promise)
          throw new Error("Asynchronous transform encountered during synchronous parse operation. Use .parseAsync instead.");
        return { status: r.value, value: o };
      } else
        return this._def.schema._parseAsync({ data: n.data, path: n.path, parent: n }).then((i) => ce(i) ? Promise.resolve(s.transform(i.value, a)).then((o) => ({
          status: r.value,
          value: o
        })) : v);
    x.assertNever(s);
  }
}
X.create = (t, e, r) => new X({
  schema: t,
  typeName: _.ZodEffects,
  effect: e,
  ...b(r)
});
X.createWithPreprocess = (t, e, r) => new X({
  schema: e,
  effect: { type: "preprocess", transform: t },
  typeName: _.ZodEffects,
  ...b(r)
});
class G extends w {
  _parse(e) {
    return this._getType(e) === p.undefined ? D(void 0) : this._def.innerType._parse(e);
  }
  unwrap() {
    return this._def.innerType;
  }
}
G.create = (t, e) => new G({
  innerType: t,
  typeName: _.ZodOptional,
  ...b(e)
});
class pe extends w {
  _parse(e) {
    return this._getType(e) === p.null ? D(null) : this._def.innerType._parse(e);
  }
  unwrap() {
    return this._def.innerType;
  }
}
pe.create = (t, e) => new pe({
  innerType: t,
  typeName: _.ZodNullable,
  ...b(e)
});
class gt extends w {
  _parse(e) {
    const { ctx: r } = this._processInputParams(e);
    let n = r.data;
    return r.parsedType === p.undefined && (n = this._def.defaultValue()), this._def.innerType._parse({
      data: n,
      path: r.path,
      parent: r
    });
  }
  removeDefault() {
    return this._def.innerType;
  }
}
gt.create = (t, e) => new gt({
  innerType: t,
  typeName: _.ZodDefault,
  defaultValue: typeof e.default == "function" ? e.default : () => e.default,
  ...b(e)
});
class _t extends w {
  _parse(e) {
    const { ctx: r } = this._processInputParams(e), n = {
      ...r,
      common: {
        ...r.common,
        issues: []
      }
    }, s = this._def.innerType._parse({
      data: n.data,
      path: n.path,
      parent: {
        ...n
      }
    });
    return Se(s) ? s.then((a) => ({
      status: "valid",
      value: a.status === "valid" ? a.value : this._def.catchValue({
        get error() {
          return new H(n.common.issues);
        },
        input: n.data
      })
    })) : {
      status: "valid",
      value: s.status === "valid" ? s.value : this._def.catchValue({
        get error() {
          return new H(n.common.issues);
        },
        input: n.data
      })
    };
  }
  removeCatch() {
    return this._def.innerType;
  }
}
_t.create = (t, e) => new _t({
  innerType: t,
  typeName: _.ZodCatch,
  catchValue: typeof e.catch == "function" ? e.catch : () => e.catch,
  ...b(e)
});
class vt extends w {
  _parse(e) {
    if (this._getType(e) !== p.nan) {
      const n = this._getOrReturnCtx(e);
      return f(n, {
        code: l.invalid_type,
        expected: p.nan,
        received: n.parsedType
      }), v;
    }
    return { status: "valid", value: e.data };
  }
}
vt.create = (t) => new vt({
  typeName: _.ZodNaN,
  ...b(t)
});
class La extends w {
  _parse(e) {
    const { ctx: r } = this._processInputParams(e), n = r.data;
    return this._def.type._parse({
      data: n,
      path: r.path,
      parent: r
    });
  }
  unwrap() {
    return this._def.type;
  }
}
class Mt extends w {
  _parse(e) {
    const { status: r, ctx: n } = this._processInputParams(e);
    if (n.common.async)
      return (async () => {
        const a = await this._def.in._parseAsync({
          data: n.data,
          path: n.path,
          parent: n
        });
        return a.status === "aborted" ? v : a.status === "dirty" ? (r.dirty(), be(a.value)) : this._def.out._parseAsync({
          data: a.value,
          path: n.path,
          parent: n
        });
      })();
    {
      const s = this._def.in._parseSync({
        data: n.data,
        path: n.path,
        parent: n
      });
      return s.status === "aborted" ? v : s.status === "dirty" ? (r.dirty(), {
        status: "dirty",
        value: s.value
      }) : this._def.out._parseSync({
        data: s.value,
        path: n.path,
        parent: n
      });
    }
  }
  static create(e, r) {
    return new Mt({
      in: e,
      out: r,
      typeName: _.ZodPipeline
    });
  }
}
class bt extends w {
  _parse(e) {
    const r = this._def.innerType._parse(e), n = (s) => (ce(s) && (s.value = Object.freeze(s.value)), s);
    return Se(r) ? r.then((s) => n(s)) : n(r);
  }
  unwrap() {
    return this._def.innerType;
  }
}
bt.create = (t, e) => new bt({
  innerType: t,
  typeName: _.ZodReadonly,
  ...b(e)
});
R.lazycreate;
var _;
(function(t) {
  t.ZodString = "ZodString", t.ZodNumber = "ZodNumber", t.ZodNaN = "ZodNaN", t.ZodBigInt = "ZodBigInt", t.ZodBoolean = "ZodBoolean", t.ZodDate = "ZodDate", t.ZodSymbol = "ZodSymbol", t.ZodUndefined = "ZodUndefined", t.ZodNull = "ZodNull", t.ZodAny = "ZodAny", t.ZodUnknown = "ZodUnknown", t.ZodNever = "ZodNever", t.ZodVoid = "ZodVoid", t.ZodArray = "ZodArray", t.ZodObject = "ZodObject", t.ZodUnion = "ZodUnion", t.ZodDiscriminatedUnion = "ZodDiscriminatedUnion", t.ZodIntersection = "ZodIntersection", t.ZodTuple = "ZodTuple", t.ZodRecord = "ZodRecord", t.ZodMap = "ZodMap", t.ZodSet = "ZodSet", t.ZodFunction = "ZodFunction", t.ZodLazy = "ZodLazy", t.ZodLiteral = "ZodLiteral", t.ZodEnum = "ZodEnum", t.ZodEffects = "ZodEffects", t.ZodNativeEnum = "ZodNativeEnum", t.ZodOptional = "ZodOptional", t.ZodNullable = "ZodNullable", t.ZodDefault = "ZodDefault", t.ZodCatch = "ZodCatch", t.ZodPromise = "ZodPromise", t.ZodBranded = "ZodBranded", t.ZodPipeline = "ZodPipeline", t.ZodReadonly = "ZodReadonly";
})(_ || (_ = {}));
const g = W.create, Gr = ue.create;
vt.create;
const Ia = de.create, ae = at.create;
Te.create;
it.create;
ot.create;
ct.create;
const Qr = ut.create, or = dt.create;
Q.create;
lt.create;
const L = q.create, T = R.create;
R.strictCreate;
const Be = Pe.create;
Ne.create;
const Z = ee.create, Ke = Ae.create;
pt.create;
le.create;
ht.create;
const ie = mt.create, wt = fe.create, $a = yt.create;
Le.create;
X.create;
G.create;
pe.create;
X.createWithPreprocess;
const Xr = /^0x[0-9a-f]+$/i, Kr = /^\d+$/, ja = g().min(1, "The short string cannot be empty").max(31, "The short string cannot exceed 31 characters").refine(
  (t) => !Xr.test(t),
  "The shortString should not be a hex string"
).refine(
  (t) => !Kr.test(t),
  "The shortString should not be an integer string"
), M = Be([
  g().regex(
    Xr,
    "Only hex, integers and bigint are supported in calldata"
  ),
  g().regex(
    Kr,
    "Only hex, integers and bigint are supported in calldata"
  ),
  ja,
  Gr().int("Only hex, integers and bigint are supported in calldata"),
  Ia()
]), Ce = T({
  contractAddress: g(),
  entrypoint: g(),
  calldata: L(M.or(L(M))).optional()
}), xt = L(Ce).nonempty(), en = T({
  types: Ke(
    L(
      Be([
        T({
          name: g(),
          type: ie("merkletree"),
          contains: g()
        }),
        T({
          name: g(),
          type: ie("enum"),
          contains: g()
        }),
        T({
          name: g(),
          type: g()
        })
      ])
    )
  ),
  primaryType: g(),
  domain: Ke(or()),
  message: Ke(or()).or(T({}))
}), kt = T({
  type: ie("ERC20"),
  options: T({
    address: g(),
    symbol: g().optional(),
    decimals: Gr().optional(),
    image: g().optional(),
    name: g().optional()
  })
}), Ma = Be([
  T({
    id: g(),
    chain_id: g(),
    chain_name: g(),
    rpc_urls: L(g()).optional(),
    native_currency: kt.optional(),
    block_explorer_url: L(g()).optional()
  }),
  T({
    id: g(),
    chainId: g(),
    chainName: g(),
    rpcUrls: L(g()).optional(),
    nativeCurrency: kt.optional(),
    blockExplorerUrl: L(g()).optional()
  }).transform((t) => ({
    id: t.id,
    chain_id: t.chainId,
    chain_name: t.chainName,
    rpc_urls: t.rpcUrls,
    native_currency: t.nativeCurrency,
    block_explorer_url: t.blockExplorerUrl
  }))
]), et = {
  enable: Z([
    T({
      starknetVersion: Be([ie("v3"), ie("v4"), ie("v5")]).optional()
    }).optional()
  ]).or(Z([])),
  addStarknetChain: Z([Ma]),
  switchStarknetChain: Z([
    T({
      chainId: g()
    })
  ]),
  watchAsset: Z([kt]),
  requestAccounts: Z([
    T({
      silent_mode: ae().optional()
    })
  ]),
  execute: Z([
    xt.or(Ce),
    T({
      nonce: M.optional(),
      maxFee: M.optional(),
      version: M.optional()
    }).optional()
  ]),
  signMessage: Z([en])
};
Z([
  xt.or(Ce),
  T({
    nonce: M.optional(),
    maxFee: M.optional(),
    version: M.optional()
  }).optional()
]).or(
  Z([
    xt.or(Ce),
    L(Qr()).optional(),
    T({
      nonce: M.optional(),
      maxFee: M.optional(),
      version: M.optional()
    }).optional()
  ])
);
const tn = T({
  contract_address: g(),
  entry_point: g(),
  calldata: L(M).optional()
}).transform(({ contract_address: t, entry_point: e, calldata: r }) => ({
  contractAddress: t,
  entrypoint: e,
  calldata: r || []
})), Da = L(tn).nonempty(), Za = {
  ZERO: 0,
  ONE: 1
}, Ua = T({
  address: g(),
  class_hash: g(),
  salt: g(),
  calldata: L(g()),
  sigdata: L(g()).optional(),
  //version: z.literal([0, 1]),
  version: $a(Za)
  // allow only 0 | 1, workaround since zod doesn't support literals as numbers
}), qa = T({
  callbackData: g().optional(),
  approvalRequests: L(
    T({
      tokenAddress: g(),
      amount: g(),
      spender: g()
    })
  ),
  sessionTypedData: en
}), za = T({
  account: g().array().optional(),
  chainId: g().optional(),
  signature: g().array().optional(),
  approvalTransactionHash: g().optional(),
  deploymentPayload: Qr().optional(),
  approvalRequestsCalls: L(Ce).optional(),
  errorCode: wt([
    "USER_REJECTED",
    "ACCOUNT_NOT_DEPLOYED",
    "NOT_ENOUGH_BALANCE",
    "NOT_ENOUGH_BALANCE_DEPLOYMENT",
    "GENERIC_ERROR"
  ]).optional()
}), S = Hn.create({
  isServer: !1,
  allowOutsideOfServer: !0
});
let Tt = an, Ct = "", rn = "";
const Xa = ({
  width: t = 775,
  height: e = 385,
  origin: r,
  location: n,
  atLeftBottom: s = !1
}) => {
  const a = window?.outerWidth ?? window?.innerWidth ?? window?.screen.width ?? 0, i = window?.outerHeight ?? window?.innerHeight ?? window?.screen.height ?? 0, o = window?.screenLeft ?? window?.screenX ?? 0, c = window?.screenTop ?? window?.screenY ?? 0, u = s ? 0 : o + a / 2 - t / 2, d = s ? window.screen.availHeight + 10 : c + i / 2 - e / 2;
  Tt = r ?? Tt, Ct = n ?? Ct, rn = `width=${t},height=${e},top=${d},left=${u},toolbar=no,menubar=no,scrollbars=no,location=no,status=no,popup=1`;
};
S.router({
  authorize: S.procedure.output(ae()).mutation(async () => !0),
  connect: S.procedure.mutation(async () => ""),
  connectWebwallet: S.procedure.input(
    T({
      theme: wt(["light", "dark", "auto"]).optional()
    })
  ).output(
    T({
      account: g().array().optional(),
      chainId: g().optional()
    })
  ).mutation(async () => ({})),
  connectWebwalletSSO: S.procedure.input(
    T({ token: g(), authorizedPartyId: g().optional() })
  ).output(
    T({
      account: g().array().optional(),
      chainId: g().optional()
    })
  ).mutation(async () => ({})),
  connectAndSignSession: S.procedure.input(qa).output(za).mutation(async () => ({})),
  enable: S.procedure.output(g()).mutation(async () => ""),
  execute: S.procedure.input(et.execute).output(g()).mutation(async () => ""),
  signMessage: S.procedure.input(et.signMessage).output(g().array()).mutation(async () => []),
  getLoginStatus: S.procedure.output(
    T({
      isLoggedIn: ae(),
      hasSession: ae().optional(),
      isPreauthorized: ae().optional()
    })
  ).mutation(async () => ({
    isLoggedIn: !0
  })),
  // RPC Messages
  requestAccounts: S.procedure.input(T({ silent_mode: ae().optional() })).output(g().array()).mutation(async () => []),
  requestChainId: S.procedure.output(g()).mutation(async () => ""),
  signTypedData: S.procedure.input(et.signMessage).output(g().array()).mutation(async () => []),
  getPermissions: S.procedure.output(L(wt([qt.ACCOUNTS]))).mutation(async () => [qt.ACCOUNTS]),
  addInvokeTransaction: S.procedure.input(tn.or(Da)).output(g()).mutation(async (t) => ""),
  addStarknetChain: S.procedure.mutation((t) => {
    throw Error("not implemented");
  }),
  switchStarknetChain: S.procedure.mutation((t) => {
    throw Error("not implemented");
  }),
  watchAsset: S.procedure.mutation((t) => {
    throw Error("not implemented");
  }),
  updateModal: S.procedure.subscription(async () => {
  }),
  deploymentData: S.procedure.output(Ua).mutation(async () => ({
    address: "",
    calldata: [],
    version: 0,
    class_hash: "",
    salt: ""
  }))
});
const Ka = ({
  iframe: t
}) => An({
  links: [
    Rn({
      enabled: (e) => process.env.NODE_ENV === "development" && typeof window < "u" || process.env.NODE_ENV === "development" && e.direction === "down" && e.result instanceof Error
    }),
    hn({
      condition(e) {
        if (!t && e.type === "subscription")
          throw new Error(
            "subscription is not supported without an iframe window"
          );
        return !!t;
      },
      true: tt.windowLink({
        window,
        postWindow: t,
        postOrigin: "*"
      }),
      false: tt.popupLink({
        listenWindow: window,
        createPopup: () => {
          let e = null;
          const r = document.createElement("button");
          if (r.style.display = "none", r.addEventListener("click", () => {
            e = window.open(
              `${Tt}${Ct}`,
              "popup",
              rn
            );
          }), r.click(), (async () => {
            for (; !e; )
              await new Promise((n) => setTimeout(n, 100));
          })(), !e)
            throw new Error("Could not open popup");
          return e;
        },
        postOrigin: "*"
      })
    })
  ]
});
export {
  Ha as D,
  Ya as M,
  Ja as T,
  an as a,
  Wa as g,
  Ga as m,
  Xa as s,
  Ka as t
};
