// @argent/x-ui/dist/src/features/simulation/activity/row/TokenActivitySummaryRow.js
import { jsx as t, jsxs as e } from "react/jsx-runtime";
import { ensureDecimals as k } from "../../../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/ensureDecimals.js";
import { prettifyCurrencyValue as v } from "../../../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/prettifyCurrencyValue.js";
import { prettifyTokenAmount as S } from "../../../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/prettifyTokenAmount.js";
import { getCurrencySymbol as b } from "../../../../../node_modules/.pnpm/@argent_x-shared@1.73.0_@scure_base@1.2.4_@scure_bip39@1.5.4_async-retry@1.3.3_lodash-e_b34e798a09d95058b17e0b32da902dbd/node_modules/@argent/x-shared/dist/src/tokens/getCurrencySymbol.js";
import { useTokenInfo as x } from "../../../providers/TokenProvider.js";
import { Flex as n } from "@chakra-ui/react";
import { useTranslation as T } from "react-i18next";
import { Hideable as w } from "../Hideable.js";
import { TokenActivitySummaryIcon as I } from "../icon/TokenActivitySummaryIcon.js";
import { ActivityRowSubtitle as C } from "./ActivityRowSubtitle.js";
import { getAttributes as R } from "./getAttributes.js";
import { UnknownTokenIcon as j } from "../../../../components/TokenIcon.js";
import { B3 as i } from "../../../../components/typography.generated.js";
const N = ({ summary: o, networkId: m, hideComponent: f }) => {
  var a, u;
  const { t: p } = T(), d = o.asset.tokenAddress, r = x({
    networkId: m,
    address: d
  });
  if (!r)
    return /* @__PURE__ */ t(n, { alignItems: "center", w: "full", children: /* @__PURE__ */ e(n, { gap: "2", alignItems: "center", children: [
      /* @__PURE__ */ t(j, { size: 7 }),
      /* @__PURE__ */ t(i, { children: p("tokenActivitySummaryRow.unknownToken") })
    ] }) });
  const c = o.asset.amount || 0, s = (a = o.asset.fiatAmount) == null ? void 0 : a.currencyAmount, y = b(((u = o.asset.fiatAmount) == null ? void 0 : u.currency) ?? "USD"), l = S({
    amount: c,
    decimals: k(r.decimals),
    symbol: r.symbol
  }), { color: A, prefix: g } = R(o), h = v(s, y);
  return /* @__PURE__ */ e(n, { alignItems: "center", w: "full", children: [
    /* @__PURE__ */ e(n, { gap: "2", alignItems: "center", children: [
      /* @__PURE__ */ t(I, { summary: o, size: 7, networkId: m }),
      /* @__PURE__ */ t(i, { children: r.name })
    ] }),
    /* @__PURE__ */ t(n, { ml: "auto", gap: 1, direction: "column", textAlign: "right", children: /* @__PURE__ */ e(w, { hideComponent: f, children: [
      /* @__PURE__ */ e(i, { color: A, "data-tx-value": c, "data-fe-value": l, children: [
        g,
        l
      ] }),
      s !== void 0 && /* @__PURE__ */ t(C, { children: h })
    ] }) })
  ] });
};
export {
  N as TokenActivitySummaryRow
};
