// @argent/x-ui/dist/src/features/tailwind/colors.generated.js
const r = {
  "black-0": "rgba(var(--color-black-0))",
  "black-4": "rgba(var(--color-black-4))",
  "black-6": "rgba(var(--color-black-6))",
  "black-12": "rgba(var(--color-black-12))",
  "black-30": "rgba(var(--color-black-30))",
  "black-50": "rgba(var(--color-black-50))",
  "black-100": "rgb(var(--color-black-100))",
  "deprecated-accent-accessible-green": "rgb(var(--color-deprecated-accent-accessible-green))",
  "deprecated-accent-accessible-sky-blue": "rgb(var(--color-deprecated-accent-accessible-sky-blue))",
  "deprecated-accent-accessible-yellow": "rgb(var(--color-deprecated-accent-accessible-yellow))",
  "deprecated-accent-blue": "rgb(var(--color-deprecated-accent-blue))",
  "deprecated-accent-coral": "rgb(var(--color-deprecated-accent-coral))",
  "deprecated-accent-green": "rgb(var(--color-deprecated-accent-green))",
  "deprecated-accent-hot-pink": "rgb(var(--color-deprecated-accent-hot-pink))",
  "deprecated-accent-light-orange": "rgb(var(--color-deprecated-accent-light-orange))",
  "deprecated-accent-red-on-black": "rgb(var(--color-deprecated-accent-red-text-on-black))",
  "deprecated-accent-red-on-white": "rgb(var(--color-deprecated-accent-red-text-on-white))",
  "deprecated-accent-sky-blue": "rgb(var(--color-deprecated-accent-sky-blue))",
  "deprecated-accent-yellow": "rgb(var(--color-deprecated-accent-yellow))",
  "deprecated-black": "rgb(var(--color-deprecated-black))",
  "deprecated-black-3": "rgba(var(--color-deprecated-black-3))",
  "deprecated-black-8": "rgba(var(--color-deprecated-black-8))",
  "deprecated-black-30": "rgba(var(--color-deprecated-black-30))",
  "deprecated-black-50": "rgba(var(--color-deprecated-black-50))",
  "deprecated-dark-green-shade": "rgb(var(--color-deprecated-dark-green-shade))",
  "deprecated-darker-blue-shade": "rgb(var(--color-deprecated-darker-blue-shade))",
  "deprecated-darker-orange": "rgb(var(--color-deprecated-darker-orange))",
  "deprecated-experimental-vault": "rgb(var(--color-deprecated-experimental-vault))",
  "deprecated-lighter-blue-shade": "rgb(var(--color-deprecated-lighter-blue-shade))",
  "deprecated-lighter-sky-blue": "rgb(var(--color-deprecated-lighter-sky-blue))",
  "deprecated-neutrals-100": "rgb(var(--color-deprecated-neutrals-100))",
  "deprecated-neutrals-200": "rgb(var(--color-deprecated-neutrals-200))",
  "deprecated-neutrals-300": "rgb(var(--color-deprecated-neutrals-300))",
  "deprecated-neutrals-400": "rgb(var(--color-deprecated-neutrals-400))",
  "deprecated-neutrals-500": "rgb(var(--color-deprecated-neutrals-500))",
  "deprecated-neutrals-600": "rgb(var(--color-deprecated-neutrals-600))",
  "deprecated-neutrals-700": "rgb(var(--color-deprecated-neutrals-700))",
  "deprecated-neutrals-800": "rgb(var(--color-deprecated-neutrals-800))",
  "deprecated-neutrals-900": "rgb(var(--color-deprecated-neutrals-900))",
  "deprecated-neutrals-1000": "rgb(var(--color-deprecated-neutrals-1000))",
  "deprecated-primary-accessible-orange": "rgb(var(--color-deprecated-primary-accessible-orange))",
  "deprecated-primary-orange": "rgb(var(--color-deprecated-primary-orange))",
  "deprecated-white": "rgb(var(--color-deprecated-white))",
  "deprecated-white-12": "rgba(var(--color-deprecated-white-12))",
  "deprecated-white-30": "rgba(var(--color-deprecated-white-30))",
  "deprecated-white-50": "rgba(var(--color-deprecated-white-50))",
  "neutrals-100": "rgb(var(--color-neutrals-100))",
  "neutrals-200": "rgb(var(--color-neutrals-200))",
  "neutrals-300": "rgb(var(--color-neutrals-300))",
  "neutrals-400": "rgb(var(--color-neutrals-400))",
  "neutrals-500": "rgb(var(--color-neutrals-500))",
  "neutrals-600": "rgb(var(--color-neutrals-600))",
  "neutrals-700": "rgb(var(--color-neutrals-700))",
  "neutrals-800": "rgb(var(--color-neutrals-800))",
  "neutrals-900": "rgb(var(--color-neutrals-900))",
  "neutrals-1000": "rgb(var(--color-neutrals-1000))",
  "primary-blue-200": "rgb(var(--color-primary-blue-200))",
  "primary-blue-400": "rgb(var(--color-primary-blue-400))",
  "primary-blue-600": "rgb(var(--color-primary-blue-600))",
  "primary-blue-800": "rgb(var(--color-primary-blue-800))",
  "primary-blue-1000": "rgb(var(--color-primary-blue-1000))",
  "primary-green-200": "rgb(var(--color-primary-green-200))",
  "primary-green-400": "rgb(var(--color-primary-green-400))",
  "primary-green-600": "rgb(var(--color-primary-green-600))",
  "primary-green-800": "rgb(var(--color-primary-green-800))",
  "primary-green-1000": "rgb(var(--color-primary-green-1000))",
  "primary-orange-6": "rgba(var(--color-primary-orange-6))",
  "primary-orange-18": "rgba(var(--color-primary-orange-18))",
  "primary-orange-200": "rgb(var(--color-primary-orange-200))",
  "primary-orange-300": "rgb(var(--color-primary-orange-300))",
  "primary-orange-400": "rgb(var(--color-primary-orange-400))",
  "primary-orange-600": "rgb(var(--color-primary-orange-600))",
  "primary-orange-800": "rgb(var(--color-primary-orange-800))",
  "primary-orange-1000": "rgb(var(--color-primary-orange-1000))",
  "primary-red-200": "rgb(var(--color-primary-red-200))",
  "primary-red-400": "rgb(var(--color-primary-red-400))",
  "primary-red-600": "rgb(var(--color-primary-red-600))",
  "primary-red-800": "rgb(var(--color-primary-red-800))",
  "primary-red-1000": "rgb(var(--color-primary-red-1000))",
  "primary-red-dark": "rgb(var(--color-primary-red-dark))",
  "primary-yellow-200": "rgb(var(--color-primary-yellow-200))",
  "primary-yellow-400": "rgb(var(--color-primary-yellow-400))",
  "primary-yellow-600": "rgb(var(--color-primary-yellow-600))",
  "primary-yellow-800": "rgb(var(--color-primary-yellow-800))",
  "primary-yellow-1000": "rgb(var(--color-primary-yellow-1000))",
  "screen-border": "rgba(var(--color-screen-border))",
  "secondary-clementine": "rgb(var(--color-secondary-clementine))",
  "secondary-coral": "rgb(var(--color-secondary-coral))",
  "secondary-hot-pink": "rgb(var(--color-secondary-hot-pink))",
  "secondary-sky-blue": "rgb(var(--color-secondary-sky-blue))",
  "white-0": "rgba(var(--color-white-0))",
  "white-6": "rgba(var(--color-white-6))",
  "white-12": "rgba(var(--color-white-12))",
  "white-18": "rgba(var(--color-white-18))",
  "white-30": "rgba(var(--color-white-30))",
  "white-50": "rgba(var(--color-white-50))",
  "white-100": "rgb(var(--color-white-100))",
  "accent-blue": "var(--color-accent-blue)",
  "accent-brand": "var(--color-accent-brand)",
  "accent-coral": "var(--color-accent-coral)",
  "accent-green": "var(--color-accent-green)",
  "accent-hot-pink": "var(--color-accent-hot-pink)",
  "accent-orange": "var(--color-accent-orange)",
  "accent-red": "var(--color-accent-red)",
  "accent-sky-blue": "var(--color-accent-sky-blue)",
  "accent-yellow": "var(--color-accent-yellow)",
  adaptive: "var(--color-text-adaptive)",
  brand: "var(--color-text-brand)",
  "button-danger": "var(--color-button-danger)",
  "button-danger-hover": "var(--color-button-danger-hover)",
  "button-info": "var(--color-button-info)",
  "button-info-hover": "var(--color-button-info-hover)",
  "button-pressed-ghost-overlay": "var(--color-button-pressed-ghost-overlay)",
  "button-pressed-overlay": "var(--color-button-pressed-overlay)",
  "button-primary": "var(--color-button-primary)",
  "button-primary-hover": "var(--color-button-primary-hover)",
  "button-secondary": "var(--color-button-secondary)",
  "button-secondary-hover": "var(--color-button-secondary-hover)",
  "button-tertiary": "var(--color-button-tertiary)",
  "button-tertiary-hover": "var(--color-button-tertiary-hover)",
  "button-tertiary-pressed": "var(--color-button-tertiary-pressed)",
  "button-warning": "var(--color-button-warning)",
  "button-warning-hover": "var(--color-button-warning-hover)",
  danger: "var(--color-text-danger)",
  "danger-accessible": "var(--color-text-danger-accessible)",
  "icon-adaptive": "var(--color-icon-adaptive)",
  "icon-background": "var(--color-icon-background)",
  "icon-background-brand": "var(--color-icon-background-brand)",
  "icon-brand": "var(--color-icon-brand)",
  "icon-danger": "var(--color-icon-danger)",
  "icon-default": "var(--color-icon-default)",
  "icon-info": "var(--color-icon-info)",
  "icon-secondary": "var(--color-icon-secondary)",
  "icon-stable": "var(--color-icon-stable)",
  "icon-subtle": "var(--color-icon-subtle)",
  "icon-success": "var(--color-icon-success)",
  "icon-warning": "var(--color-icon-warning)",
  info: "var(--color-text-info)",
  "info-accessible": "var(--color-text-info-accessible)",
  "ios-icon": "var(--color-ios-icon)",
  "ios-inverted": "var(--color-ios-text-inverted)",
  "ios-ios": "var(--color-ios-text)",
  "ios-keyboard-button": "var(--color-ios-keyboard-button)",
  "ios-primary": "var(--color-ios-primary)",
  "ios-surface-dark": "var(--color-ios-surface-dark)",
  "ios-surface-default": "var(--color-ios-surface-default)",
  "ios-surface-elevated": "var(--color-ios-surface-elevated)",
  link: "var(--color-text-link)",
  "link-pressed": "var(--color-text-link-pressed)",
  primary: "var(--color-text-primary)",
  secondary: "var(--color-text-secondary)",
  "secondary-web": "var(--color-text-secondary-web)",
  "shadow-default": "var(--color-shadow-default)",
  "shadow-medium": "var(--color-shadow-medium)",
  "shadow-strong": "var(--color-shadow-strong)",
  stable: "var(--color-text-stable)",
  "stroke-brand": "var(--color-stroke-brand)",
  "stroke-danger": "var(--color-stroke-danger)",
  "stroke-default": "var(--color-stroke-default)",
  "stroke-default-web": "var(--color-stroke-default-web)",
  "stroke-focused": "var(--color-stroke-focused)",
  "stroke-subtle": "var(--color-stroke-subtle)",
  "stroke-success": "var(--color-stroke-success)",
  "stroke-warning": "var(--color-stroke-warning)",
  subtle: "var(--color-text-subtle)",
  success: "var(--color-text-success)",
  "success-accessible": "var(--color-text-success-accessible)",
  "surface-brand": "var(--color-surface-brand)",
  "surface-danger-default": "var(--color-surface-danger-default)",
  "surface-danger-vibrant": "var(--color-surface-danger-vibrant)",
  "surface-default": "var(--color-surface-default)",
  "surface-default-web": "var(--color-surface-default-web)",
  "surface-elevated": "var(--color-surface-elevated)",
  "surface-elevated-web": "var(--color-surface-elevated-web)",
  "surface-info-default": "var(--color-surface-info-default)",
  "surface-info-vibrant": "var(--color-surface-info-vibrant)",
  "surface-input": "var(--color-surface-input)",
  "surface-inverted": "var(--color-surface-inverted)",
  "surface-match-mode": "var(--color-surface-match-mode)",
  "surface-pressed-overlay": "var(--color-surface-pressed-overlay)",
  "surface-scrimedge": "var(--color-surface-scrimedge)",
  "surface-stable-dark": "var(--color-surface-stable-dark)",
  "surface-stable-light": "var(--color-surface-stable-light)",
  "surface-success-default": "var(--color-surface-success-default)",
  "surface-success-vibrant": "var(--color-surface-success-vibrant)",
  "surface-sunken": "var(--color-surface-sunken)",
  "surface-surface": "var(--color-surface-text)",
  "surface-transparent": "var(--color-surface-transparent)",
  "surface-warning-default": "var(--color-surface-warning-default)",
  "surface-warning-vibrant": "var(--color-surface-warning-vibrant)",
  warning: "var(--color-text-warning)",
  "warning-accessible": "var(--color-text-warning-accessible)"
};
export {
  r as colors
};
