// @argent/x-ui/dist/src/features/tailwind/fontSize.web.generated.js
const e = {
  b1: [
    "1.125rem",
    {
      fontWeight: "600",
      lineHeight: "1.125rem"
    }
  ],
  b2: [
    "0.9375rem",
    {
      fontWeight: "600",
      lineHeight: "1rem"
    }
  ],
  b3: [
    "0.8125rem",
    {
      fontWeight: "600",
      lineHeight: "0.875rem"
    }
  ],
  button1: [
    "1.125rem",
    {
      fontWeight: "600",
      lineHeight: "1.125rem"
    }
  ],
  button2: [
    "0.9375rem",
    {
      fontWeight: "600",
      lineHeight: "1rem"
    }
  ],
  button3: [
    "0.8125rem",
    {
      fontWeight: "600",
      lineHeight: "0.875rem"
    }
  ],
  "feature-title": [
    "2.5rem",
    {
      fontWeight: "700",
      letterSpacing: "-0.075rem",
      lineHeight: "3.1875rem"
    }
  ],
  h1: [
    "2.5rem",
    {
      fontWeight: "700",
      lineHeight: "2.5625rem"
    }
  ],
  h2: [
    "2rem",
    {
      fontWeight: "700",
      lineHeight: "2.125rem"
    }
  ],
  h3: [
    "1.75rem",
    {
      fontWeight: "700",
      lineHeight: "1.75rem"
    }
  ],
  h4: [
    "1.375rem",
    {
      fontWeight: "700",
      lineHeight: "1.5625rem"
    }
  ],
  h5: [
    "1.125rem",
    {
      fontWeight: "600",
      lineHeight: "1.375rem"
    }
  ],
  "h3-medium": [
    "1.75rem",
    {
      fontWeight: "500",
      lineHeight: "1.75rem"
    }
  ],
  "h4-medium": [
    "1.375rem",
    {
      fontWeight: "500",
      lineHeight: "1.5625rem"
    }
  ],
  heading1: [
    "2.5rem",
    {
      fontWeight: "700",
      lineHeight: "2.5625rem"
    }
  ],
  heading2: [
    "2rem",
    {
      fontWeight: "700",
      lineHeight: "2.125rem"
    }
  ],
  heading3: [
    "1.75rem",
    {
      fontWeight: "700",
      lineHeight: "1.75rem"
    }
  ],
  heading4: [
    "1.375rem",
    {
      fontWeight: "700",
      lineHeight: "1.5625rem"
    }
  ],
  heading5: [
    "1.125rem",
    {
      fontWeight: "600",
      lineHeight: "1.375rem"
    }
  ],
  "heading3-medium": [
    "1.75rem",
    {
      fontWeight: "500",
      lineHeight: "1.75rem"
    }
  ],
  "heading4-medium": [
    "1.375rem",
    {
      fontWeight: "500",
      lineHeight: "1.5625rem"
    }
  ],
  "headings-feature-title": [
    "2.5rem",
    {
      fontWeight: "700",
      letterSpacing: "-0.075rem",
      lineHeight: "3.1875rem"
    }
  ],
  l1: [
    "0.875rem",
    {
      fontWeight: "400",
      lineHeight: "1.125rem"
    }
  ],
  l2: [
    "0.75rem",
    {
      fontWeight: "400",
      lineHeight: "1rem"
    }
  ],
  l3: [
    "0.6875rem",
    {
      fontWeight: "400",
      lineHeight: "0.875rem"
    }
  ],
  "l1-bold": [
    "0.875rem",
    {
      fontWeight: "600",
      lineHeight: "1.125rem"
    }
  ],
  "l2-bold": [
    "0.75rem",
    {
      fontWeight: "600",
      lineHeight: "1rem"
    }
  ],
  "l3-bold": [
    "0.6875rem",
    {
      fontWeight: "600",
      lineHeight: "0.875rem"
    }
  ],
  label1: [
    "0.875rem",
    {
      fontWeight: "400",
      lineHeight: "1.125rem"
    }
  ],
  label2: [
    "0.75rem",
    {
      fontWeight: "400",
      lineHeight: "1rem"
    }
  ],
  label3: [
    "0.6875rem",
    {
      fontWeight: "400",
      lineHeight: "0.875rem"
    }
  ],
  "label1-bold": [
    "0.875rem",
    {
      fontWeight: "600",
      lineHeight: "1.125rem"
    }
  ],
  "label2-bold": [
    "0.75rem",
    {
      fontWeight: "600",
      lineHeight: "1rem"
    }
  ],
  "label3-bold": [
    "0.6875rem",
    {
      fontWeight: "600",
      lineHeight: "0.875rem"
    }
  ],
  "massive-title": [
    "4.5rem",
    {
      fontWeight: "700",
      lineHeight: "3.5rem"
    }
  ],
  p1: [
    "1.25rem",
    {
      fontWeight: "400",
      lineHeight: "1.625rem"
    }
  ],
  p2: [
    "1.125rem",
    {
      fontWeight: "400",
      lineHeight: "1.375rem"
    }
  ],
  p3: [
    "1rem",
    {
      fontWeight: "400",
      lineHeight: "1.3125rem"
    }
  ],
  p4: [
    "0.9375rem",
    {
      fontWeight: "400",
      lineHeight: "1.25rem"
    }
  ],
  "p2-bold": [
    "1.125rem",
    {
      fontWeight: "600",
      lineHeight: "1.375rem"
    }
  ],
  "p3-bold": [
    "1rem",
    {
      fontWeight: "600",
      lineHeight: "1.3125rem"
    }
  ],
  "p4-bold": [
    "0.9375rem",
    {
      fontWeight: "600",
      lineHeight: "1.25rem"
    }
  ],
  paragraph1: [
    "1.25rem",
    {
      fontWeight: "400",
      lineHeight: "1.625rem"
    }
  ],
  paragraph2: [
    "1.125rem",
    {
      fontWeight: "400",
      lineHeight: "1.375rem"
    }
  ],
  paragraph3: [
    "1rem",
    {
      fontWeight: "400",
      lineHeight: "1.3125rem"
    }
  ],
  paragraph4: [
    "0.9375rem",
    {
      fontWeight: "400",
      lineHeight: "1.25rem"
    }
  ],
  "paragraph2-bold": [
    "1.125rem",
    {
      fontWeight: "600",
      lineHeight: "1.375rem"
    }
  ],
  "paragraph3-bold": [
    "1rem",
    {
      fontWeight: "600",
      lineHeight: "1.3125rem"
    }
  ],
  "paragraph4-bold": [
    "0.9375rem",
    {
      fontWeight: "600",
      lineHeight: "1.25rem"
    }
  ]
};
export {
  e as fontSize
};
